// @revolut/ui-kit@11.21.2/esm/legacy/components/AvatarCircle/sizes.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _AVATAR_CIRCLE_SIZE_S;

import { rem } from 'polished';
import { AvatarSize, Size } from "../../../constants";
export var AvatarCircleSize = {
  DEFAULT: AvatarSize.DEFAULT,
  XS: AvatarSize.XS,
  SM: AvatarSize.SM,
  MD: AvatarSize.MD,
  LG: AvatarSize.LG,
  XL: AvatarSize.XL,
  XXL: AvatarSize.XXL
};
var AVATAR_DEFAULT = {
  size: Size.AVATAR_DEFAULT,
  fontSize: rem(24)
};
var AVATAR_XS = {
  size: Size.AVATAR_XS,
  fontSize: rem(12)
};
var AVATAR_SM = {
  size: [Size.AVATAR_SM, null, Size.AVATAR_MD],
  fontSize: [rem(20), null, rem(24)]
};
var AVATAR_MD = {
  size: [Size.AVATAR_MD, null, Size.AVATAR_MD_PLUS],
  fontSize: [rem(24), null, rem(28)]
};
var AVATAR_LG = {
  size: [Size.AVATAR_LG, null, Size.AVATAR_LG_PLUS],
  fontSize: rem(32)
};
var AVATAR_XL = {
  size: Size.AVATAR_XL,
  fontSize: rem(48)
};
var AVATAR_XXL = {
  size: [Size.AVATAR_XL, null, Size.AVATAR_XXL],
  fontSize: [rem(48), null, rem(64)]
};
export var AVATAR_CIRCLE_SIZE_STYLES = (_AVATAR_CIRCLE_SIZE_S = {}, _AVATAR_CIRCLE_SIZE_S[AvatarCircleSize.DEFAULT] = AVATAR_DEFAULT, _AVATAR_CIRCLE_SIZE_S[AvatarCircleSize.XS] = AVATAR_XS, _AVATAR_CIRCLE_SIZE_S[AvatarCircleSize.SM] = AVATAR_SM, _AVATAR_CIRCLE_SIZE_S[AvatarCircleSize.MD] = AVATAR_MD, _AVATAR_CIRCLE_SIZE_S[AvatarCircleSize.LG] = AVATAR_LG, _AVATAR_CIRCLE_SIZE_S[AvatarCircleSize.XL] = AVATAR_XL, _AVATAR_CIRCLE_SIZE_S[AvatarCircleSize.XXL] = AVATAR_XXL, _AVATAR_CIRCLE_SIZE_S);
