// @revolut/ui-kit@11.21.2/esm/components/Balance/BlinkingText.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled, { keyframes } from 'styled-components';
import { Color } from "../../constants";
import { themeColor } from "../../utils";
import { Text } from "../Text";

var blink = function blink(color, blinkColor) {
  return keyframes(["0%,100%{color:", ";}50%{color:", ";}"], color, blinkColor);
};

export var BlinkingType;

(function (BlinkingType) {
  BlinkingType["NONE"] = "NONE";
  BlinkingType["POSITIVE"] = "POSITIVE";
  BlinkingType["NEGATIVE"] = "NEGATIVE";
})(BlinkingType || (BlinkingType = {}));

export var BlinkingText = /*#__PURE__*/styled(Text).withConfig({
  displayName: "BlinkingText",
  componentId: "sc-1cx8l5m-0"
})(["animation-name:", ";animation-duration:1000ms;animation-timing-function:linear;animation-iteration-count:initial;"], function (props) {
  if (props.blinkingType === BlinkingType.NONE) {
    return 'none';
  }

  return blink(themeColor(Color.FOREGROUND)(props), themeColor(props.blinkingType === BlinkingType.POSITIVE ? Color.BLUE : Color.PINK)(props));
});
