// @revolut/ui-kit@11.21.2/esm/components/Slider/Thumb.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import React from 'react';
import styled from 'styled-components';
import { themeColor } from "../../utils";
import { Color } from "../../constants";
import { Box } from '..';
import { THUMB_SIZE } from "./constants";
var Inner = /*#__PURE__*/styled(Box).withConfig({
  displayName: "Thumb__Inner",
  componentId: "sksd3t-0"
})(["--slider-shadow-focus:0 0 0 2px ", ",0 0 0 4px ", ";", ""], themeColor(Color.BACKGROUND), themeColor(Color.BLUE_OPAQUE_50), function (_ref) {
  var focused = _ref.focused;
  return focused ? 'box-shadow: var(--slider-shadow-focus);' : '';
});
var Outer = /*#__PURE__*/styled.div.attrs(function (_ref2) {
  var left = _ref2.left;
  return {
    style: {
      left: left
    }
  };
}).withConfig({
  displayName: "Thumb__Outer",
  componentId: "sksd3t-1"
})(["position:absolute;pointer-events:none;"]);
export var Thumb = function Thumb(_ref3) {
  var position = _ref3.position,
      _ref3$focused = _ref3.focused,
      focused = _ref3$focused === void 0 ? false : _ref3$focused;
  return /*#__PURE__*/React.createElement(Outer, {
    left: position
  }, /*#__PURE__*/React.createElement(Inner, {
    focused: focused,
    width: THUMB_SIZE,
    height: THUMB_SIZE,
    ml: -THUMB_SIZE / 2,
    borderRadius: "50%",
    bg: Color.BLUE
  }));
};
