// @revolut/ui-kit@11.21.2/esm/components/Calendar/CalendarNavbar.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import React from 'react';
import * as Icons from '@revolut/icons';
import styled from 'styled-components';
import css from '@styled-system/css';
import { rem } from 'polished';
import { Color, Space } from "../../constants";
import { Absolute } from "../Position";
import { IconButton } from "../IconButton";
import { HEADING_HEIGHT } from "./constants";
var Container = /*#__PURE__*/styled(Absolute).withConfig({
  displayName: "CalendarNavbar__Container",
  componentId: "sc-186tdbq-0"
})(css({
  display: 'flex',
  justifyContent: 'space-between',
  width: '100%',
  alignItems: 'center',
  height: rem(HEADING_HEIGHT),
  top: Space.S_16,
  color: Color.GREY_TONE_20,
  zIndex: 1,
  pointerEvents: 'none',
  '*': {
    pointerEvents: 'auto'
  }
}));
export var CalendarNavbar = function CalendarNavbar(_ref) {
  var onPreviousClick = _ref.onPreviousClick,
      onNextClick = _ref.onNextClick,
      labels = _ref.labels;
  return /*#__PURE__*/React.createElement(Container, null, /*#__PURE__*/React.createElement(IconButton, {
    useIcon: Icons.ShevronLeft,
    onClick: function onClick() {
      return onPreviousClick();
    },
    "aria-label": labels.previousMonth
  }), /*#__PURE__*/React.createElement(IconButton, {
    useIcon: Icons.ShevronRight,
    onClick: function onClick() {
      return onNextClick();
    },
    "aria-label": labels.nextMonth
  }));
};
