// @revolut/ui-kit@11.21.2/esm/components/Calendar/CalendarYear.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import React, { useMemo, useState } from 'react';
import { DateUtils } from 'react-day-picker';
import { useControlled } from "../../hooks";
import { Space } from "../../constants";
import { CalendarHeader } from "./CalendarHeader";
import { CalendarNavbar } from "./CalendarNavbar";
import { CalendarCell } from "./CalendarCell";
import { TableContainer } from "./TableContainer";
import { INITIAL_YEAR_OFFSET, MONTHS_IN_YEAR, YEAR_PICKER_YEARS_QUANTITY, classNames, modifiers } from "./constants";
import { useBaseCalendarLabels, useYearsLayout } from "./hooks";
import { CalendarWrapper } from "./styles";
import { isSameYear } from "./utils";
export function CalendarYear(_ref) {
  var controlledValue = _ref.value,
      _ref$defaultValue = _ref.defaultValue,
      defaultValue = _ref$defaultValue === void 0 ? null : _ref$defaultValue,
      onChange = _ref.onChange,
      labelEmptyValue = _ref.labelEmptyValue,
      labelButtonClear = _ref.labelButtonClear,
      labelPrev = _ref.labelPrev,
      labelNext = _ref.labelNext,
      disabledDays = _ref.disabledDays,
      hideHeader = _ref.hideHeader;

  var _useControlled = useControlled({
    value: controlledValue,
    defaultValue: defaultValue,
    onChange: onChange
  }),
      value = _useControlled[0],
      setValue = _useControlled[1];

  var _useState = useState(value ? new Date(value.getFullYear() - INITIAL_YEAR_OFFSET, 0, 1) : new Date(new Date().getFullYear() - INITIAL_YEAR_OFFSET, 0, 1)),
      periodStart = _useState[0],
      setPeriodStart = _useState[1];

  var yearsLayout = useYearsLayout(periodStart);
  var modifiersObj = useMemo(function () {
    var _ref2;

    return _ref2 = {}, _ref2[classNames.disabled] = disabledDays, _ref2[classNames.today] = function (date) {
      return date.getFullYear() === new Date().getFullYear();
    }, _ref2[modifiers.SELECTED_CUSTOM] = function (date) {
      return value ? isSameYear(value, date) : false;
    }, _ref2;
  }, [disabledDays, value]);
  var labels = useBaseCalendarLabels({
    labelPrev: labelPrev,
    labelNext: labelNext
  });
  return /*#__PURE__*/React.createElement(CalendarWrapper, null, /*#__PURE__*/React.createElement(CalendarHeader, {
    hideHeader: hideHeader,
    valueLabel: value && value.getFullYear(),
    labelEmptyValue: labelEmptyValue,
    labelButtonClear: labelButtonClear,
    onClear: function onClear() {
      return setValue(null);
    }
  }), /*#__PURE__*/React.createElement(TableContainer, null, /*#__PURE__*/React.createElement(TableContainer.InnerWrapper, null, /*#__PURE__*/React.createElement(CalendarNavbar, {
    onPreviousClick: function onPreviousClick() {
      return setPeriodStart(DateUtils.addMonths(periodStart, -MONTHS_IN_YEAR * YEAR_PICKER_YEARS_QUANTITY));
    },
    onNextClick: function onNextClick() {
      return setPeriodStart(DateUtils.addMonths(periodStart, MONTHS_IN_YEAR * YEAR_PICKER_YEARS_QUANTITY));
    },
    labels: labels
  }), /*#__PURE__*/React.createElement(TableContainer.Table, null, /*#__PURE__*/React.createElement(TableContainer.Caption, null, periodStart.getFullYear() + " - " + (periodStart.getFullYear() + YEAR_PICKER_YEARS_QUANTITY - 1)), /*#__PURE__*/React.createElement(TableContainer.TableBody, null, yearsLayout.map(function (row) {
    return /*#__PURE__*/React.createElement(TableContainer.TableRow, {
      key: row[0].title
    }, row.map(function (_ref3) {
      var title = _ref3.title,
          yearValue = _ref3.yearValue;
      return /*#__PURE__*/React.createElement(TableContainer.TableCell, {
        key: title,
        value: yearValue,
        onChange: setValue,
        label: title,
        modifiersObj: modifiersObj
      }, /*#__PURE__*/React.createElement(CalendarCell, {
        width: "fit-content",
        px: Space.S_8
      }, title));
    }));
  }))))));
}
