// @revolut/ui-kit@11.21.2/esm/components/Calendar/constants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

export var WEEKDAY_HEIGHT = 56;
export var HEADING_HEIGHT = 36;
export var DEFAULT_FIRST_DAY_OF_WEEK = 1;
export var DATE_OF_MONTH_INITIAL_MONTH = new Date('1973-01-01');
export var MONTH_PICKER_MONTHS_AT_ROW = 4;
export var YEAR_PICKER_YEARS_AT_ROW = 4;
export var YEAR_PICKER_YEARS_QUANTITY = 12;
export var MONTHS_IN_YEAR = 12;
export var INITIAL_YEAR_OFFSET = 5;
export var classNames = {
  container: 'DayPicker',
  wrapper: 'DayPicker-wrapper',
  interactionDisabled: 'DayPicker--interactionDisabled',
  months: 'DayPicker-Months',
  month: 'DayPicker-Month',
  navBar: 'DayPicker-NavBar',
  navButtonPrev: 'DayPicker-NavButton DayPicker-NavButton--prev',
  navButtonNext: 'DayPicker-NavButton DayPicker-NavButton--next',
  navButtonInteractionDisabled: 'DayPicker-NavButton--interactionDisabled',
  caption: 'DayPicker-Caption',
  weekdays: 'DayPicker-Weekdays',
  weekdaysRow: 'DayPicker-WeekdaysRow',
  weekday: 'DayPicker-Weekday',
  body: 'DayPicker-Body',
  week: 'DayPicker-Week',
  weekNumber: 'DayPicker-WeekNumber',
  day: 'DayPicker-Day',
  footer: 'DayPicker-Footer',
  todayButton: 'DayPicker-TodayButton',
  // default modifiers
  today: 'today',
  selected: 'selected',
  disabled: 'disabled',
  outside: 'outside'
};
export var modifiers = {
  SELECTED_CUSTOM: 'selected-custom',
  RANGE: 'range',
  RANGE_FROM: 'range-from',
  RANGE_TO: 'range-to',
  WEEK_START: 'week-start',
  WEEK_END: 'week-end',
  MONTH_START: 'month-start',
  MONTH_END: 'month-end'
};
