// @revolut/ui-kit@11.21.2/esm/components/Calendar/TableContainer.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import React from 'react';
import styled from 'styled-components';
import { ModifiersUtils } from 'react-day-picker';
import { KeyboardEventKey, Space, TextStyle } from "../../constants";
import { themeSpace, themeTextStyle } from "../../utils";
import { CAPTION_STYLE, CONTAINER_STYLE, TABLE_BODY_STYLE, TABLE_CELL_STYLE, TABLE_ROW_STYLE, TABLE_STYLE, WRAPPER_STYLE } from "./styles";
import { classNames, modifiers } from "./constants";
var Container = /*#__PURE__*/styled.div.withConfig({
  displayName: "TableContainer__Container",
  componentId: "sc-1qquewf-0"
})(CONTAINER_STYLE);
var InnerWrapper = /*#__PURE__*/styled.div.withConfig({
  displayName: "TableContainer__InnerWrapper",
  componentId: "sc-1qquewf-1"
})(WRAPPER_STYLE);
var CaptionStyled = /*#__PURE__*/styled.div.withConfig({
  displayName: "TableContainer__CaptionStyled",
  componentId: "sc-1qquewf-2"
})(["", " ", " margin-bottom:", ";"], CAPTION_STYLE, themeTextStyle(TextStyle.PRIMARY), themeSpace(Space.S_8));
CaptionStyled.defaultProps = {
  role: 'heading'
};

var Caption = function Caption(_ref) {
  var children = _ref.children;
  return /*#__PURE__*/React.createElement(CaptionStyled, null, /*#__PURE__*/React.createElement("div", null, children));
};

var Table = /*#__PURE__*/styled.div.withConfig({
  displayName: "TableContainer__Table",
  componentId: "sc-1qquewf-3"
})(TABLE_STYLE);
Table.defaultProps = {
  role: 'grid'
};
var TableBody = /*#__PURE__*/styled.div.withConfig({
  displayName: "TableContainer__TableBody",
  componentId: "sc-1qquewf-4"
})(TABLE_BODY_STYLE);
TableBody.defaultProps = {
  role: 'rowgroup'
};
var TableRow = /*#__PURE__*/styled.div.withConfig({
  displayName: "TableContainer__TableRow",
  componentId: "sc-1qquewf-5"
})(TABLE_ROW_STYLE);
TableRow.defaultProps = {
  role: 'row'
};
var TableCellBase = /*#__PURE__*/styled.div.withConfig({
  displayName: "TableContainer__TableCellBase",
  componentId: "sc-1qquewf-6"
})(["", " padding-top:", ";"], TABLE_CELL_STYLE, themeSpace(Space.S_8));
TableCellBase.defaultProps = {
  role: 'gridcell'
};

var TableCell = function TableCell(_ref2) {
  var children = _ref2.children,
      label = _ref2.label,
      modifiersObj = _ref2.modifiersObj,
      onChange = _ref2.onChange,
      value = _ref2.value;
  var dayModifiers = ModifiersUtils.getModifiersForDay(value, modifiersObj);
  var disabled = dayModifiers.indexOf(classNames.disabled) > -1;

  var onClickHandler = function onClickHandler() {
    if (disabled) {
      return;
    }

    onChange(value);
  };

  var onKeyDownHandler = function onKeyDownHandler(event) {
    if (event.key === KeyboardEventKey.ENTER || event.key === KeyboardEventKey.SPACE) {
      onClickHandler();
    }
  };

  return /*#__PURE__*/React.createElement(TableCellBase, {
    onClick: onClickHandler,
    onKeyDown: onKeyDownHandler,
    "aria-selected": dayModifiers.indexOf(modifiers.SELECTED_CUSTOM) > -1,
    "aria-label": label,
    "aria-disabled": disabled,
    tabIndex: disabled ? -1 : 0,
    className: dayModifiers.join(' ')
  }, children);
};

export var TableContainer = Object.assign(Container, {
  InnerWrapper: InnerWrapper,
  Caption: Caption,
  Table: Table,
  TableBody: TableBody,
  TableRow: TableRow,
  TableCell: TableCell
});
