// @revolut/ui-kit@11.21.2/esm/components/Calendar/Weekdays.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import React, { useMemo } from 'react';
import { addDays, startOfWeek } from 'date-fns';
import { Color, Size, Space, TextStyle } from "../../constants";
import { useSupportedLocale } from "../../hooks";
import { Flex } from "../Flex";
import { useFormatDayOfWeekHead } from "./hooks";

function getWeekdays(firstDayOfWeek) {
  var start = startOfWeek(new Date(), {
    weekStartsOn: firstDayOfWeek
  });
  var days = [];

  for (var i = 0; i < 7; i++) {
    var day = addDays(start, i);
    days.push(day);
  }

  return days;
}

export function Weekdays(_ref) {
  var localeProp = _ref.locale,
      firstDayOfWeek = _ref.firstDayOfWeek;
  var locale = useSupportedLocale(localeProp, [Intl.DateTimeFormat]);
  var weekdays = useMemo(function () {
    return getWeekdays(firstDayOfWeek);
  }, [firstDayOfWeek]);
  var format = useFormatDayOfWeekHead(locale);
  return /*#__PURE__*/React.createElement(Flex, {
    justifyContent: "space-around",
    alignItems: "center",
    bg: Color.GREY_TONE_2,
    textStyle: TextStyle.PRIMARY,
    px: Space.S_16,
    py: Space.S_8
  }, weekdays.map(function (weekday, index) {
    return /*#__PURE__*/React.createElement(Flex // eslint-disable-next-line react/no-array-index-key
    , {
      key: index,
      size: Size.S_40,
      justifyContent: "center",
      alignItems: "center"
    }, format(weekday));
  }));
}
