// @revolut/ui-kit@11.21.2/esm/components/PlanWidget/PlanWidgetTitle.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import styled from 'styled-components';
import { TextStyle } from "../../constants";
import { themeTextStyle } from "../../utils";
import { Box } from "../Box";
import { useAccessibleId } from "../AccessibleId";
import { getLabelId } from "./utils";
export var PlanWidgetTitleBase = /*#__PURE__*/styled(Box).withConfig({
  displayName: "PlanWidgetTitle__PlanWidgetTitleBase",
  componentId: "zws0yr-0"
})(themeTextStyle(TextStyle.H1));
export var PlanWidgetTitle = function PlanWidgetTitle(props) {
  var id = useAccessibleId();
  return /*#__PURE__*/React.createElement(PlanWidgetTitleBase, _extends({
    id: getLabelId(id)
  }, props));
};
PlanWidgetTitle.defaultProps = {
  gridArea: '1 / 1',
  placeSelf: 'center start'
};
