// @revolut/ui-kit@11.21.2/esm/components/PlanWidget/PlanWidget.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { Color } from "../../constants";
import { AccessibleId } from "../AccessibleId";
import { PlanWidgetMediaContent } from "./PlanWidgetMediaContent";
import { PlanWidgetDescription } from "./PlanWidgetDescription";
import { PlanWidgetLayout } from "./PlanWidgetLayout";
import { PlanWidgetAction } from "./PlanWidgetAction";
import { PlanWidgetBottomAction } from "./PlanWidgetBottomAction";
import { PlanWidgetTitle } from "./PlanWidgetTitle";
import { PlanWidgetWrapper } from "./PlanWidgetWrapper";
import { PlanWidgetVariant } from "./variants";
import { getDescriptionId, getLabelId } from "./utils";

var PlanWidgetBase = function PlanWidgetBase(_ref) {
  var image = _ref.image,
      video = _ref.video,
      videoRef = _ref.videoRef,
      _ref$bg = _ref.bg,
      bg = _ref$bg === void 0 ? Color.WIDGET_BACKGROUND : _ref$bg,
      _ref$variant = _ref.variant,
      variant = _ref$variant === void 0 ? PlanWidgetVariant.LARGE : _ref$variant,
      _ref$use = _ref.use,
      use = _ref$use === void 0 ? 'div' : _ref$use,
      children = _ref.children,
      rest = _objectWithoutPropertiesLoose(_ref, ["image", "video", "videoRef", "bg", "variant", "use", "children"]);

  return /*#__PURE__*/React.createElement(AccessibleId, null, function (_ref2) {
    var id = _ref2.id;
    return /*#__PURE__*/React.createElement(PlanWidgetWrapper, _extends({
      variant: variant,
      bg: bg,
      "aria-labelledby": rest.role === 'presentation' ? undefined : getLabelId(id),
      "aria-describedby": rest.role === 'presentation' ? undefined : getDescriptionId(id),
      hasMediaContent: Boolean(image || video),
      use: use
    }, rest), /*#__PURE__*/React.createElement(PlanWidgetMediaContent, {
      image: image,
      video: video,
      videoRef: videoRef
    }), /*#__PURE__*/React.createElement(PlanWidgetLayout, null, children));
  });
};

export var PlanWidget = Object.assign(PlanWidgetBase, {
  Action: PlanWidgetAction,
  BottomAction: PlanWidgetBottomAction,
  Description: PlanWidgetDescription,
  Title: PlanWidgetTitle
});
