// @revolut/ui-kit@11.21.2/esm/components/PlanWidget/PlanWidgetMediaContent.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import React from 'react';
import styled from 'styled-components';
import { Box } from "../Box";
import { Video } from "../Video";
import { Image, ImageVariant } from "../Image";
import { layerStyle } from "./styles";
var GRADIENT_OVERLAY = 'linear-gradient(180deg, rgba(0, 0, 0, 0.1) 0%, rgba(0, 0, 0, 0.25) 100%)';
var GradientOverlay = /*#__PURE__*/styled(Box).withConfig({
  displayName: "PlanWidgetMediaContent__GradientOverlay",
  componentId: "hwnk6q-0"
})(["height:100%;background:", ";", ""], GRADIENT_OVERLAY, layerStyle);
var StyledImage = /*#__PURE__*/styled(Image).withConfig({
  displayName: "PlanWidgetMediaContent__StyledImage",
  componentId: "hwnk6q-1"
})(["background-size:cover;"]);
export var PlanWidgetMediaContent = function PlanWidgetMediaContent(_ref) {
  var image = _ref.image,
      video = _ref.video,
      videoRef = _ref.videoRef;
  return /*#__PURE__*/React.createElement(React.Fragment, null, (image || video) && /*#__PURE__*/React.createElement(React.Fragment, null, image && /*#__PURE__*/React.createElement(StyledImage, {
    variant: ImageVariant.BACKGROUND,
    height: "100%",
    role: "presentation",
    src: image
  }), video && /*#__PURE__*/React.createElement(Video, {
    ref: videoRef,
    video: video
  }), /*#__PURE__*/React.createElement(GradientOverlay, null)));
};
