// @revolut/ui-kit@11.21.2/esm/components/Story/StoryDescription.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import styled from 'styled-components';
import css from '@styled-system/css';
import { Color, Space, TextStyle } from "../../constants";
import { Text } from "../Text";
import { themeTextStyle } from "../../utils";
import { getDescriptionId } from "./utils";
import { useAccessibleId } from "../AccessibleId";
var StoryDescriptionBase = /*#__PURE__*/styled(Text).withConfig({
  displayName: "StoryDescription__StoryDescriptionBase",
  componentId: "sc-1pxok0z-0"
})(themeTextStyle(TextStyle.PRIMARY), css({
  color: Color.WHITE_80,
  marginTop: Space.S_16,
  paddingX: Space.S_24,
  flexGrow: 0,
  flexShrink: 0
}));
export var StoryDescription = function StoryDescription(props) {
  var id = useAccessibleId();
  return /*#__PURE__*/React.createElement(StoryDescriptionBase, _extends({
    id: getDescriptionId(id)
  }, props));
};
