// @revolut/ui-kit@11.21.2/esm/components/Story/variants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _extends2, _extends3, _extends4, _STORY_STYLES;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { rem } from 'polished';
import { Color } from "../../constants";
import { StoryHeaderWrapper } from "./StoryHeaderWrapper";
import { StoryLayoutWrapper } from "./StoryLayoutWrapper";
export var StoryVariant = {
  DEFAULT: 'default',
  PROGRESS: 'progress',
  INSIGHT: 'insight',
  PROMOTION: 'promotion'
};
var GRADIENT = 'linear-gradient(180deg, rgba(5, 9, 29, 0.8) 0%, rgba(4, 11, 33, 0.4) 67.82%, rgba(18, 25, 53, 0) 100%)';
var STORY_BASE_STYLE = {
  overflow: 'hidden',
  backgroundSize: 'cover',
  backgroundPosition: 'center',
  height: '100%'
};

var STORY_DEFAULT_STYLE = _extends({}, STORY_BASE_STYLE, (_extends2 = {}, _extends2[StoryHeaderWrapper] = {
  backgroundImage: GRADIENT,
  mb: rem(-200),
  pb: rem(200),
  flexGrow: 0
}, _extends2));

var STORY_PROGRESS_STYLE = _extends({}, STORY_BASE_STYLE, (_extends3 = {}, _extends3[StoryLayoutWrapper] = {
  backgroundImage: GRADIENT
}, _extends3));

var STORY_INSIGHT_STYLE = _extends({}, STORY_BASE_STYLE);

var STORY_PROMOTION_STYLE = _extends({}, STORY_BASE_STYLE, (_extends4 = {}, _extends4[StoryLayoutWrapper] = {
  backgroundColor: Color.BLACK_50
}, _extends4));

export var STORY_STYLES = (_STORY_STYLES = {}, _STORY_STYLES[StoryVariant.DEFAULT] = STORY_DEFAULT_STYLE, _STORY_STYLES[StoryVariant.PROGRESS] = STORY_PROGRESS_STYLE, _STORY_STYLES[StoryVariant.INSIGHT] = STORY_INSIGHT_STYLE, _STORY_STYLES[StoryVariant.PROMOTION] = STORY_PROMOTION_STYLE, _STORY_STYLES);
