// @revolut/ui-kit@11.21.2/esm/components/ImageWidget/ImageWidget.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled from 'styled-components';
import { css } from '@styled-system/css';
import { themeTextStyle } from "../../utils";
import { Color, Radius, Space, TextStyle } from "../../constants";
import { Action } from "../Action";
import { Image } from "../Image";
import { Text } from "../Text";
import { Widget } from "../Widget";
export var ImageWidget = Object.assign(styled(Widget).withConfig({
  displayName: "ImageWidget",
  componentId: "sc-1kb0bzk-0"
})(css({
  display: 'grid',
  rowGap: Space.S_16,
  gridTemplateColumns: '1fr auto',
  padding: Space.S_16
})), {
  Title: styled(Text).withConfig({
    displayName: "ImageWidget__Title",
    componentId: "sc-1kb0bzk-1"
  })(themeTextStyle(TextStyle.H6), css({
    alignSelf: 'start',
    color: Color.GREY_TONE_50,
    gridRow: '1',
    marginRight: Space.S_16
  })),
  Action: styled(Action).withConfig({
    displayName: "ImageWidget__Action",
    componentId: "sc-1kb0bzk-2"
  })(css({
    alignSelf: 'start',
    gridRow: '1',
    justifySelf: 'end'
  })),
  Description: styled(Text).withConfig({
    displayName: "ImageWidget__Description",
    componentId: "sc-1kb0bzk-3"
  })(themeTextStyle(TextStyle.CAPTION), css({
    color: Color.FOREGROUND,
    gridColumn: '1 / span 2'
  })),
  Image: styled(Image).withConfig({
    displayName: "ImageWidget__Image",
    componentId: "sc-1kb0bzk-4"
  })(css({
    borderRadius: Radius.IMAGE,
    gridColumn: '1 / span 2',
    overflow: 'hidden'
  }))
});
ImageWidget.Image.defaultProps = {
  role: 'presentation'
};
