// @revolut/ui-kit@11.21.2/esm/components/Bar/variants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _BAR_STYLES;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { Space } from "../../constants";
var BASE_STYLE = {
  display: 'grid',
  alignItems: 'center',
  gridAutoFlow: 'column',
  gridAutoColumns: 'max-content'
};

var CODE_OR_ACTIONS_STYLE = _extends({}, BASE_STYLE, {
  gridColumnGap: Space.S_8
});

var ATTACHMENTS_STYLE = _extends({}, BASE_STYLE, {
  gridColumnGap: Space.S_16
});

var SUGGESTIONS_STYLE = _extends({}, BASE_STYLE, {
  paddingY: Space.S_8,
  gridColumnGap: Space.S_16
});

var ICONS_STYLE = _extends({}, BASE_STYLE, {
  gridColumnGap: Space.S_24
});

var TABLE_TAGS_STYLE = _extends({}, BASE_STYLE, {
  gridColumnGap: Space.S_4
});

export var BarVariant = {
  ACTIONS: 'actions',
  CODE: 'code',
  ATTACHMENTS: 'attachments',
  ICONS: 'icons',
  SUGGESTIONS: 'suggestions',
  TABLE_TAGS: 'table-tags'
};
export var BAR_STYLES = (_BAR_STYLES = {}, _BAR_STYLES[BarVariant.ACTIONS] = CODE_OR_ACTIONS_STYLE, _BAR_STYLES[BarVariant.CODE] = CODE_OR_ACTIONS_STYLE, _BAR_STYLES[BarVariant.ATTACHMENTS] = ATTACHMENTS_STYLE, _BAR_STYLES[BarVariant.SUGGESTIONS] = SUGGESTIONS_STYLE, _BAR_STYLES[BarVariant.ICONS] = ICONS_STYLE, _BAR_STYLES[BarVariant.TABLE_TAGS] = TABLE_TAGS_STYLE, _BAR_STYLES);
