// @revolut/ui-kit@11.21.2/esm/components/ContactGrid/ContactGrid.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled from 'styled-components';
import { css } from '@styled-system/css';
import { Space } from "../../constants";
import { Box } from "../Box";
import { WIDTH } from "../Contact/constants";
export var ContactGrid = /*#__PURE__*/styled(Box).withConfig({
  displayName: "ContactGrid",
  componentId: "sc-1c2sklx-0"
})(css({
  display: 'grid',
  columnGap: Space.S_8,
  gridTemplateColumns: "repeat(auto-fill, minmax(" + WIDTH + ", auto))",
  gridTemplateRows: 'repeat(2, auto)',
  justifyItems: 'center',
  overflow: 'hidden',
  gridAutoRows: 0,
  '& > *': {
    marginBottom: Space.S_16
  }
}));
