// @revolut/ui-kit@11.21.2/esm/components/BottomNav/BottomNav.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import React from 'react';
import { ElevatedThemeProvider } from "../ThemeProvider";
import { Flex } from "../Flex";
import { TransitionSlide } from "../Transitions";
import { useFloatingMenuContext } from "../Layout/FloatingMenuContext";
import { BottomNavVariant } from "./variants";
import { BottomNavItem } from "./BottomNavItem";
export var BottomNav = function BottomNav(props) {
  var floatingMenuVisible = useFloatingMenuContext();
  return /*#__PURE__*/React.createElement(ElevatedThemeProvider, null, props.variant === BottomNavVariant.FLOATING && floatingMenuVisible !== null ? /*#__PURE__*/React.createElement(TransitionSlide, {
    unmountOnExit: false,
    "in": floatingMenuVisible,
    appear: false,
    offsetY: "-100%"
  }, /*#__PURE__*/React.createElement(Flex, props)) : /*#__PURE__*/React.createElement(Flex, props));
};
BottomNav.defaultProps = {
  variant: BottomNavVariant.FIXED,
  variantKey: 'bottomNavStyles'
};
BottomNav.Item = BottomNavItem;
