// @revolut/ui-kit@11.21.2/esm/components/BottomNav/BottomNavItem.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { forwardRef } from 'react';
import styled from 'styled-components';
import { Color, Space } from "../../constants";
import { themeColor, themeSpace } from "../../utils";
import { Badge, BadgeVariant } from "../Badge";
import { ButtonBase } from "../ButtonBase";
export var BottomNavItemBase = /*#__PURE__*/styled(ButtonBase).withConfig({
  displayName: "BottomNavItem__BottomNavItemBase",
  componentId: "i9sd5i-0"
})(["&[aria-pressed='true'],&[aria-selected='true'],&[aria-current='page']{color:", ";}&:disabled,&[aria-disabled='true']{color:", ";}"], themeColor(Color.BLUE), themeColor(Color.GREY_TONE_20));
var BottomNavDot = /*#__PURE__*/styled(Badge).withConfig({
  displayName: "BottomNavItem__BottomNavDot",
  componentId: "i9sd5i-1"
})(["position:absolute;bottom:", ";"], themeSpace(Space.S_2));
BottomNavDot.defaultProps = {
  variant: BadgeVariant.DOT,
  use: 'span'
};
export var BottomNavItem = /*#__PURE__*/forwardRef(function (_ref, ref) {
  var Icon = _ref.useIcon,
      hasDot = _ref.hasDot,
      children = _ref.children,
      rest = _objectWithoutPropertiesLoose(_ref, ["useIcon", "hasDot", "children"]);

  return /*#__PURE__*/React.createElement(BottomNavItemBase, _extends({
    ref: ref
  }, rest), Icon ? /*#__PURE__*/React.createElement(Icon, {
    size: 24
  }) : children, hasDot && /*#__PURE__*/React.createElement(BottomNavDot, null));
});
