// @revolut/ui-kit@11.21.2/esm/components/BottomNav/variants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _extends2, _extends3, _BOTTOM_NAV_STYLES;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { rem } from 'polished';
import { rgbaSafe, themeColor } from "../../utils";
import { Color, Radius, Space } from "../../constants";
import { BottomNavItemBase } from "./BottomNavItem";
export var BottomNavVariant = {
  /** @deprecated use `floating` variant instead */
  FIXED: 'fixed',
  FLOATING: 'floating'
};
var BOTTOM_NAV_ITEM_BASE_STYLE = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center'
};
var BOTTOM_NAV_BASE_STYLE = {
  position: 'relative',
  zIndex: '0',
  color: Color.GREY_TONE_50,
  backdropFilter: 'blur(16px)'
};

var BOTTOM_NAV_FIXED_STYLE = _extends({}, BOTTOM_NAV_BASE_STYLE, (_extends2 = {
  height: rem(49),
  backgroundColor: Color.BOTTOM_NAV
}, _extends2[BottomNavItemBase] = _extends({}, BOTTOM_NAV_ITEM_BASE_STYLE, {
  borderRadius: Radius.CELL,
  minWidth: 0,
  flex: '1 1 auto'
}), _extends2));

var BOTTOM_NAV_FLOATING_STYLE = _extends({}, BOTTOM_NAV_BASE_STYLE, (_extends3 = {
  width: 'fit-content',
  paddingX: Space.S_16,
  paddingY: Space.S_8,
  marginX: 'auto',
  marginY: Space.S_16,
  backgroundColor: function backgroundColor(theme) {
    return rgbaSafe(themeColor(Color.WIDGET_BACKGROUND)({
      theme: theme
    }), 0.9);
  },
  gap: 16,
  borderRadius: Radius.BOTTOM_NAV,
  boxShadow: '0px 4px 24px rgba(0, 0, 0, 0.1)'
}, _extends3[BottomNavItemBase] = _extends({}, BOTTOM_NAV_ITEM_BASE_STYLE, {
  borderRadius: Radius.ROUND,
  width: 40,
  height: 40,
  ':active': {
    backgroundColor: Color.GREY_TONE_5
  }
}), _extends3));

export var BOTTOM_NAV_STYLES = (_BOTTOM_NAV_STYLES = {}, _BOTTOM_NAV_STYLES[BottomNavVariant.FIXED] = BOTTOM_NAV_FIXED_STYLE, _BOTTOM_NAV_STYLES[BottomNavVariant.FLOATING] = BOTTOM_NAV_FLOATING_STYLE, _BOTTOM_NAV_STYLES);
