// @revolut/ui-kit@11.21.2/esm/components/Group/Group.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _groupBackgrounds;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { forwardRef } from 'react';
import styled from 'styled-components';
import { Color, Radius } from "../../constants";
import { themeColor, themeRadius } from "../../utils";
import { Box } from "../Box";
import { GroupVariant } from "./variants";
var groupBackgrounds = (_groupBackgrounds = {}, _groupBackgrounds[GroupVariant.DEFAULT] = Color.WIDGET_BACKGROUND, _groupBackgrounds[GroupVariant.GREY] = Color.GREY_TONE_8, _groupBackgrounds);
var GroupBase = /*#__PURE__*/styled(Box).withConfig({
  displayName: "Group__GroupBase",
  componentId: "k2h0d0-0"
})(["&& > *{border-radius:0;background-color:", ";transition:margin-top 300ms,margin-bottom 300ms;}>:first-child{border-top-left-radius:", ";border-top-right-radius:", ";}>:last-child{border-bottom-left-radius:", ";border-bottom-right-radius:", ";}[aria-expanded='true']{&:not(:first-child){margin-top:8px;}&:not(:last-child){margin-bottom:8px;}border-radius:", ";& + *{border-top-left-radius:", ";border-top-right-radius:", ";}}[data-before-expanded='true']{border-bottom-left-radius:", ";border-bottom-right-radius:", ";}"], function (props) {
  return themeColor(props.elementBg);
}, themeRadius(Radius.CELL), themeRadius(Radius.CELL), themeRadius(Radius.CELL), themeRadius(Radius.CELL), themeRadius(Radius.CELL), themeRadius(Radius.CELL), themeRadius(Radius.CELL), themeRadius(Radius.CELL), themeRadius(Radius.CELL));
/**
 * Docs: {@link https://ui-kit.revolut.codes/components/group|Group}
 */

export var Group = /*#__PURE__*/forwardRef(function (_ref, ref) {
  var _ref$variant = _ref.variant,
      variant = _ref$variant === void 0 ? GroupVariant.DEFAULT : _ref$variant,
      rest = _objectWithoutPropertiesLoose(_ref, ["variant"]);

  return /*#__PURE__*/React.createElement(GroupBase, _extends({
    ref: ref,
    elementBg: groupBackgrounds[variant],
    color: Color.FOREGROUND
  }, rest));
});
