// @revolut/ui-kit@11.21.2/esm/components/ErrorWidget/ErrorWidget.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { Space } from "../../constants";
import { useMessage } from "../../hooks";
import { Placeholder } from "../Placeholder";
import { Widget } from "../Widget";
export var ErrorWidgetVariant = {
  DEFAULT: 'default',

  /** @deprecated */
  COMPACT: 'compact'
};
/** @deprecated */

export var ErrorWidget = function ErrorWidget(_ref) {
  var children = _ref.children,
      rest = _objectWithoutPropertiesLoose(_ref, ["children"]);

  return /*#__PURE__*/React.createElement(Widget, _extends({
    pt: Space.S_16,
    px: Space.S_16,
    pb: Space.S_24
  }, rest), /*#__PURE__*/React.createElement(Placeholder, {
    p: Space.NONE
  }, children));
};

var ErrorWidgetTitle = function ErrorWidgetTitle(_ref2) {
  var children = _ref2.children,
      rest = _objectWithoutPropertiesLoose(_ref2, ["children"]);

  var m = useMessage();
  return /*#__PURE__*/React.createElement(Placeholder.Title, rest, children != null ? children : m('label.failed-state'));
};

var ErrorWidgetAction = function ErrorWidgetAction(_ref3) {
  var children = _ref3.children,
      rest = _objectWithoutPropertiesLoose(_ref3, ["children"]);

  var m = useMessage();
  return /*#__PURE__*/React.createElement(Placeholder.Action, rest, children != null ? children : m('action.retry'));
};

ErrorWidget.Title = ErrorWidgetTitle;
ErrorWidget.Description = Placeholder.Description;
ErrorWidget.Action = ErrorWidgetAction;
ErrorWidget.Image = Placeholder.Image;
