// @revolut/ui-kit@11.21.2/esm/components/ProductWidget/ProductWidgetSkeleton.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { rem } from 'polished';
import { Space } from "../../constants";
import { Skeleton } from "../Skeleton";
import { Subheader } from "../Subheader";
import { ItemSkeleton } from "../Item";
import { BarSkeleton } from "../Bar";
import { AvatarSkeleton } from "../Avatar";
import { MoreBarSkeleton } from "../MoreBar";
import { ProductWidget } from "./ProductWidget";

/** @deprecated  use `ProductWidgetSkeleton.Bar` */
var ProductWidgetActionBarSkeleton = function ProductWidgetActionBarSkeleton(_ref) {
  var children = _ref.children,
      rest = _objectWithoutPropertiesLoose(_ref, ["children"]);

  return /*#__PURE__*/React.createElement(ProductWidget.ActionBar, rest, /*#__PURE__*/React.createElement(BarSkeleton, null));
};

var ProductWidgetBarSkeleton = function ProductWidgetBarSkeleton(_ref2) {
  var children = _ref2.children,
      rest = _objectWithoutPropertiesLoose(_ref2, ["children"]);

  return /*#__PURE__*/React.createElement(ProductWidget.Bar, rest, /*#__PURE__*/React.createElement(MoreBarSkeleton, null));
};

/** @deprecated  use `ProductWidgetSkeleton.Avatar` */
export var ProductWidgetAvatarSkeleton = function ProductWidgetAvatarSkeleton(_ref3) {
  var children = _ref3.children,
      rest = _objectWithoutPropertiesLoose(_ref3, ["children"]);

  return /*#__PURE__*/React.createElement(ProductWidget.Avatar, rest, /*#__PURE__*/React.createElement(AvatarSkeleton, null));
};

/** @deprecated  use `ProductWidgetSkeleton.Title` */
export var ProductWidgetTitleSkeleton = function ProductWidgetTitleSkeleton(_ref4) {
  var children = _ref4.children,
      rest = _objectWithoutPropertiesLoose(_ref4, ["children"]);

  return /*#__PURE__*/React.createElement(ProductWidget.Title, rest, /*#__PURE__*/React.createElement(Skeleton, {
    mt: Space.S_8,
    height: rem(24),
    width: "50%"
  }));
};

/** @deprecated  use `ProductWidgetSkeleton.Description` */
export var ProductWidgetDescriptionSkeleton = function ProductWidgetDescriptionSkeleton(props) {
  return /*#__PURE__*/React.createElement(ProductWidget.Description, props, "\xA0");
};

/** @deprecated  use `ProductWidgetSkeleton.Header` */
export var ProductWidgetHeaderSkeleton = function ProductWidgetHeaderSkeleton(_ref5) {
  var children = _ref5.children,
      rest = _objectWithoutPropertiesLoose(_ref5, ["children"]);

  return /*#__PURE__*/React.createElement(ProductWidget.Header, rest, children != null ? children : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(ProductWidgetTitleSkeleton, null), /*#__PURE__*/React.createElement(ProductWidgetAvatarSkeleton, null), /*#__PURE__*/React.createElement(ProductWidgetDescriptionSkeleton, null), /*#__PURE__*/React.createElement(ProductWidgetBarSkeleton, null)));
};

var ProductWidgetSkeletonBase = function ProductWidgetSkeletonBase(_ref6) {
  var children = _ref6.children,
      rest = _objectWithoutPropertiesLoose(_ref6, ["children"]);

  return /*#__PURE__*/React.createElement(ProductWidget, rest, children != null ? children : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(ProductWidgetHeaderSkeleton, null), /*#__PURE__*/React.createElement(Subheader, null), /*#__PURE__*/React.createElement(ItemSkeleton, null)));
};

export var ProductWidgetSkeleton = Object.assign(ProductWidgetSkeletonBase, {
  /** @deprecated  use `ProductWidgetSkeleton.Actions` */
  ActionBar: ProductWidgetActionBarSkeleton,
  Bar: ProductWidgetBarSkeleton,
  Avatar: ProductWidgetAvatarSkeleton,
  Header: ProductWidgetHeaderSkeleton,
  Title: ProductWidgetTitleSkeleton,
  Description: ProductWidgetDescriptionSkeleton
});
