// @revolut/ui-kit@11.21.2/esm/components/CounterWidget/CounterWidget.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { Color, Radius } from "../../constants";
import { HStack } from "../HStack";
import { CounterWidgetColumn } from "./CounterWidgetColumn";
import { CounterWidgetDescription } from "./CounterWidgetDescription";
import { CounterWidgetSubtitle } from "./CounterWidgetSubtitle";
import { CounterWidgetTitle } from "./CounterWidgetTitle";

var CounterWidgetBase = function CounterWidgetBase(_ref) {
  var _ref$use = _ref.use,
      use = _ref$use === void 0 ? 'div' : _ref$use,
      children = _ref.children,
      rest = _objectWithoutPropertiesLoose(_ref, ["use", "children"]);

  return /*#__PURE__*/React.createElement(HStack, _extends({
    bg: Color.WIDGET_BACKGROUND,
    radius: Radius.WIDGET
  }, rest), children);
};

export var CounterWidget = Object.assign(CounterWidgetBase, {
  Column: CounterWidgetColumn,
  Description: CounterWidgetDescription,
  Subtitle: CounterWidgetSubtitle,
  Title: CounterWidgetTitle
});
