// @revolut/ui-kit@11.21.2/esm/components/CounterWidget/CounterWidgetSkeleton.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { rem } from 'polished';
import { Size } from "../../constants";
import { Skeleton } from "../Skeleton";
import { Flex } from "../Flex";
import { CounterWidget } from "./CounterWidget";

var CounterWidgetTitleSkeleton = function CounterWidgetTitleSkeleton(props) {
  return /*#__PURE__*/React.createElement(CounterWidget.Title, {
    width: "40%"
  }, /*#__PURE__*/React.createElement(Flex, {
    height: rem(22),
    width: "100%",
    alignItems: "center"
  }, /*#__PURE__*/React.createElement(Skeleton, _extends({
    height: Size.S_16,
    width: "100%"
  }, props))));
};

var CounterWidgetDescriptionSkeleton = function CounterWidgetDescriptionSkeleton(props) {
  return /*#__PURE__*/React.createElement(CounterWidget.Description, {
    width: "90%"
  }, /*#__PURE__*/React.createElement(Flex, {
    height: rem(36),
    width: "100%",
    alignItems: "center"
  }, /*#__PURE__*/React.createElement(Skeleton, _extends({
    height: Size.S_16,
    width: "100%"
  }, props))));
};

var CounterWidgetSubtitleSkeleton = function CounterWidgetSubtitleSkeleton(props) {
  return /*#__PURE__*/React.createElement(CounterWidget.Subtitle, {
    width: "40%"
  }, /*#__PURE__*/React.createElement(Flex, {
    height: Size.S_20,
    width: "100%",
    alignItems: "center"
  }, /*#__PURE__*/React.createElement(Skeleton, _extends({
    height: Size.S_16,
    width: "100%"
  }, props))));
};

export var CounterWidgetSkeletonBase = function CounterWidgetSkeletonBase(_ref) {
  var children = _ref.children,
      rest = _objectWithoutPropertiesLoose(_ref, ["children"]);

  return /*#__PURE__*/React.createElement(CounterWidget, rest, children != null ? children : /*#__PURE__*/React.createElement(CounterWidget.Column, {
    role: "presentation",
    width: rem(168)
  }, /*#__PURE__*/React.createElement(CounterWidgetTitleSkeleton, null), /*#__PURE__*/React.createElement(CounterWidgetDescriptionSkeleton, null), /*#__PURE__*/React.createElement(CounterWidgetSubtitleSkeleton, null)));
};
export var CounterWidgetSkeleton = Object.assign(CounterWidgetSkeletonBase, {
  Title: CounterWidgetTitleSkeleton,
  Description: CounterWidgetDescriptionSkeleton,
  Subtitle: CounterWidgetSubtitleSkeleton
});
