// @revolut/ui-kit@11.21.2/esm/legacy/components/CircleButton/sizes.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _CIRCLE_BUTTON_SIZE_S;

import { ButtonSize, Radius, Size } from "../../../constants";
export var CircleButtonSizeStyle = {
  SM: ButtonSize.SM,
  MD: ButtonSize.MD
};
var BUTTON_SM = {
  size: Size.BUTTON_SM,
  borderRadius: Radius.BUTTON_SM
};
var BUTTON_MD = {
  size: [Size.BUTTON_MD, null, Size.BUTTON_MD_PLUS],
  borderRadius: [Radius.BUTTON_MD, null, Radius.BUTTON_LG]
};
export var CIRCLE_BUTTON_SIZE_STYLES = (_CIRCLE_BUTTON_SIZE_S = {}, _CIRCLE_BUTTON_SIZE_S[CircleButtonSizeStyle.SM] = BUTTON_SM, _CIRCLE_BUTTON_SIZE_S[CircleButtonSizeStyle.MD] = BUTTON_MD, _CIRCLE_BUTTON_SIZE_S);
