// @revolut/ui-kit@11.21.2/esm/future/styles/grid.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import { system } from '@styled-system/core';
import { isNumber } from "../../utils";

function transformGridTemplate(value) {
  return isNumber(value) ? "repeat(" + value + ", 1fr)" : value;
}

export var grid = system({
  template: {
    property: 'gridTemplate'
  },
  columns: {
    property: 'gridTemplateColumns',
    transform: transformGridTemplate
  },
  rows: {
    property: 'gridTemplateRows',
    transform: transformGridTemplate
  },
  autoFlow: {
    property: 'gridAutoFlow'
  },
  autoRows: {
    property: 'gridAutoRows'
  },
  autoColumns: {
    property: 'gridAutoColumns'
  },
  placeItems: true,
  placeContent: true
});
