// @revolut/ui-kit@11.21.2/esm/components/Circle/Circle.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import styled, { css } from 'styled-components';
import { system } from 'styled-system';
import { Flex } from "../Flex";
import { CircleVariant } from "./variants";
export var Circle = /*#__PURE__*/styled(Flex).withConfig({
  displayName: "Circle",
  componentId: "sc-1bqbzna-0"
})(["align-items:center;justify-content:center;border-radius:9999px;", ""], function (props) {
  return props.image && css(["background-size:cover;background-position:center;background-repeat:no-repeat;", ""], system({
    image: {
      property: 'backgroundImage',
      transform: function transform(value) {
        return "url(" + value + ")";
      }
    }
  }));
});
Circle.displayName = 'Circle';
Circle.defaultProps = _extends({}, Flex.defaultProps, {
  variantKey: 'circleStyles',
  variant: CircleVariant.DEFAULT
});
