// @revolut/ui-kit@11.21.2/esm/components/Circle/variants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _CIRCLE_VARIANTS;

export var CircleVariant = {
  DEFAULT: 'default',
  FILLED: 'filled',
  OUTLINED: 'outlined'
};
var FILLED_STYLE = {
  color: 'white',
  bg: 'primary'
};
var OUTLINED_STYLE = {
  color: 'grey-tone-20',
  border: '2px solid'
};
export var CIRCLE_VARIANTS = (_CIRCLE_VARIANTS = {}, _CIRCLE_VARIANTS[CircleVariant.DEFAULT] = null, _CIRCLE_VARIANTS[CircleVariant.FILLED] = FILLED_STYLE, _CIRCLE_VARIANTS[CircleVariant.OUTLINED] = OUTLINED_STYLE, _CIRCLE_VARIANTS);
