// @revolut/ui-kit@11.21.2/esm/components/Dropdown/DropdownBox.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled, { css } from 'styled-components';
import { rem } from 'polished';
import { Color, ElevationLevel, Radius, Space, TextStyle } from "../../constants";
import { isUnifiedTheme, themeColor } from "../../utils";
import { scrollbar } from "../../styles";
import { Box } from "../Box";
export var DropdownBox = /*#__PURE__*/styled(Box).withConfig({
  displayName: "DropdownBox",
  componentId: "sc-1fvebdr-0"
})(["overflow-x:hidden;overflow-y:auto;", ""], function (props) {
  if (isUnifiedTheme(props.theme)) {
    return css(["background-color:", ";", ""], themeColor(Color.WIDGET_BACKGROUND), scrollbar(Space.S_4));
  }

  return null;
});
DropdownBox.defaultProps = {
  maxHeight: rem(344),
  radius: Radius.DROPDOWN,
  bg: Color.BACKGROUND,
  color: Color.FOREGROUND,
  textStyle: TextStyle.SECONDARY,
  elevation: ElevationLevel.HIGHER
};
