// @revolut/ui-kit@11.21.2/esm/components/Dropdown/DropdownItem.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { forwardRef } from 'react';
import styled, { css } from 'styled-components';
import { rem } from 'polished';
import { isUnifiedTheme, themeColor, themeSpace } from "../../utils";
import { Color, Space } from "../../constants";
import { useTheme } from "../../hooks";
import { Box } from "../Box";
import { Cell } from "../Cell";
export var DropdownItemCell = /*#__PURE__*/styled(Cell).withConfig({
  displayName: "DropdownItem__DropdownItemCell",
  componentId: "efr6jd-0"
})(["padding-top:", ";padding-bottom:", ";"], themeSpace(Space.S_12), themeSpace(Space.S_12));
/** @deprecated */

export var DropdownItemBase = /*#__PURE__*/styled(Box).withConfig({
  displayName: "DropdownItem__DropdownItemBase",
  componentId: "efr6jd-1"
})(["appearance:none;border:none;background:none;text-align:inherit;font:inherit;display:block;min-width:100%;cursor:", ";"], function (props) {
  return props.disabled && 'default';
});
DropdownItemBase.defaultProps = {
  textStyle: 'default',
  minHeight: rem(56),
  px: Space.S_24,
  py: Space.S_16
};
/** @deprecated */

/** @deprecated */
export var DropdownLegacyItem = /*#__PURE__*/styled(DropdownItemBase).withConfig({
  displayName: "DropdownItem__DropdownLegacyItem",
  componentId: "efr6jd-2"
})(["cursor:pointer;background-color:", ";&:hover:not([disabled]){background-color:", ";}", ""], function (props) {
  return (props.isHighlighted || props.isSelected) && themeColor(Color.GREY_TONE_5);
}, themeColor(Color.GREY_TONE_5), function (props) {
  return props.disabled && css(["color:", ";pointer-events:none;"], themeColor(Color.GREY_TONE_20));
});
export var DropdownItem = /*#__PURE__*/forwardRef(function (_ref, ref) {
  var isHighlighted = _ref.isHighlighted,
      isSelected = _ref.isSelected,
      children = _ref.children,
      unsafe__enableSelectedSelector = _ref.unsafe__enableSelectedSelector,
      Icon = _ref.useIcon,
      color = _ref.color,
      rest = _objectWithoutPropertiesLoose(_ref, ["isHighlighted", "isSelected", "children", "unsafe__enableSelectedSelector", "useIcon", "color"]);

  var theme = useTheme();

  if (isUnifiedTheme(theme)) {
    return /*#__PURE__*/React.createElement(DropdownItemCell, _extends({
      ref: ref,
      unsafe__enableSelectedSelector: unsafe__enableSelectedSelector
    }, rest), Icon && /*#__PURE__*/React.createElement(Box, {
      mr: children ? Space.S_16 : undefined,
      color: color
    }, /*#__PURE__*/React.createElement(Icon, {
      size: 24
    })), children && /*#__PURE__*/React.createElement(Box, {
      width: "100%",
      color: color
    }, children));
  }

  return /*#__PURE__*/React.createElement(DropdownLegacyItem, _extends({
    ref: ref,
    isHighlighted: isHighlighted,
    isSelected: isSelected
  }, rest), children);
});
