// @revolut/ui-kit@11.21.2/esm/components/RewardSticker/RewardSticker.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import React from 'react';
import css from '@styled-system/css';
import styled from 'styled-components';
import { rem } from 'polished';
import { Box } from "../Box";
import { Flex } from "../Flex";
import { Color, Radius, Space, TextStyle } from "../../constants";
import { themeTextStyle } from "../../utils";
var Wrapper = /*#__PURE__*/styled(Box).withConfig({
  displayName: "RewardSticker__Wrapper",
  componentId: "zk2555-0"
})(function (_ref) {
  var image = _ref.image;
  return {
    background: "linear-gradient(180deg, rgba(0, 0, 0, 0) 0%, rgba(0, 0, 0, 0.6) 100%), url(" + image + ")"
  };
}, css({
  width: rem(289),
  height: rem(362),
  display: 'grid',
  gridTemplateRows: 'minmax(min-content, 1fr)',
  rowGap: Space.S_8,
  backgroundSize: 'cover',
  backgroundPosition: 'center',
  borderRadius: Radius.REWARD_STICKER,
  p: Space.S_24
}));
export var RewardSticker = function RewardSticker(props) {
  return /*#__PURE__*/React.createElement(Wrapper, props);
};
var StickerAvatar = /*#__PURE__*/styled(Box).withConfig({
  displayName: "RewardSticker__StickerAvatar",
  componentId: "zk2555-1"
})(css({
  justifySelf: 'start',
  order: 1
}));
var StickerTitle = /*#__PURE__*/styled(Box).withConfig({
  displayName: "RewardSticker__StickerTitle",
  componentId: "zk2555-2"
})(themeTextStyle(TextStyle.H4), css({
  color: Color.WHITE,
  order: 2
}));
var StickerDescription = /*#__PURE__*/styled(Flex).withConfig({
  displayName: "RewardSticker__StickerDescription",
  componentId: "zk2555-3"
})(themeTextStyle(TextStyle.PRIMARY), css({
  color: Color.WHITE_80,
  order: 3
}));
RewardSticker.Avatar = StickerAvatar;
RewardSticker.Title = StickerTitle;
RewardSticker.Description = StickerDescription;
