// @revolut/ui-kit@11.21.2/esm/components/Countdown/Timer.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _timeFormats, _TextSizes, _TextColors;

import React from 'react';
import styled from 'styled-components';
import { Color, DEFAULT_LOCALE, TextStyle } from "../../constants";
import { Text } from "../Text";
import { CountdownSize } from "./sizes";
var timeFormatMD = new Intl.DateTimeFormat(DEFAULT_LOCALE, {
  minute: 'numeric',
  second: 'numeric'
});
var timeFormatSM = new Intl.DateTimeFormat(DEFAULT_LOCALE, {
  second: 'numeric'
});
var timeFormats = (_timeFormats = {}, _timeFormats[CountdownSize.MD] = timeFormatMD, _timeFormats[CountdownSize.SM] = timeFormatSM, _timeFormats);

var formatTime = function formatTime(size, input) {
  return timeFormats[size].format(Math.ceil(input / 1000) * 1000).replace(/^(0)(\d:)/, function (_match, _p1, p2) {
    return p2;
  });
};

var TextSizes = (_TextSizes = {}, _TextSizes[CountdownSize.MD] = TextStyle.H5, _TextSizes[CountdownSize.SM] = TextStyle.H6, _TextSizes);
var TextColors = (_TextColors = {}, _TextColors[CountdownSize.MD] = undefined, _TextColors[CountdownSize.SM] = Color.FOREGROUND, _TextColors);
var TimerText = /*#__PURE__*/styled(Text).withConfig({
  displayName: "Timer__TimerText",
  componentId: "y8tpoq-0"
})(["position:absolute;left:50%;top:50%;transform:translate(-50%,-50%);"]);
export var Timer = function Timer(_ref) {
  var value = _ref.value,
      size = _ref.size;
  return /*#__PURE__*/React.createElement(TimerText, {
    variant: TextSizes[size],
    color: TextColors[size]
  }, formatTime(size, value));
};
