// @revolut/ui-kit@11.21.2/esm/components/Calendar/hooks/useCalendarRangeState.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import { useCallback } from 'react';
import { DateUtils } from 'react-day-picker';
import { useControlled } from "../../../hooks";

function rangeUpdate(selectedDate, state) {
  if (!selectedDate) {
    return null;
  }

  if (!state) {
    return {
      from: selectedDate,
      to: selectedDate
    };
  }

  var result = DateUtils.addDayToRange(selectedDate, state);

  if (result.from === null) {
    result.from = selectedDate;
  }

  if (result.to === null) {
    result.to = selectedDate;
  }

  return result;
}

export function useCalendarRangeState(_ref) {
  var controlledValue = _ref.controlledValue,
      defaultValue = _ref.defaultValue,
      onChange = _ref.onChange;

  var _useControlled = useControlled({
    value: controlledValue,
    defaultValue: defaultValue,
    onChange: onChange
  }),
      value = _useControlled[0],
      setValue = _useControlled[1];

  var update = useCallback(function (selectedDate) {
    return setValue(rangeUpdate(selectedDate, value));
  }, [value, setValue]);
  return [value, update];
}
