// @revolut/ui-kit@11.21.2/esm/components/RateWidget/RateWidget.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import styled from 'styled-components';
import { css } from '@styled-system/css';
import { Color, Space, TextStyle } from "../../constants";
import { themeTextStyle } from "../../utils";
import { Box } from "../Box";
import { Flex } from "../Flex";
import { Widget } from "../Widget";
var RateWidgetBase = /*#__PURE__*/styled(Widget).withConfig({
  displayName: "RateWidget__RateWidgetBase",
  componentId: "sc-1ilpee3-0"
})(css({
  padding: Space.S_16
}));
export var RateWidgetContent = /*#__PURE__*/styled(Box).withConfig({
  displayName: "RateWidget__RateWidgetContent",
  componentId: "sc-1ilpee3-1"
})(themeTextStyle(TextStyle.H6), css({
  color: Color.GREY_TONE_50
}));
export var RateWidgetActionsContainer = /*#__PURE__*/styled(Box).withConfig({
  displayName: "RateWidget__RateWidgetActionsContainer",
  componentId: "sc-1ilpee3-2"
})(css({
  columnGap: Space.S_8,
  display: 'grid',
  gridAutoFlow: 'column',
  marginTop: Space.S_16,
  justifyItems: 'center'
}));
export var RateWidgetActions = function RateWidgetActions(_ref) {
  var children = _ref.children,
      rest = _objectWithoutPropertiesLoose(_ref, ["children"]);

  return /*#__PURE__*/React.createElement(Flex, _extends({
    justifyContent: "center"
  }, rest), /*#__PURE__*/React.createElement(RateWidgetActionsContainer, null, children));
};
export var RateWidgetStars = function RateWidgetStars(props) {
  return /*#__PURE__*/React.createElement(Flex, _extends({
    justifyContent: "center",
    alignItems: "center",
    mt: Space.S_16
  }, props));
};
export var RateWidget = Object.assign(RateWidgetBase, {
  Content: RateWidgetContent,
  Actions: RateWidgetActions,
  Stars: RateWidgetStars
});
