// @revolut/ui-kit@11.21.2/esm/components/Popup/variants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _POPUP_DIALOG_BASE_ST, _extends2, _extends3, _extends4, _ref, _POPUP_STYLES;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { Color, Radius, Space } from "../../constants";
import { themeColor } from "../../utils";
import { Modal } from "../Modal";
import { BACKGROUND_COLOR_VAR } from "../Header/constants";
export var PopupVariant = {
  MODAL_VIEW: 'modal-view',
  BOTTOM_SHEET: 'bottom-sheet',

  /** @deprecated use `bottom-sheet` variant */
  DIALOG: 'dialog',

  /** @deprecated use `Prompt` */
  COLORFUL: 'colorful',

  /** @private  */
  RESULT: 'result'
};
var POPUP_DIALOG_BASE_STYLE = (_POPUP_DIALOG_BASE_ST = {
  marginX: 'auto',
  marginTop: 'auto',
  marginBottom: [Space.NONE, null, 'auto'],
  width: '100%',
  maxWidth: ['100%', null, '25rem'],
  padding: [Space.S_16, null, Space.S_24],
  maxHeight: ['unset', null, '42rem'],
  borderRadius: [Radius.POPUP_TOP, null, Radius.POPUP],
  backgroundColor: Color.GROUPED_BACKGROUND,
  color: Color.FOREGROUND
}, _POPUP_DIALOG_BASE_ST[BACKGROUND_COLOR_VAR] = function (theme) {
  return themeColor(Color.GROUPED_BACKGROUND)({
    theme: theme
  });
}, _POPUP_DIALOG_BASE_ST);

var POPUP_DIALOG_MODAL_VIEW_STYLE = _extends({}, POPUP_DIALOG_BASE_STYLE, {
  minHeight: ['unset', null, '37.5rem'],
  borderRadius: [Radius.POPUP_FULL_SCREEN, null, Radius.POPUP]
});

var POPUP_DIALOG_COLORFUL_STYLE = _extends({}, POPUP_DIALOG_MODAL_VIEW_STYLE, (_extends2 = {
  backgroundColor: Color.BLUE,
  color: Color.WHITE
}, _extends2[BACKGROUND_COLOR_VAR] = function (theme) {
  return themeColor(Color.BLUE)({
    theme: theme
  });
}, _extends2));

var POPUP_DIALOG_BOTTOM_SHEET_STYLE = _extends({}, POPUP_DIALOG_BASE_STYLE);

var POPUP_DIALOG_DIALOG_STYLE = _extends({}, POPUP_DIALOG_BASE_STYLE, (_extends3 = {
  backgroundColor: Color.WIDGET_BACKGROUND
}, _extends3[BACKGROUND_COLOR_VAR] = function (theme) {
  return themeColor(Color.WIDGET_BACKGROUND)({
    theme: theme
  });
}, _extends3));

var POPUP_DIALOG_RESULT_STYLE = _extends({}, POPUP_DIALOG_BASE_STYLE, {
  borderRadius: Radius.POPUP,
  backgroundColor: Color.WIDGET_BACKGROUND
});

var IS_TEST = process.env.NODE_ENV === 'test';

var POPUP_BASE_STYLE = _extends((_extends4 = {
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  paddingTop: [Space.S_8, null, Space.S_48],
  paddingBottom: [Space.NONE, null, Space.S_48],
  '[role=dialog]': POPUP_DIALOG_MODAL_VIEW_STYLE
}, _extends4[Modal.CloseButton] = {
  opacity: 0,
  transition: 'opacity 0.3s'
}, _extends4), !IS_TEST && (_ref = {}, _ref["&:hover [role=dialog]:not(:hover) ~ " + Modal.CloseButton] = {
  opacity: 1
}, _ref));

var POPUP_MODAL_VIEW_STYLE = _extends({}, POPUP_BASE_STYLE);

var POPUP_COLORFUL_STYLE = _extends({}, POPUP_BASE_STYLE, {
  '[role=dialog]': POPUP_DIALOG_COLORFUL_STYLE
});

var POPUP_BOTTOM_SHEET_STYLE = _extends({}, POPUP_BASE_STYLE, {
  paddingTop: Space.S_48,
  '[role=dialog]': POPUP_DIALOG_BOTTOM_SHEET_STYLE
});

var POPUP_DIALOG_STYLE = _extends({}, POPUP_BASE_STYLE, {
  '[role=dialog]': POPUP_DIALOG_DIALOG_STYLE
});

var POPUP_RESULT_STYLE = _extends({}, POPUP_BASE_STYLE, {
  paddingBottom: [Space.S_24, null, Space.S_40],
  paddingX: Space.S_16,
  '[role=dialog]': POPUP_DIALOG_RESULT_STYLE
});

export var POPUP_STYLES = (_POPUP_STYLES = {}, _POPUP_STYLES[PopupVariant.MODAL_VIEW] = POPUP_MODAL_VIEW_STYLE, _POPUP_STYLES[PopupVariant.BOTTOM_SHEET] = POPUP_BOTTOM_SHEET_STYLE, _POPUP_STYLES[PopupVariant.RESULT] = POPUP_RESULT_STYLE, _POPUP_STYLES[PopupVariant.DIALOG] = POPUP_DIALOG_STYLE, _POPUP_STYLES[PopupVariant.COLORFUL] = POPUP_COLORFUL_STYLE, _POPUP_STYLES);
