// @revolut/ui-kit@11.21.2/esm/components/FanChart/FanChartCurrentValueLabel.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import React, { useRef } from 'react';
import { BarRounded } from '@visx/shape';
import { Text } from '@visx/text';
import { Color } from "../../constants";
import { useTheme } from "../../hooks";
import { themeColor } from "../../utils";
var BAR_PADDING = 2;
var INITIAL_RECT = {
  width: 40,
  height: 16
};
export var FanChartCurrentValueLabel = function FanChartCurrentValueLabel(_ref) {
  var x = _ref.x,
      y = _ref.y,
      label = _ref.label;
  var theme = useTheme();
  var textColor = themeColor(Color.GREY_TONE_50)({
    theme: theme
  });
  var fillColor = themeColor(Color.BACKGROUND)({
    theme: theme
  });
  var labelRef = useRef(null);

  var _ref2 = labelRef.current ? labelRef.current.getBoundingClientRect() : INITIAL_RECT,
      width = _ref2.width,
      height = _ref2.height;

  var barWidth = width + BAR_PADDING * 2;
  var barHeight = height + BAR_PADDING * 2;
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(BarRounded, {
    x: x,
    y: y - barHeight / 2,
    radius: 4,
    width: barWidth,
    height: barHeight,
    fill: fillColor,
    all: true
  }), /*#__PURE__*/React.createElement(Text, {
    x: x + BAR_PADDING,
    y: y,
    fill: textColor,
    fontSize: 10,
    fontFamily: theme.fonts["default"],
    fontWeight: 500,
    verticalAnchor: "middle",
    innerRef: labelRef
  }, label));
};
