// @revolut/ui-kit@11.21.2/esm/themes/compat.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import { LEGACY_SPACE } from "../tokens/default";
export var CompatTheme = {
  type: 'compat',
  space: LEGACY_SPACE,
  colorModes: {},
  // make it compatible with shape of `default` and `unified` themes
  textStyles: {
    "default": {
      variant: 'textStyles.primary'
    },
    body1: {
      variant: 'textStyles.primary'
    },
    body2: {
      variant: 'textStyles.secondary'
    },
    tableCaps: {
      variant: 'textStyles.h5'
    }
  },
  buttonStyles: {
    link: {
      variant: 'buttonStyles.text'
    }
  }
};
