// @revolut/ui-kit@11.21.2/esm/themes/unified.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import { UnifiedColorMode } from "../constants";
import { AVATAR_CIRCLE_SIZE_STYLES } from "../legacy/components/AvatarCircle/sizes";
import { BAR_STYLES } from "../components/Bar/variants";
import { BADGE_STYLES } from "../components/Badge/variants";
import { CIRCLE_VARIANTS } from "../components/Circle/variants";
import { CARD_STYLES } from "../legacy/components/Card/variants";
import { TAB_BAR_STYLES } from "../components/TabBar/variants";
import { LIST_STYLES } from "../components/List/variants";
import { CIRCLE_BUTTON_STYLES } from "../legacy/components/CircleButton/variants";
import { TEXT_BUTTON_STYLES } from "../components/TextButton/variants";
import { POPUP_STYLES } from "../components/Popup/variants";
import { STORY_STYLES } from "../components/Story/variants";
import { BOTTOM_NAV_STYLES } from "../components/BottomNav/variants";
import { BUTTON_SIZE_STYLES, BUTTON_STYLES, COLORS, COLOR_MODES, ELEVATIONS, FONTS, FONT_SIZES, LETTER_SPACINGS, LINE_HEIGHTS, RADII, SIZES, SPACE, TEXT_STYLES } from "../tokens/unified";
export var UnifiedTheme = {
  type: 'unified',
  mode: UnifiedColorMode.LIGHT,
  avatarCircleSizeStyles: AVATAR_CIRCLE_SIZE_STYLES,
  barStyles: BAR_STYLES,
  badgeStyles: BADGE_STYLES,
  buttonStyles: BUTTON_STYLES,
  buttonSizeStyles: BUTTON_SIZE_STYLES,
  cardStyles: CARD_STYLES,
  circleButtonStyles: CIRCLE_BUTTON_STYLES,
  popupStyles: POPUP_STYLES,
  storyStyles: STORY_STYLES,
  circleStyles: CIRCLE_VARIANTS,
  bottomNavStyles: BOTTOM_NAV_STYLES,
  colorModes: COLOR_MODES,
  colors: COLORS,
  elevations: ELEVATIONS,
  fonts: FONTS,
  fontSizes: FONT_SIZES,
  lineHeights: LINE_HEIGHTS,
  letterSpacings: LETTER_SPACINGS,
  radii: RADII,
  sizes: SIZES,
  space: SPACE,
  tabBarStyles: TAB_BAR_STYLES,
  listStyles: LIST_STYLES,
  textButtonStyles: TEXT_BUTTON_STYLES,
  textStyles: TEXT_STYLES
};
