// @revolut/ui-kit@11.21.2/esm/components/Subheader/Subheader.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import styled from 'styled-components';
import css from '@styled-system/css';
import { Color, Space, TextStyle } from "../../constants";
import { Flex } from "../Flex";
import { Text } from "../Text";

/** @private */
export var SubheaderBase = /*#__PURE__*/styled(Flex).withConfig({
  displayName: "Subheader__SubheaderBase",
  componentId: "sc-1coiuwy-0"
})(css({
  paddingY: Space.S_16,
  alignItems: 'center',
  justifyContent: 'space-between',
  color: Color.GREY_TONE_50
}));
export function Subheader(_ref) {
  var children = _ref.children,
      rest = _objectWithoutPropertiesLoose(_ref, ["children"]);

  return /*#__PURE__*/React.createElement(SubheaderBase, _extends({
    textStyle: TextStyle.H6
  }, rest), children != null ? children : /*#__PURE__*/React.createElement(React.Fragment, null, "\xA0"));
}
export function SubheaderTitle(props) {
  return /*#__PURE__*/React.createElement(Text, _extends({
    mr: Space.S_8,
    ellipsis: true
  }, props));
}
Subheader.Title = SubheaderTitle;
export function SubheaderSide(props) {
  return /*#__PURE__*/React.createElement(Text // room for focus styles ~4px
  , _extends({
    p: Space.S_4,
    m: Space.N_4,
    ellipsis: true
  }, props));
}
Subheader.Side = SubheaderSide;
