// @revolut/ui-kit@11.21.2/esm/components/Subheader/SubheaderSkeleton.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import React from 'react';
import { rem } from 'polished';
import { Skeleton } from "../Skeleton";
import { Subheader } from "./Subheader";
import { Color } from "../../constants";

var SubheaderTitleSkeleton = function SubheaderTitleSkeleton() {
  return /*#__PURE__*/React.createElement(Subheader.Title, null, /*#__PURE__*/React.createElement(Skeleton, {
    height: rem(16),
    width: rem(73),
    bg: Color.GREY_TONE_10,
    my: rem(3)
  }));
};

var SubheaderSideSkeleton = function SubheaderSideSkeleton() {
  return /*#__PURE__*/React.createElement(Subheader.Side, null, /*#__PURE__*/React.createElement(Skeleton, {
    height: rem(16),
    width: rem(43),
    bg: Color.GREY_TONE_10,
    my: rem(3)
  }));
};

export var SubheaderSkeleton = function SubheaderSkeleton(_ref) {
  var children = _ref.children;
  return /*#__PURE__*/React.createElement(Subheader, null, children != null ? children : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(SubheaderSkeleton.Title, null), /*#__PURE__*/React.createElement(SubheaderSkeleton.Side, null)));
};
SubheaderSkeleton.Title = SubheaderTitleSkeleton;
SubheaderSkeleton.Side = SubheaderSideSkeleton;
