// @revolut/ui-kit@11.21.2/esm/components/Table/constants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _ROW_HEIGHT;

export var BUTTON_SIZE = 16;
export var EXPAND_PADDING_STEP = 4;
export var SHOW_LOCK_SHADOW_ATTR = 'data-show-lock-shadow';
export var CellAlign = {
  START: 'start',
  END: 'end',
  CENTER: 'center'
};
export var TableRowHeight = {
  CONDENSED: 'condensed',
  NORMAL: 'normal',
  STRETCHED: 'stretched'
};
export var ROW_HEIGHT = (_ROW_HEIGHT = {}, _ROW_HEIGHT[TableRowHeight.CONDENSED] = 40, _ROW_HEIGHT[TableRowHeight.NORMAL] = 48, _ROW_HEIGHT[TableRowHeight.STRETCHED] = 56, _ROW_HEIGHT);
export var DEFAULT_ROW_HEIGHT = ROW_HEIGHT[TableRowHeight.NORMAL];
