// @revolut/ui-kit@11.21.2/esm/components/Table/TableFailedState.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import React from 'react';
import { TextStyle } from "../../constants";
import { Placeholder } from "../Placeholder";
export var TableFailedState = function TableFailedState(_ref) {
  var labelDescription = _ref.labelDescription,
      labelRetry = _ref.labelRetry,
      onRetry = _ref.onRetry;
  return /*#__PURE__*/React.createElement(Placeholder, null, /*#__PURE__*/React.createElement(Placeholder.Image, {
    src: "https://assets.revolut.com/web-ui-kit/images/3D018.png",
    srcSet: "https://assets.revolut.com/web-ui-kit/images/3D018@2x.png 2x, https://assets.revolut.com/web-ui-kit/images/3D018@3x.png 3x",
    size: 104
  }), /*#__PURE__*/React.createElement(Placeholder.Title, {
    use: "p",
    variant: TextStyle.TILE
  }, labelDescription), Boolean(onRetry) && /*#__PURE__*/React.createElement(Placeholder.Action, {
    onClick: onRetry
  }, labelRetry));
};
