// @revolut/ui-kit@11.21.2/esm/components/Table/TableBodySkeleton.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { forwardRef } from 'react';
import styled, { keyframes } from 'styled-components';
import { rem } from 'polished';
import { Color, Radius } from "../../constants";
import { themeColor } from "../../utils";
import { Box } from "../Box";
import { Skeleton } from "../Skeleton";
import { DEFAULT_ROW_HEIGHT } from "./constants";
var moveGradient = /*#__PURE__*/keyframes(["100%{transform:translateX(60%);}"]);
var RowSkeleton = /*#__PURE__*/styled(Skeleton).withConfig({
  displayName: "TableBodySkeleton__RowSkeleton",
  componentId: "sc-1pa4qct-0"
})(["border-bottom:1px solid ", ";height:var(--rui-table-row-height,", ");&::after{transform:translateX(-60%);background-image:linear-gradient( 90deg,rgba(255,245,255,0) 40%,", " 50%,rgba(255,255,255,0) 60% );animation-name:", ";}"], themeColor(Color.GREY_TONE_8), rem(DEFAULT_ROW_HEIGHT), themeColor(Color.GREY_20_20), moveGradient);
var DURATION = 3000;
export var TableBodySkeleton = /*#__PURE__*/forwardRef(function (props, ref) {
  return /*#__PURE__*/React.createElement(Box, _extends({
    ref: ref,
    role: "presentation"
  }, props), /*#__PURE__*/React.createElement(RowSkeleton, {
    bg: Color.BACKGROUND,
    shimmer: Color.GREY_20_20,
    radius: Radius.UNSET,
    duration: DURATION
  }), /*#__PURE__*/React.createElement(RowSkeleton, {
    bg: Color.BACKGROUND,
    shimmer: Color.GREY_20_20,
    radius: Radius.UNSET,
    duration: DURATION,
    delay: 200
  }), /*#__PURE__*/React.createElement(RowSkeleton, {
    bg: Color.BACKGROUND,
    shimmer: Color.GREY_20_20,
    radius: Radius.UNSET,
    duration: DURATION,
    delay: 400
  }));
});
