// @revolut/ui-kit@11.21.2/esm/components/PieChart/constants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _PIE_SIZE, _INNER_PIE_SIZE, _EMPTY_SPACE_SIZE, _GRAPH_WIDTH, _GRAPH_HEIGHT;

import { PieChartVariant } from "./variants";
export var PIE_SIZE = (_PIE_SIZE = {}, _PIE_SIZE[PieChartVariant.LARGE] = 186, _PIE_SIZE[PieChartVariant.SMALL] = 68, _PIE_SIZE);
export var INNER_PIE_SIZE = (_INNER_PIE_SIZE = {}, _INNER_PIE_SIZE[PieChartVariant.LARGE] = 117, _INNER_PIE_SIZE[PieChartVariant.SMALL] = 43, _INNER_PIE_SIZE);
export var EMPTY_SPACE_SIZE = (_EMPTY_SPACE_SIZE = {}, _EMPTY_SPACE_SIZE[PieChartVariant.LARGE] = 82, _EMPTY_SPACE_SIZE[PieChartVariant.SMALL] = 30, _EMPTY_SPACE_SIZE);
export var PIE_OPACITY = 0.8;
export var GRAPH_WIDTH = (_GRAPH_WIDTH = {}, _GRAPH_WIDTH[PieChartVariant.LARGE] = 343, _GRAPH_WIDTH[PieChartVariant.SMALL] = 68, _GRAPH_WIDTH);
export var GRAPH_HEIGHT = (_GRAPH_HEIGHT = {}, _GRAPH_HEIGHT[PieChartVariant.LARGE] = 270, _GRAPH_HEIGHT[PieChartVariant.SMALL] = 68, _GRAPH_HEIGHT);
