// @revolut/ui-kit@11.21.2/esm/styles/aspect-ratio.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import { system } from 'styled-system';
var aspectRatioSystem = system({
  aspectRatio: {
    // @ts-expect-error
    property: '--aspect-ratio',
    transform: function transform(value) {
      return "" + value;
    }
  }
});
export var aspectRatio = function aspectRatio(props) {
  return [aspectRatioSystem(props), props.aspectRatio ? {
    '&::before': {
      content: "''",
      "float": 'left',
      paddingBottom: "calc(1 / var(--aspect-ratio) * 100%)",
      width: 0
    },
    '&::after': {
      content: "''",
      clear: 'both',
      display: 'block'
    }
  } : null];
};
