// @revolut/ui-kit@11.21.2/esm/styles/stack.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import { system } from '@styled-system/core';
import { BASE_SPACE } from "../constants";

function transformAlign(input) {
  switch (input) {
    case 'start':
      return 'flex-start';

    case 'end':
      return 'flex-end';

    default:
      return input;
  }
}
/** @private */


export var stack = system({
  gap: {
    property: 'gap',
    scale: 'space',
    defaultScale: BASE_SPACE
  },
  align: {
    property: 'alignItems',
    transform: transformAlign
  }
});
