// @revolut/ui-kit@11.21.2/esm/components/Item/ItemSkeleton.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { rem } from 'polished';
import { Space } from "../../constants";
import { Box } from "../Box";
import { Skeleton } from "../Skeleton";
import { AvatarSkeleton } from "../Avatar";
import { Item } from "./Item";
export var ItemAvatarSkeleton = function ItemAvatarSkeleton(props) {
  return /*#__PURE__*/React.createElement(Item.Avatar, props, /*#__PURE__*/React.createElement(AvatarSkeleton, null));
};

var ItemTitleSkeleton = function ItemTitleSkeleton(props) {
  return /*#__PURE__*/React.createElement(Item.Title, props, /*#__PURE__*/React.createElement(Skeleton, {
    height: rem(16),
    width: rem(320),
    maxWidth: "68%"
  }));
};

var ItemDescriptionSkeleton = function ItemDescriptionSkeleton(props) {
  return /*#__PURE__*/React.createElement(Item.Description, _extends({
    mt: Space.S_6
  }, props), /*#__PURE__*/React.createElement(Skeleton, {
    height: rem(16),
    width: rem(160),
    maxWidth: "34%"
  }));
};

var ItemValueSkeleton = function ItemValueSkeleton(props) {
  return /*#__PURE__*/React.createElement(Item.Value, props, /*#__PURE__*/React.createElement(Skeleton, {
    height: rem(16),
    width: rem(40),
    maxWidth: "100%"
  }));
};

export var ItemContentSkeleton = function ItemContentSkeleton(_ref) {
  var children = _ref.children,
      rest = _objectWithoutPropertiesLoose(_ref, ["children"]);

  return /*#__PURE__*/React.createElement(Item.Content, _extends({
    height: children ? undefined : rem(44)
  }, rest), /*#__PURE__*/React.createElement(Box, {
    my: "auto"
  }, children != null ? children : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(ItemTitleSkeleton, null), /*#__PURE__*/React.createElement(ItemDescriptionSkeleton, null))));
};
export var ItemSideSkeleton = function ItemSideSkeleton(_ref2) {
  var children = _ref2.children,
      rest = _objectWithoutPropertiesLoose(_ref2, ["children"]);

  return /*#__PURE__*/React.createElement(Item.Side, rest, children != null ? children : /*#__PURE__*/React.createElement(ItemValueSkeleton, null));
};
export var ItemSkeleton = function ItemSkeleton(_ref3) {
  var children = _ref3.children,
      rest = _objectWithoutPropertiesLoose(_ref3, ["children"]);

  return /*#__PURE__*/React.createElement(Item, rest, children != null ? children : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(ItemSkeleton.Avatar, null), /*#__PURE__*/React.createElement(ItemSkeleton.Content, null), /*#__PURE__*/React.createElement(ItemSkeleton.Side, null)));
};
ItemSkeleton.Avatar = ItemAvatarSkeleton;
ItemSkeleton.Content = ItemContentSkeleton;
ItemSkeleton.Title = ItemTitleSkeleton;
ItemSkeleton.Description = ItemDescriptionSkeleton;
ItemSkeleton.Side = ItemSideSkeleton;
ItemSkeleton.Value = ItemValueSkeleton;
