// @revolut/ui-kit@11.21.2/esm/components/List/variants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _LIST_STYLES;

import { Space, TextStyle } from "../../constants";
var DEFAULT_STYLE = {
  listStyle: 'none',
  '> li': {
    variant: "textStyles." + TextStyle.PRIMARY,
    paddingY: Space.S_16
  }
};
var COMPACT_STYLE = {
  listStyle: 'none',
  '> li': {
    variant: "textStyles." + TextStyle.CAPTION
  },
  '> li + li': {
    marginTop: Space.S_4
  }
};
export var ListVariant = {
  DEFAULT: 'default',
  COMPACT: 'compact'
};
export var LIST_STYLES = (_LIST_STYLES = {}, _LIST_STYLES[ListVariant.DEFAULT] = DEFAULT_STYLE, _LIST_STYLES[ListVariant.COMPACT] = COMPACT_STYLE, _LIST_STYLES);
