// @revolut/ui-kit@11.21.2/esm/components/Transitions/TransitionFade.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { Transition } from 'react-transition-group';
import { Duration } from "../../constants";
import { DURATIONS } from "../../tokens/default";
import { noop } from "../../utils";
import { addAnimationEndListener, animate } from "./utils";
export var TransitionFade = function TransitionFade(_ref) {
  var _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? DURATIONS[Duration.SM] : _ref$duration,
      _ref$delay = _ref.delay,
      delay = _ref$delay === void 0 ? 0 : _ref$delay,
      _ref$easing = _ref.easing,
      easing = _ref$easing === void 0 ? 'linear' : _ref$easing,
      enterAnimation = _ref.enterAnimation,
      exitAnimation = _ref.exitAnimation,
      _ref$unmountOnExit = _ref.unmountOnExit,
      unmountOnExit = _ref$unmountOnExit === void 0 ? true : _ref$unmountOnExit,
      _ref$onEnter = _ref.onEnter,
      _onEnter = _ref$onEnter === void 0 ? noop : _ref$onEnter,
      _ref$onExit = _ref.onExit,
      _onExit = _ref$onExit === void 0 ? noop : _ref$onExit,
      _ref$appear = _ref.appear,
      appear = _ref$appear === void 0 ? true : _ref$appear,
      rest = _objectWithoutPropertiesLoose(_ref, ["duration", "delay", "easing", "enterAnimation", "exitAnimation", "unmountOnExit", "onEnter", "onExit", "appear"]);

  return /*#__PURE__*/React.createElement(Transition, _extends({
    addEndListener: addAnimationEndListener,
    unmountOnExit: unmountOnExit,
    appear: appear,
    onEnter: function onEnter(node, isAppearing) {
      _onEnter(node, isAppearing);

      node.style.opacity = '0';
      return animate(_extends({
        targets: node,
        opacity: [0, 1],
        duration: duration,
        delay: delay,
        easing: easing
      }, enterAnimation));
    },
    onExit: function onExit(node) {
      _onExit(node);

      animate(_extends({
        targets: node,
        opacity: [1, 0],
        duration: duration,
        delay: delay,
        easing: easing
      }, exitAnimation));
    }
  }, rest));
};
