// @revolut/ui-kit@11.21.2/esm/components/Position/Position.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled from 'styled-components';
import { bottom, left, right, top, zIndex } from 'styled-system';
import { Box } from "../Box";
export var Position = /*#__PURE__*/styled(Box).withConfig({
  displayName: "Position",
  componentId: "nh2ik7-0"
})(["", " ", " ", " ", " ", ""], top, bottom, left, right, zIndex);
export var Fixed = /*#__PURE__*/styled(Position).withConfig({
  displayName: "Position__Fixed",
  componentId: "nh2ik7-1"
})(["position:fixed;"]);
export var Absolute = /*#__PURE__*/styled(Position).withConfig({
  displayName: "Position__Absolute",
  componentId: "nh2ik7-2"
})(["position:absolute;"]);
export var Relative = /*#__PURE__*/styled(Position).withConfig({
  displayName: "Position__Relative",
  componentId: "nh2ik7-3"
})(["position:relative;"]);
export var Sticky = /*#__PURE__*/styled(Position).withConfig({
  displayName: "Position__Sticky",
  componentId: "nh2ik7-4"
})(["position:sticky;"]);
