// @revolut/ui-kit@11.21.2/esm/components/Banner/BannerSkeleton.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { rem } from 'polished';
import { Space } from "../../constants";
import { AvatarSkeleton } from "../Avatar";
import { Skeleton } from "../Skeleton";
import { ItemSkeleton } from "../Item";
import { Banner } from "./Banner";

var BannerAvatarSkeleton = function BannerAvatarSkeleton(props) {
  return /*#__PURE__*/React.createElement(Banner.Avatar, null, /*#__PURE__*/React.createElement(AvatarSkeleton, props));
};

var BannerDescriptionSkeleton = function BannerDescriptionSkeleton(props) {
  return /*#__PURE__*/React.createElement(Banner.Description, _extends({
    mt: Space.S_6
  }, props), /*#__PURE__*/React.createElement(Skeleton, {
    height: rem(16),
    width: rem(160),
    maxWidth: "34%"
  }));
};

var BannerContentSkeleton = function BannerContentSkeleton(_ref) {
  var children = _ref.children,
      rest = _objectWithoutPropertiesLoose(_ref, ["children"]);

  return /*#__PURE__*/React.createElement(Banner.Content, _extends({
    minHeight: children ? undefined : rem(44)
  }, rest), children != null ? children : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(BannerSkeleton.Title, null), /*#__PURE__*/React.createElement(BannerDescriptionSkeleton, null)));
};

export var BannerSkeleton = function BannerSkeleton(_ref2) {
  var children = _ref2.children,
      rest = _objectWithoutPropertiesLoose(_ref2, ["children"]);

  return /*#__PURE__*/React.createElement(Banner, rest, children != null ? children : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(BannerSkeleton.Avatar, null), /*#__PURE__*/React.createElement(BannerSkeleton.Content, null)));
};
BannerSkeleton.Avatar = BannerAvatarSkeleton;
BannerSkeleton.Description = BannerDescriptionSkeleton;
BannerSkeleton.Content = BannerContentSkeleton;
BannerSkeleton.Title = ItemSkeleton.Title;
