// @revolut/ui-kit@11.21.2/esm/components/ProgressWidget/ProgressWidget.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _css;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import styled from 'styled-components';
import { css } from '@styled-system/css';
import { Color, Space, TextStyle } from "../../constants";
import { themeTextStyle } from "../../utils";
import { Text } from "../Text";
import { Widget } from "../Widget";
import { Progress } from "../Progress";
import { ProgressSteps } from "../ProgressSteps";
import { AccessibleId, useAccessibleId } from "../AccessibleId";
import { getDescriptionId, getLabelId } from "./utils";
// Converts the component to styled to be able select it in parent CSS
var ProgressWidgetLabel = /*#__PURE__*/styled(Text).withConfig({
  displayName: "ProgressWidget__ProgressWidgetLabel",
  componentId: "ra2io6-0"
})([""]);
ProgressWidgetLabel.defaultProps = {
  color: Color.FOREGROUND,
  variant: TextStyle.CAPTION
};
var ProgressWidgetTitleBase = /*#__PURE__*/styled(Text).withConfig({
  displayName: "ProgressWidget__ProgressWidgetTitleBase",
  componentId: "ra2io6-1"
})({
  gridRow: 1
});

var ProgressWidgetTitle = function ProgressWidgetTitle(props) {
  var id = useAccessibleId();
  return /*#__PURE__*/React.createElement(ProgressWidgetTitleBase, _extends({
    id: getLabelId(id)
  }, props));
};

ProgressWidgetTitle.defaultProps = {
  color: Color.FOREGROUND,
  variant: TextStyle.TILE
};
var ProgressWidgetDescriptionBase = /*#__PURE__*/styled(Text).withConfig({
  displayName: "ProgressWidget__ProgressWidgetDescriptionBase",
  componentId: "ra2io6-2"
})(themeTextStyle(TextStyle.CAPTION), {
  gridRow: 1,
  textAlign: 'right'
});
ProgressWidgetDescriptionBase.defaultProps = {
  placeSelf: 'center'
};

var ProgressWidgetDescription = function ProgressWidgetDescription(props) {
  var id = useAccessibleId();
  return /*#__PURE__*/React.createElement(ProgressWidgetDescriptionBase, _extends({
    id: getDescriptionId(id)
  }, props));
};

var StyledWidget = /*#__PURE__*/styled(Widget).withConfig({
  displayName: "ProgressWidget__StyledWidget",
  componentId: "ra2io6-3"
})(css((_css = {
  display: 'grid',
  gap: Space.S_16,
  gridTemplateColumns: '1fr auto',
  padding: Space.S_16
}, _css[ProgressWidgetLabel + ":nth-of-type(even)"] = {
  textAlign: 'right'
}, _css[ProgressWidgetLabel + ":last-of-type:not(:nth-of-type(even))"] = {
  gridColumn: '1 / span 2'
}, _css)));

var ProgressWidgetBase = function ProgressWidgetBase(props) {
  return /*#__PURE__*/React.createElement(AccessibleId, null, /*#__PURE__*/React.createElement(StyledWidget, props));
};

var ProgressBase = function ProgressBase(props) {
  var id = useAccessibleId();
  return /*#__PURE__*/React.createElement(Progress, _extends({
    "aria-labelledby": getLabelId(id),
    "aria-describedby": getDescriptionId(id)
  }, props));
};

export var ProgressWidget = Object.assign(ProgressWidgetBase, {
  Title: ProgressWidgetTitle,
  Description: ProgressWidgetDescription,
  Label: ProgressWidgetLabel,
  Progress: styled(ProgressBase).withConfig({
    displayName: "ProgressWidget__Progress",
    componentId: "ra2io6-4"
  })({
    gridRow: 2,
    gridColumn: '1 / span 2'
  }),
  Steps: styled(ProgressSteps).withConfig({
    displayName: "ProgressWidget__Steps",
    componentId: "ra2io6-5"
  })({
    gridRow: 2,
    gridColumn: '1 / span 2'
  })
});
