// @revolut/ui-kit@11.21.2/esm/tokens/default/button-size-styles.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _BUTTON_SIZE_STYLES;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { ButtonSizeStyle, Radius, Size, Space } from "../../constants";
var BUTTON_XS = {
  variant: 'textStyles.secondary',
  height: Size.BUTTON_XS,
  width: 'max-content',
  maxWidth: '100%',
  paddingX: Space.S_4,
  borderRadius: Radius.BUTTON_XS
};
var BUTTON_SM = {
  variant: 'textStyles.secondary',
  height: Size.BUTTON_SM,
  width: 'max-content',
  maxWidth: '100%',
  paddingX: Space.S_16,
  borderRadius: Radius.BUTTON_SM
};
var BUTTON_MD = {
  variant: 'textStyles.primary',
  height: [Size.BUTTON_MD, null, Size.BUTTON_MD_PLUS],
  width: '100%',
  paddingX: Space.S_16,
  borderRadius: Radius.BUTTON_MD
};

var BUTTON_TAG = _extends({}, BUTTON_SM, {
  height: 'auto',
  variant: 'textStyles.secondary',
  paddingX: Space.S_8,
  paddingY: Space.S_4
});

export var BUTTON_SIZE_STYLES = (_BUTTON_SIZE_STYLES = {}, _BUTTON_SIZE_STYLES[ButtonSizeStyle.XS] = BUTTON_XS, _BUTTON_SIZE_STYLES[ButtonSizeStyle.SM] = BUTTON_SM, _BUTTON_SIZE_STYLES[ButtonSizeStyle.MD] = BUTTON_MD, _BUTTON_SIZE_STYLES[ButtonSizeStyle.TAG] = BUTTON_TAG, _BUTTON_SIZE_STYLES);
