// @revolut/ui-kit@11.21.2/esm/tokens/default/text-styles.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _TEXT_STYLES;

import { Font, FontSize, LetterSpace, LineHeight, TextStyle } from "../../constants";
export var TEXT_STYLES = (_TEXT_STYLES = {}, _TEXT_STYLES[TextStyle.DEFAULT] = null, _TEXT_STYLES[TextStyle.PRIMARY] = {
  fontFamily: Font.DEFAULT,
  fontWeight: 400,
  fontSize: FontSize.PRIMARY,
  lineHeight: LineHeight.PRIMARY,
  letterSpacing: LetterSpace.PRIMARY
}, _TEXT_STYLES[TextStyle.SECONDARY] = {
  fontFamily: Font.DEFAULT,
  fontWeight: 400,
  fontSize: FontSize.SECONDARY,
  lineHeight: LineHeight.SECONDARY,
  letterSpacing: LetterSpace.SECONDARY
}, _TEXT_STYLES[TextStyle.BULLET] = {
  fontFamily: Font.BRAND,
  fontWeight: 600,
  fontSize: FontSize.BULLET,
  lineHeight: LineHeight.BULLET,
  letterSpacing: LetterSpace.BULLET
}, _TEXT_STYLES[TextStyle.H0] = {
  fontFamily: Font.DEFAULT,
  fontWeight: 700,
  fontSize: [FontSize.H0, null, FontSize.H0_PLUS],
  lineHeight: [LineHeight.H0, null, LineHeight.H0_PLUS],
  letterSpacing: LetterSpace.H0
}, _TEXT_STYLES[TextStyle.H1] = {
  fontFamily: Font.DEFAULT,
  fontWeight: 600,
  fontSize: [FontSize.H1, null, FontSize.H1_PLUS],
  lineHeight: [LineHeight.H1, null, LineHeight.H1_PLUS],
  letterSpacing: LetterSpace.H1
}, _TEXT_STYLES[TextStyle.H2] = {
  fontFamily: Font.DEFAULT,
  fontWeight: 600,
  fontSize: [FontSize.H2, null, FontSize.H2_PLUS],
  lineHeight: [LineHeight.H2, null, LineHeight.H2_PLUS],
  letterSpacing: LetterSpace.H2
}, _TEXT_STYLES[TextStyle.H3] = {
  fontFamily: Font.DEFAULT,
  fontWeight: 600,
  fontSize: [FontSize.H3, null, FontSize.H3_PLUS],
  lineHeight: [LineHeight.H3, null, LineHeight.H3_PLUS],
  letterSpacing: LetterSpace.H3
}, _TEXT_STYLES[TextStyle.H4] = {
  fontFamily: Font.DEFAULT,
  fontWeight: 500,
  fontSize: [FontSize.H4, null, FontSize.H4_PLUS],
  lineHeight: [LineHeight.H4, null, LineHeight.H4_PLUS],
  letterSpacing: LetterSpace.H4
}, _TEXT_STYLES[TextStyle.H5] = {
  fontFamily: Font.DEFAULT,
  fontWeight: 500,
  fontSize: FontSize.H5,
  lineHeight: LineHeight.H5,
  textTransform: 'uppercase',
  letterSpacing: LetterSpace.H5
}, _TEXT_STYLES[TextStyle.H6] = {
  fontFamily: Font.DEFAULT,
  fontWeight: 500,
  fontSize: FontSize.PRIMARY,
  lineHeight: LineHeight.PRIMARY,
  letterSpacing: LetterSpace.PRIMARY
}, _TEXT_STYLES[TextStyle.HELPER] = {
  fontFamily: Font.DEFAULT,
  fontWeight: 400,
  fontSize: FontSize.HELPER,
  lineHeight: LineHeight.HELPER,
  letterSpacing: LetterSpace.HELPER
}, _TEXT_STYLES[TextStyle.AVATAR] = {
  fontFamily: Font.DEFAULT,
  fontSize: FontSize.AVATAR,
  lineHeight: LineHeight.AVATAR,
  fontWeight: 500,
  textTransform: 'uppercase'
}, _TEXT_STYLES[TextStyle.BUTTON] = {
  fontWeight: 500
}, _TEXT_STYLES[TextStyle.ACTION] = {
  variant: "textStyles." + TextStyle.PRIMARY,
  fontWeight: 500
}, _TEXT_STYLES[TextStyle.TEXT_BUTTON] = {
  variant: "textStyles." + TextStyle.PRIMARY
}, _TEXT_STYLES[TextStyle.SMALL] = {
  variant: "textStyles." + TextStyle.HELPER
}, _TEXT_STYLES[TextStyle.TINY] = {
  variant: "textStyles." + TextStyle.HELPER
}, _TEXT_STYLES[TextStyle.CAPTION] = {
  variant: "textStyles." + TextStyle.HELPER
}, _TEXT_STYLES[TextStyle.TILE] = {
  variant: "textStyles." + TextStyle.PRIMARY
}, _TEXT_STYLES[TextStyle.INPUT] = {
  variant: "textStyles." + TextStyle.PRIMARY
}, _TEXT_STYLES);
