// @revolut/ui-kit@11.21.2/esm/components/Bullet/Bullet.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled from 'styled-components';
import css from '@styled-system/css';
import { Radius, Size, TextStyle } from "../../constants";
import { themeTextStyle } from "../../utils";
import { Circle } from "../Circle";
export var Bullet = /*#__PURE__*/styled(Circle).withConfig({
  displayName: "Bullet",
  componentId: "ko2x6c-0"
})(themeTextStyle(TextStyle.BULLET), css({
  borderRadius: Radius.ROUND,
  borderStyle: 'solid',
  borderWidth: Size.S_2
}));
Bullet.defaultProps = {
  size: Size.S_24
};
