// @revolut/ui-kit@11.21.2/esm/components/Badge/variants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _BADGE_STYLES;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { Color, Space } from "../../constants";
export var BadgeVariant = {
  DEFAULT: 'default',
  DOT: 'dot',
  OUTLINE: 'outline'
};
var DEFAULT_STYLE = {
  color: Color.BACKGROUND,
  backgroundColor: Color.RED,
  ':hover:enabled': {
    cursor: 'pointer'
  }
};

var DOT_STYLE = _extends({}, DEFAULT_STYLE, {
  margin: Space.S_4
});

var OUTLINE_STYLE = _extends({}, DEFAULT_STYLE, {
  backgroundColor: Color.BLUE,
  boxSizing: 'content-box',
  borderStyle: 'solid',
  borderColor: Color.BACKGROUND,
  borderWidth: Space.S_2
});

export var BADGE_STYLES = (_BADGE_STYLES = {}, _BADGE_STYLES[BadgeVariant.DEFAULT] = DEFAULT_STYLE, _BADGE_STYLES[BadgeVariant.DOT] = DOT_STYLE, _BADGE_STYLES[BadgeVariant.OUTLINE] = OUTLINE_STYLE, _BADGE_STYLES);
