// @revolut/ui-kit@11.21.2/esm/legacy/components/Grid/Grid.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import React from 'react';
import styled from 'styled-components';
import { css } from '@styled-system/css';
import { colsAndOffsetStyle } from "./utils";
import { Box } from "../../../components/Box";
import { Flex } from "../../../components/Flex";
var GridContainer = /*#__PURE__*/styled(Box).withConfig({
  displayName: "Grid__GridContainer",
  componentId: "sc-1m1x97q-0"
})(["", ""], css({
  paddingX: 'grid'
}));
GridContainer.defaultProps = {
  mx: 'auto',
  width: '100%'
};
var GridRow = /*#__PURE__*/styled(Flex).withConfig({
  displayName: "Grid__GridRow",
  componentId: "sc-1m1x97q-1"
})(["", ""], css({
  marginX: '-grid'
}));
GridRow.defaultProps = {
  flex: '0 0 auto',
  flexWrap: 'wrap'
};

var GridColBase = function GridColBase(props) {
  return /*#__PURE__*/React.createElement(Box, props);
};

var GridCol = /*#__PURE__*/styled(GridColBase).withConfig({
  displayName: "Grid__GridCol",
  componentId: "sc-1m1x97q-2"
})(["", " ", ""], css({
  paddingX: 'grid'
}), colsAndOffsetStyle);
GridCol.displayName = 'Grid.Col';
GridCol.defaultProps = {
  flex: '1 0 auto'
};
/** @deprecated */

export var Grid = Object.assign(GridContainer, {
  Row: GridRow,
  Col: GridCol
}); // COMPAT

/** @deprecated Use `Grid` instead */

export var Container = Grid;
/** @deprecated Use `Grid.Row` instead */

export var Row = GridRow;
/** @deprecated Use `Grid.Col` instead */

export var Col = GridCol;
