// @revolut/ui-kit@11.21.2/esm/components/Carousel/styles.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled from 'styled-components';
import css from '@styled-system/css';
import { Space } from "../../constants";
import { themeSpace } from "../../utils";
import { Flex } from "../Flex";
import { Box } from "../Box";
import { CarouselAlignVariant } from "./types";
export var CarouselItem = /*#__PURE__*/styled(Box).withConfig({
  displayName: "styles__CarouselItem",
  componentId: "sc-1eb9j43-0"
})(css({
  flex: '0 0 auto',
  // room for focus styles
  paddingX: [Space.S_4, null, Space.S_8],
  paddingY: Space.S_4
}));
CarouselItem.defaultProps = {
  role: 'group',
  minWidth: 0
};
export var CarouselRadiusClip = /*#__PURE__*/styled(Box).withConfig({
  displayName: "styles__CarouselRadiusClip",
  componentId: "sc-1eb9j43-1"
})(function (_ref) {
  var radius = _ref.radius;
  return css({
    ':not(:focus-within)': {
      overflow: [null, null, 'hidden'],
      borderRadius: radius
    }
  });
});
export var CarouselSpace = /*#__PURE__*/styled(Box).withConfig({
  displayName: "styles__CarouselSpace",
  componentId: "sc-1eb9j43-2"
})(css({
  marginY: Space.N_4,
  marginX: [Space.N_20, null, Space.N_8]
}));
export var CarouselScroll = /*#__PURE__*/styled(Flex).withConfig({
  displayName: "styles__CarouselScroll",
  componentId: "sc-1eb9j43-3"
})(function (_ref2) {
  var _css;

  var align = _ref2.align,
      alignWidth = _ref2.alignWidth;
  return css((_css = {
    overflowX: 'scroll',
    scrollSnapType: 'x mandatory',
    msOverflowStyle: 'none',
    scrollbarWidth: 'none',
    '&::-webkit-scrollbar': {
      display: 'none'
    },
    scrollPaddingLeft: function scrollPaddingLeft(theme) {
      return alignWidth ? undefined : [themeSpace(Space.S_16)({
        theme: theme
      }), null, themeSpace(Space.NONE)({
        theme: theme
      })];
    },
    paddingX: alignWidth ? "calc((100% - " + alignWidth + ") / 4 + 12px)" : [Space.S_16, null, Space.NONE]
  }, _css[CarouselItem] = {
    scrollSnapAlign: align === CarouselAlignVariant.SIDE ? 'start' : 'center'
  }, _css));
});
export var CarouselItemPlaceholder = /*#__PURE__*/styled(CarouselItem).withConfig({
  displayName: "styles__CarouselItemPlaceholder",
  componentId: "sc-1eb9j43-4"
})(["scroll-snap-align:start;"]);
CarouselItemPlaceholder.defaultProps = {
  hide: 'md-*',
  height: 1,
  minWidth: {
    all: Space.S_16,
    md: Space.S_24
  }
};
