// @revolut/ui-kit@11.21.2/esm/components/Text/styles.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import { style } from 'styled-system';
import { mq } from "../../styles";
import { isString } from "../../utils";
export var whiteSpace = style({
  prop: 'whiteSpace'
});
export var textOverflow = style({
  prop: 'textOverflow'
});
export var ellipsis = function ellipsis(props) {
  return props.ellipsis && {
    maxWidth: props.ellipsis === true ? '100%' : props.ellipsis,
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap',
    wordWrap: 'normal'
  };
};
export function lineClamp(props) {
  if (typeof props.lineClamp !== 'number') return null;
  var result = {
    display: '-webkit-box',
    '-webkit-line-clamp': "" + props.lineClamp,
    '-webkit-box-orient': 'vertical',
    overflow: 'hidden'
  };

  if (isString(props.hide)) {
    var _ref;

    return _ref = {}, _ref["@media not " + mq(props.hide)(props)] = result, _ref;
  }

  return props.hide === true ? null : result;
}
