// @revolut/ui-kit@11.21.2/esm/components/Box/Box.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled from 'styled-components';
import { css } from '@styled-system/css';
import { alignSelf, border, color, display, flex, fontSize, gridArea, height, maxHeight, maxWidth, minHeight, minWidth, order, overflow, size, space, width } from 'styled-system';
import { aspectRatio, elevation, hide, placeSelf, radius, textStyle, variantKey } from "../../styles";
import { Base } from "../Base";
export var Box = /*#__PURE__*/styled(Base).withConfig({
  displayName: "Box",
  componentId: "sc-1g1k12l-0"
})({
  boxSizing: 'border-box',
  margin: 0,
  padding: 0,
  font: 'inherit'
}, function (props) {
  return props.__css != null ? css(props.__css) : null;
}, variantKey, space, color, width, fontSize, flex, order, overflow, alignSelf, height, minWidth, maxWidth, minHeight, maxHeight, size, radius, elevation, border, textStyle, display, hide, aspectRatio, gridArea, placeSelf);
Box.displayName = 'Box';
Box.defaultProps = {
  variantKey: 'boxStyles',
  variant: 'default'
};
