// @revolut/ui-kit@11.21.2/esm/components/TabBar/variants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _TAB_BAR_STYLES;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { mix } from 'polished';
import { Color, Radius, Size, Space, TextStyle, Transition } from "../../constants";
import { TabBarVariant } from "./constants";
import { rgbaSafe, themeColor } from "../../utils";
var TAB_BASE_STYLE = {
  variant: "textStyles." + TextStyle.H6,
  fontWeight: 500,
  color: Color.GREY_TONE_50,
  transition: function transition(theme) {
    var _theme$transitions;

    return (_theme$transitions = theme.transitions) == null ? void 0 : _theme$transitions[Transition.BUTTON];
  }
};

var TAB_DEFAULT_STYLE = _extends({}, TAB_BASE_STYLE, {
  paddingY: Space.S_16,
  '&[aria-selected=true], &:hover': {
    color: Color.PRIMARY
  }
});

var TAB_SEGMENTED_STYLE = _extends({}, TAB_BASE_STYLE, {
  flex: 1,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  minHeight: Size.TAB_SEGMENTED,
  margin: Space.S_2,
  paddingX: Space.S_4,
  borderRadius: Radius.TAB_INNER,
  '&[aria-selected=true], &:hover': {
    color: Color.FOREGROUND
  },
  '&:disabled, &[aria-disabled=true]': {
    color: Color.GREY_TONE_20
  },
  '&[aria-selected=true]:disabled, &[aria-selected=true][aria-disabled=true]': {
    color: function color(theme) {
      return rgbaSafe(themeColor(Color.FOREGROUND)({
        theme: theme
      }), 0.3);
    }
  },
  '&[aria-selected=true]:hover': {
    backgroundColor: function backgroundColor(theme) {
      return mix(0.02, theme.colors[Color.FOREGROUND], theme.colors[Color.SEGMENTED_ACTIVE]);
    }
  },
  '&[aria-selected=true]:active': {
    backgroundColor: function backgroundColor(theme) {
      return mix(0.03, theme.colors[Color.FOREGROUND], theme.colors[Color.SEGMENTED_ACTIVE]);
    }
  },
  '&:focus-visible, &.focus-visible': {
    color: Color.FOREGROUND,
    backgroundColor: Color.SEGMENTED_ACTIVE
  }
});

var TAB_NAVIGATION_STYLE = _extends({}, TAB_BASE_STYLE, {
  display: 'flex',
  alignItems: 'center',
  minHeight: Size.TAB_NAVIGATION,
  // room for focus styles ~4px
  margin: Space.S_4,
  paddingY: Space.S_4,
  paddingX: Space.S_16,
  lineHeight: 1,
  borderRadius: Radius.TAB_INNER,
  '&:last-child': {
    marginRight: Space.NONE
  },
  '&[aria-selected=true], &:hover': {
    color: Color.FOREGROUND
  },
  '&:disabled, &[aria-disabled=true]': {
    color: Color.GREY_TONE_20
  },
  '&[aria-selected=true]:disabled, &[aria-selected=true][aria-disabled=true]': {
    color: function color(theme) {
      return rgbaSafe(themeColor(Color.FOREGROUND)({
        theme: theme
      }), 0.3);
    }
  },
  '&[aria-selected=true]:hover': {
    backgroundColor: function backgroundColor(theme) {
      return mix(0.02, theme.colors[Color.FOREGROUND], theme.colors[Color.WIDGET_BACKGROUND]);
    },
    marginY: Space.NONE,
    borderRadius: Radius.TAB_OUTER
  },
  '&[aria-selected=true]:active': {
    backgroundColor: function backgroundColor(theme) {
      return mix(0.03, theme.colors[Color.FOREGROUND], theme.colors[Color.WIDGET_BACKGROUND]);
    },
    marginY: Space.NONE,
    borderRadius: Radius.TAB_OUTER
  },
  '&:focus-visible, &.focus-visible': {
    color: Color.FOREGROUND,
    backgroundColor: Color.WIDGET_BACKGROUND
  }
});

var INDICATOR_DEFAULT_HORIZONTAL_STYLE = {
  bottom: 0,
  borderBottom: '2px solid',
  borderBottomColor: Color.PRIMARY
};
var INDICATOR_DEFAULT_VERTICAL_STYLE = {
  borderLeft: '2px solid',
  borderLeftColor: Color.PRIMARY
};
var INDICATOR_SEGMENTED_BASE_STYLE = {
  backgroundColor: Color.SEGMENTED_ACTIVE,
  borderRadius: Radius.TAB_INNER,
  height: Size.INDICATOR_SEGMENTED
};

var INDICATOR_SEGMENTED_HORIZONTAL_STYLE = _extends({}, INDICATOR_SEGMENTED_BASE_STYLE, {
  top: 0,
  bottom: 0,
  marginY: Space.S_2
});

var INDICATOR_SEGMENTED_VERTICAL_STYLE = _extends({}, INDICATOR_SEGMENTED_BASE_STYLE, {
  left: 0,
  right: 0,
  marginX: Space.S_2
});

var INDICATOR_NAVIGATION_BASE_STYLE = {
  backgroundColor: Color.WIDGET_BACKGROUND,
  borderRadius: Radius.TAB_OUTER,
  height: Size.INDICATOR_NAVIGATION
};

var INDICATOR_NAVIGATION_HORIZONTAL_STYLE = _extends({}, INDICATOR_NAVIGATION_BASE_STYLE, {
  top: 0,
  bottom: 0
});

var INDICATOR_NAVIGATION_VERTICAL_STYLE = _extends({}, INDICATOR_NAVIGATION_BASE_STYLE, {
  left: 0,
  right: 0
});

var TAB_BAR_DEFAULT_STYLE = {
  borderColor: Color.GREY_TONE_10,
  '[role=tab]': TAB_DEFAULT_STYLE,
  '&[aria-orientation=horizontal]::before': INDICATOR_DEFAULT_HORIZONTAL_STYLE,
  '&[aria-orientation=horizontal] [role=tab] + [role=tab]': {
    marginLeft: Space.S_32
  },
  '&[aria-orientation=vertical]::before': INDICATOR_DEFAULT_VERTICAL_STYLE,
  '&[aria-orientation=vertical] [role=tab]': {
    paddingX: Space.S_20,
    paddingY: Space.S_8
  }
};
var TAB_BAR_SEGMENTED_STYLE = {
  backgroundColor: Color.SEGMENTED_BACKGROUND,
  borderRadius: Radius.TAB_OUTER,
  '[role=tab]': TAB_SEGMENTED_STYLE,
  '&[aria-orientation=horizontal]::before': INDICATOR_SEGMENTED_HORIZONTAL_STYLE,
  '&[aria-orientation=vertical]::before': INDICATOR_SEGMENTED_VERTICAL_STYLE
};
var TAB_BAR_NAVIGATION_STYLE = {
  marginX: Space.N_4,
  '[role=tab]': TAB_NAVIGATION_STYLE,
  '&[aria-orientation=horizontal]::before': INDICATOR_NAVIGATION_HORIZONTAL_STYLE,
  '&[aria-orientation=vertical]::before': INDICATOR_NAVIGATION_VERTICAL_STYLE,
  '&[aria-orientation=horizontal] [role=tab]': {
    justifyContent: 'center'
  },
  '&[aria-orientation=vertical] [role=tab]': {
    marginY: Space.S_4
  }
};
export var TAB_BAR_STYLES = (_TAB_BAR_STYLES = {}, _TAB_BAR_STYLES[TabBarVariant.DEFAULT] = TAB_BAR_DEFAULT_STYLE, _TAB_BAR_STYLES[TabBarVariant.SEGMENTED] = TAB_BAR_SEGMENTED_STYLE, _TAB_BAR_STYLES[TabBarVariant.NAVIGATION] = TAB_BAR_NAVIGATION_STYLE, _TAB_BAR_STYLES);
