// @revolut/ui-kit@11.21.2/esm/components/SelectBase/constants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

export var SelectSearchState;

(function (SelectSearchState) {
  SelectSearchState[SelectSearchState["IDLE"] = 0] = "IDLE";
  SelectSearchState[SelectSearchState["PENDING"] = 1] = "PENDING";
  SelectSearchState[SelectSearchState["FAILED"] = 2] = "FAILED";
})(SelectSearchState || (SelectSearchState = {}));

export var SelectBaseSize;

(function (SelectBaseSize) {
  SelectBaseSize[SelectBaseSize["SEARCH"] = 56] = "SEARCH";
  SelectBaseSize[SelectBaseSize["FOOTER"] = 68] = "FOOTER";
  SelectBaseSize[SelectBaseSize["OPTION"] = 48] = "OPTION";
  SelectBaseSize[SelectBaseSize["MIN_WIDTH"] = 240] = "MIN_WIDTH";
  SelectBaseSize[SelectBaseSize["MAX_WIDTH"] = 344] = "MAX_WIDTH";
  SelectBaseSize[SelectBaseSize["MAX_HEIGHT"] = 288] = "MAX_HEIGHT";
  SelectBaseSize[SelectBaseSize["MAX_HEIGHT_SEARCH"] = 344] = "MAX_HEIGHT_SEARCH";
  SelectBaseSize[SelectBaseSize["MAX_HEIGHT_FOOTER"] = 356] = "MAX_HEIGHT_FOOTER";
  SelectBaseSize[SelectBaseSize["MAX_HEIGHT_SEARCH_FOOTER"] = 412] = "MAX_HEIGHT_SEARCH_FOOTER";
})(SelectBaseSize || (SelectBaseSize = {}));
