// @revolut/ui-kit@11.21.2/esm/components/StorySwitcher/styles.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import css from '@styled-system/css';
import styled from 'styled-components';
import { Space } from "../../constants";
import { Modal } from "../Modal";
var DIALOG_STYLE = {
  position: 'relative',
  display: 'flex',
  flexDirection: 'column',
  flexGrow: 1,
  width: '100%',
  minWidth: 1,
  maxWidth: ['100%', null, '25rem'],
  maxHeight: ['100%', null, '42rem'],
  '@media (max-height: 400px)': {
    borderRadius: 'unset',
    maxWidth: '100%',
    maxHeight: '100%'
  }
};
var STYLE = {
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  paddingTop: [Space.NONE, null, Space.S_32],
  paddingBottom: [Space.NONE, null, Space.S_32],
  '[role=dialog]': DIALOG_STYLE,
  '@media (max-height: 400px)': {
    padding: Space.NONE
  }
};
export var StyledModal = /*#__PURE__*/styled(Modal).withConfig({
  displayName: "styles__StyledModal",
  componentId: "sc-15491gw-0"
})(css(STYLE));
