// @revolut/ui-kit@11.21.2/esm/components/HeaderBase/HeaderBaseTitle.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { forwardRef } from 'react';
import { useHeaderBaseContext } from "./HeaderBaseContext";
import { HeaderBaseVariant } from "./variants";
import { NAV_BAR_GRID_AREAS, NavBarItem } from "../NavBar";
import { Text } from "../Text";
import { TextStyle } from "../../constants";
/** @private */

function HeaderTitleText(_ref) {
  var variant = _ref.variant,
      rest = _objectWithoutPropertiesLoose(_ref, ["variant"]);

  switch (variant) {
    case HeaderBaseVariant.BOTTOM_SHEET:
      return /*#__PURE__*/React.createElement(Text, _extends({
        use: "div",
        variant: TextStyle.H5,
        lineClamp: 2
      }, rest));

    case HeaderBaseVariant.MODAL:
      return /*#__PURE__*/React.createElement(Text, _extends({
        use: "div",
        variant: TextStyle.PRIMARY,
        textAlign: "center",
        lineClamp: 2
      }, rest));

    case HeaderBaseVariant.MAIN:
    case HeaderBaseVariant.FORM:
    case HeaderBaseVariant.ITEM:
      return /*#__PURE__*/React.createElement(Text, _extends({
        use: "div",
        variant: TextStyle.H1,
        lineClamp: 3,
        textOverflow: "ellipsis"
      }, rest));
  }
}

export var HeaderBaseTitle = /*#__PURE__*/forwardRef(function (_ref2, ref) {
  var hide = _ref2.hide,
      rest = _objectWithoutPropertiesLoose(_ref2, ["hide"]);

  var variant = useHeaderBaseContext();
  return /*#__PURE__*/React.createElement(NavBarItem, {
    ref: ref,
    gridArea: NAV_BAR_GRID_AREAS.TITLE,
    justifySelf: variant === HeaderBaseVariant.MODAL ? 'center' : 'start',
    maxWidth: "100%",
    minWidth: 0,
    hide: hide
  }, /*#__PURE__*/React.createElement(HeaderTitleText, _extends({
    variant: variant
  }, rest)));
});
