// @revolut/ui-kit@11.21.2/esm/components/HeaderBase/HeaderBaseAvatar.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import React from 'react';
import { useHeaderBaseContext } from "./HeaderBaseContext";
import { HeaderBaseVariant } from "./variants";
import { NAV_BAR_GRID_AREAS, NavBarItem } from "../NavBar";
export var AVATAR_GRID_AREA = 'avatar';
/** @private */

export function HeaderBaseAvatar(_ref) {
  var children = _ref.children,
      hide = _ref.hide;
  var variant = useHeaderBaseContext();

  if (variant !== HeaderBaseVariant.ITEM && variant !== HeaderBaseVariant.MAIN) {
    return null;
  }

  return /*#__PURE__*/React.createElement(NavBarItem, {
    gridArea: variant === HeaderBaseVariant.ITEM ? AVATAR_GRID_AREA : NAV_BAR_GRID_AREAS.TITLE,
    maxWidth: "100%",
    justifySelf: variant === HeaderBaseVariant.ITEM ? 'end' : 'start',
    alignSelf: variant === HeaderBaseVariant.ITEM ? 'start' : 'center',
    hide: hide
  }, children);
}
