// @revolut/ui-kit@11.21.2/esm/components/HeaderBase/HeaderBaseDescription.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { Color, Space, TextStyle } from "../../constants";
import { useHeaderBaseContext } from "./HeaderBaseContext";
import { HeaderBaseVariant } from "./variants";
import { NavBarItem } from "../NavBar";
import { Text } from "../Text";
export var DESCRIPTION_GRID_AREA = 'description';
/** @private */

export function HeaderBaseDescription(_ref) {
  var _ref$color = _ref.color,
      color = _ref$color === void 0 ? Color.GREY_TONE_50 : _ref$color,
      rest = _objectWithoutPropertiesLoose(_ref, ["color"]);

  var variant = useHeaderBaseContext();
  return /*#__PURE__*/React.createElement(NavBarItem, {
    gridArea: DESCRIPTION_GRID_AREA,
    color: color
  }, /*#__PURE__*/React.createElement(Text, _extends({
    mt: variant === HeaderBaseVariant.MODAL ? undefined : Space.S_4,
    variant: TextStyle.CAPTION,
    maxWidth: "100%"
  }, rest)));
}
