// @revolut/ui-kit@11.21.2/esm/components/HeaderBase/HeaderBase.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { forwardRef } from 'react';
import { Space } from "../../constants";
import { NAV_BAR_GRID_AREAS, NAV_BAR_HEIGHT, NavBar, NavBarWrapper } from "../NavBar";
import { Box } from "../Box";
import { Flex } from "../Flex";
import { HeaderBaseVariant } from "./variants";
import { HeaderBaseContext, useHeaderBaseContext } from "./HeaderBaseContext";
import { BAR_GRID_AREA, HeaderBaseBar } from "./HeaderBaseBar";
import { AVATAR_GRID_AREA, HeaderBaseAvatar } from "./HeaderBaseAvatar";
import { HeaderBaseTitle } from "./HeaderBaseTitle";
import { HeaderBaseSubtitle, SUBTITLE_GRID_AREA } from "./HeaderBaseSubtitle";
import { HeaderBaseActionBar } from "./HeaderBaseActionBar";
import { DESCRIPTION_GRID_AREA, HeaderBaseDescription } from "./HeaderBaseDescription";
var BOTTOM_SHEET_GRID_TEMPLATE = "\n  \"" + NAV_BAR_GRID_AREAS.TITLE + " " + NAV_BAR_GRID_AREAS.ACTIONS + "\"\n  \"" + SUBTITLE_GRID_AREA + " " + SUBTITLE_GRID_AREA + "\"\n  \"" + DESCRIPTION_GRID_AREA + " " + DESCRIPTION_GRID_AREA + "\"\n";
var FORM_GRID_TEMPLATE = "\n  \"" + NAV_BAR_GRID_AREAS.NAV_BUTTON + " " + NAV_BAR_GRID_AREAS.ACTIONS + "\"\n  \"" + NAV_BAR_GRID_AREAS.TITLE + " " + NAV_BAR_GRID_AREAS.TITLE + "\"\n  \"" + SUBTITLE_GRID_AREA + " " + SUBTITLE_GRID_AREA + "\"\n  \"" + DESCRIPTION_GRID_AREA + " " + DESCRIPTION_GRID_AREA + "\"\n  \"" + BAR_GRID_AREA + " " + BAR_GRID_AREA + "\"\n";
var ITEM_GRID_TEMPLATE = "\n  \"" + NAV_BAR_GRID_AREAS.NAV_BUTTON + " " + NAV_BAR_GRID_AREAS.ACTIONS + "\"\n  \"" + NAV_BAR_GRID_AREAS.TITLE + " " + AVATAR_GRID_AREA + "\"\n  \"" + SUBTITLE_GRID_AREA + " " + AVATAR_GRID_AREA + "\"\n  \"" + DESCRIPTION_GRID_AREA + " " + AVATAR_GRID_AREA + "\"\n  \"" + BAR_GRID_AREA + " " + BAR_GRID_AREA + "\"\n";
var MAIN_GRID_TEMPLATE = "\n  \"" + NAV_BAR_GRID_AREAS.TITLE + " " + NAV_BAR_GRID_AREAS.ACTIONS + "\"\n  \"" + SUBTITLE_GRID_AREA + " " + AVATAR_GRID_AREA + "\"\n";

/** @private  */
export var HeaderBaseNavBar = function HeaderBaseNavBar(_ref) {
  var children = _ref.children;
  var variant = useHeaderBaseContext();

  switch (variant) {
    case HeaderBaseVariant.BOTTOM_SHEET:
      return /*#__PURE__*/React.createElement(NavBarWrapper, {
        gridTemplateAreas: BOTTOM_SHEET_GRID_TEMPLATE,
        gridTemplateColumns: "minmax(0, auto) 1fr",
        gridTemplateRows: "repeat(2, auto)",
        height: "unset"
      }, children);

    case HeaderBaseVariant.FORM:
      return /*#__PURE__*/React.createElement(NavBarWrapper, {
        gridTemplateAreas: FORM_GRID_TEMPLATE,
        gridTemplateRows: "repeat(5, auto)",
        gridTemplateColumns: "1fr minmax(0, auto)",
        height: "unset"
      }, children);

    case HeaderBaseVariant.ITEM:
      return /*#__PURE__*/React.createElement(NavBarWrapper, {
        gridTemplateAreas: ITEM_GRID_TEMPLATE,
        gridTemplateRows: "repeat(5, auto)",
        gridTemplateColumns: "1fr minmax(0, auto)",
        height: "unset"
      }, children);

    case HeaderBaseVariant.MAIN:
      return /*#__PURE__*/React.createElement(NavBarWrapper, {
        gridTemplateAreas: MAIN_GRID_TEMPLATE,
        gridTemplateRows: "minmax(" + NAV_BAR_HEIGHT + ", auto) auto",
        height: "unset"
      }, children);

    case HeaderBaseVariant.MODAL:
      return /*#__PURE__*/React.createElement(NavBar, null, children);
  }
};

var NavigationButtonWrapper = function NavigationButtonWrapper(_ref2) {
  var children = _ref2.children;
  var variant = useHeaderBaseContext();

  if (variant === HeaderBaseVariant.BOTTOM_SHEET || variant === HeaderBaseVariant.MAIN) {
    return null;
  }

  if (variant === HeaderBaseVariant.ITEM || variant === HeaderBaseVariant.FORM) {
    return /*#__PURE__*/React.createElement(Flex, {
      height: NAV_BAR_HEIGHT,
      alignItems: "center"
    }, children);
  }

  return /*#__PURE__*/React.createElement(React.Fragment, null, children);
};
/** @private */


var HeaderBaseCloseButton = /*#__PURE__*/forwardRef(function (props, ref) {
  return /*#__PURE__*/React.createElement(NavigationButtonWrapper, null, /*#__PURE__*/React.createElement(NavBar.CloseButton, _extends({
    ref: ref
  }, props)));
});
/** @private */

var HeaderBaseBackButton = /*#__PURE__*/forwardRef(function (props, ref) {
  return /*#__PURE__*/React.createElement(NavigationButtonWrapper, null, /*#__PURE__*/React.createElement(NavBar.BackButton, _extends({
    ref: ref
  }, props)));
});
/** @private */

var HeaderBaseActions = function HeaderBaseActions(props) {
  var variant = useHeaderBaseContext();

  if (variant === HeaderBaseVariant.ITEM || variant === HeaderBaseVariant.FORM) {
    return /*#__PURE__*/React.createElement(Flex, {
      height: NAV_BAR_HEIGHT,
      justifyContent: "flex-end"
    }, /*#__PURE__*/React.createElement(NavBar.Actions, props));
  }

  return /*#__PURE__*/React.createElement(NavBar.Actions, props);
};

var getPaddingBottom = function getPaddingBottom(variant) {
  switch (variant) {
    case HeaderBaseVariant.MODAL:
      return Space.S_6;

    case HeaderBaseVariant.BOTTOM_SHEET:
      return Space.S_16;

    case HeaderBaseVariant.MAIN:
      return Space.S_16;

    default:
      return Space.S_24;
  }
};

var getPaddingTop = function getPaddingTop(variant) {
  switch (variant) {
    case HeaderBaseVariant.MODAL:
      return Space.S_6;

    case HeaderBaseVariant.BOTTOM_SHEET:
      return Space.S_16;

    case HeaderBaseVariant.MAIN:
      return Space.S_16;

    default:
      return Space.NONE;
  }
};
/** @private */


/** @private */
var RefForwardingHeaderBase = /*#__PURE__*/forwardRef(function (_ref3, ref) {
  var children = _ref3.children,
      variant = _ref3.variant;
  return /*#__PURE__*/React.createElement(HeaderBaseContext.Provider, {
    value: variant
  }, /*#__PURE__*/React.createElement(Box, {
    ref: ref,
    pb: getPaddingBottom(variant),
    pt: getPaddingTop(variant)
  }, /*#__PURE__*/React.createElement(HeaderBaseNavBar, null, children)));
});
export var HeaderBase = Object.assign(RefForwardingHeaderBase, {
  CloseButton: HeaderBaseCloseButton,
  BackButton: HeaderBaseBackButton,
  Actions: HeaderBaseActions,
  Title: HeaderBaseTitle,
  Subtitle: HeaderBaseSubtitle,
  Description: HeaderBaseDescription,
  Avatar: HeaderBaseAvatar,
  Bar: HeaderBaseBar,

  /** @deprecated use `HeaderBase.Bar` */
  ActionBar: HeaderBaseActionBar
});
