// @revolut/ui-kit@11.21.2/esm/components/ButtonBase/ButtonBase.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled from 'styled-components';
import { alignItems, display, justifyContent } from 'styled-system';
import { Color, Radius, Transition } from "../../constants";
import { themeColor, themeRadius, themeTransition } from "../../utils";
import { hide } from "../../styles";
import { createBase } from "../Base";
export var DEFAULT_ELEMENT = 'button';

var attributes = function attributes(_ref) {
  var _ref$use = _ref.use,
      use = _ref$use === void 0 ? DEFAULT_ELEMENT : _ref$use,
      type = _ref.type,
      disabled = _ref.disabled,
      tabIndex = _ref.tabIndex,
      role = _ref.role;

  if (use === DEFAULT_ELEMENT) {
    return {
      type: type || 'button',
      tabIndex: disabled ? -1 : tabIndex
    };
  }

  return {
    role: role,
    'aria-disabled': disabled
  };
};

var Base = createBase(DEFAULT_ELEMENT);
export var ButtonBase = /*#__PURE__*/styled(Base).attrs(attributes).withConfig({
  displayName: "ButtonBase",
  componentId: "sc-188b3i5-0"
})(["--button-shadow-focus:0 0 0 2px ", ",0 0 0 4px ", ";box-sizing:border-box;border:none;font:inherit;background:transparent;color:inherit;text-align:inherit;text-decoration:none;padding:0;outline:none;border-radius:", ";transition:", ";:focus-visible{box-shadow:var(--button-shadow-focus);}&.focus-visible{box-shadow:var(--button-shadow-focus);}:enabled{cursor:pointer;}:link{text-decoration:none;}&:disabled,&[aria-disabled='true']{pointer-events:none;}", " ", " ", " ", ""], themeColor(Color.BACKGROUND), themeColor(Color.BLUE_OPAQUE_50), themeRadius(Radius.BUTTON_XS), themeTransition(Transition.SHADOW), display, alignItems, justifyContent, hide);
