// @revolut/ui-kit@11.21.2/esm/components/Spacer/Spacer.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled from 'styled-components';
import { system } from '@styled-system/core';

/** @private */
export var spacer = system({
  size: {
    properties: ['width', 'height'],
    scale: 'space'
  },
  minSize: {
    property: 'flexBasis',
    scale: 'space'
  }
});
export var Spacer = /*#__PURE__*/styled.div.withConfig({
  displayName: "Spacer",
  componentId: "e2cz3-0"
})(["flex:", ";margin:0 !important;", ""], function (props) {
  return props.size != null ? '0 1 auto' : '1 1 0%';
}, spacer);
