// @revolut/ui-kit@11.21.2/esm/components/AppsWidget/AppsWidget.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { forwardRef } from 'react';
import { Grid } from "../../future/components/Grid";
import { Space } from "../../constants";
import { Widget } from "../Widget";
import { AppIconSize } from "../AppIcon";
export var AppsWidget = /*#__PURE__*/forwardRef(function (_ref, ref) {
  var _ref$size = _ref.size,
      size = _ref$size === void 0 ? AppIconSize.SIZE_76 : _ref$size,
      rest = _objectWithoutPropertiesLoose(_ref, ["size"]);

  return /*#__PURE__*/React.createElement(Widget, {
    p: Space.S_16
  }, /*#__PURE__*/React.createElement(Grid, _extends({
    ref: ref,
    rows: "auto",
    autoRows: 0,
    columnGap: Space.S_16,
    columns: "repeat(auto-fit, " + size + "px)",
    placeContent: "space-between",
    overflow: "hidden"
  }, rest)));
});
