// @revolut/ui-kit@11.21.2/esm/legacy/components/Input/ClearButton.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import styled from 'styled-components';
import * as Icons from '@revolut/icons';
import { Color } from "../../../constants";
import { ButtonBase } from "../../../components/ButtonBase";

var ClearButtonBase = function ClearButtonBase(props) {
  return /*#__PURE__*/React.createElement(ButtonBase, _extends({
    tabIndex: -1
  }, props), /*#__PURE__*/React.createElement(Icons.CrossCircle, {
    size: 24,
    color: Color.GREY_TONE_20
  }));
};

export var ClearButton = /*#__PURE__*/styled(ClearButtonBase).withConfig({
  displayName: "ClearButton",
  componentId: "sc-5gdwlo-0"
})([""]);
