// @revolut/ui-kit@11.21.2/esm/legacy/components/Input/constants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

export var AppearanceTransitionDirection;

(function (AppearanceTransitionDirection) {
  AppearanceTransitionDirection["RTL"] = "right center";
  AppearanceTransitionDirection["LTR"] = "left center";
})(AppearanceTransitionDirection || (AppearanceTransitionDirection = {}));

export var OutgoingTransitionDirection;

(function (OutgoingTransitionDirection) {
  OutgoingTransitionDirection["RTL"] = "center left";
  OutgoingTransitionDirection["LTR"] = "center right";
})(OutgoingTransitionDirection || (OutgoingTransitionDirection = {}));

export var InputVariant = {
  /** @deprecated use `grey` variant */
  UNDERLINED: 'underlined',

  /** @deprecated use `grey` variant */
  FILLED: 'filled',
  GREY: 'grey'
};
export var InputSize = {
  DEFAULT: 'default',
  COMPACT: 'compact',
  PIN: 'pin'
};
