// @revolut/ui-kit@11.21.2/esm/legacy/components/Input/StatusBorder.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled, { css } from 'styled-components';
import { px, themeColor, themeTransition } from "../../../utils";
import { Color, Transition } from "../../../constants";
import { getStatusColor } from "./utils";
import { AppearanceTransitionDirection, OutgoingTransitionDirection } from "./constants";
export var INPUT_STATUS_BORDER_SIZE = px(1);
export var INPUT_STATUS_BORDER_MARGIN_TOP = px(1);
export var StatusBorder = /*#__PURE__*/styled('div').withConfig({
  displayName: "StatusBorder",
  componentId: "sc-8c72pc-0"
})(["width:100%;margin-top:", ";height:", ";transform-origin:", ";background-color:", ";transition:", ";transform:scaleX(0);", ""], INPUT_STATUS_BORDER_MARGIN_TOP, INPUT_STATUS_BORDER_SIZE, function (props) {
  return props.outgoingDirection;
}, function (props) {
  return themeColor(getStatusColor(props, Color.PRIMARY));
}, themeTransition(Transition.TRANSFORM), function (props) {
  return !props.isDisabled && (props.isFocused || props.hasError) && css(["transform:scaleX(1);transform-origin:", ";"], props.appearanceDirection);
});
StatusBorder.defaultProps = {
  appearanceDirection: AppearanceTransitionDirection.LTR,
  outgoingDirection: OutgoingTransitionDirection.RTL
};
