// @revolut/ui-kit@11.21.2/esm/components/BarChart/constants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _GRAPH_HEIGHT, _GRAPH_MARGIN;

import { BarChartVariant } from "./variants";
export var GRAPH_HEIGHT = (_GRAPH_HEIGHT = {}, _GRAPH_HEIGHT[BarChartVariant.LARGE] = 134, _GRAPH_HEIGHT[BarChartVariant.SMALL] = 68, _GRAPH_HEIGHT);
export var GRAPH_MARGIN = (_GRAPH_MARGIN = {}, _GRAPH_MARGIN[BarChartVariant.LARGE] = {
  top: 10,
  right: 49,
  bottom: 28,
  left: 4
}, _GRAPH_MARGIN[BarChartVariant.SMALL] = {
  top: 16,
  right: 0,
  bottom: 0,
  left: 0
}, _GRAPH_MARGIN);
export var BAR_WIDTH = {
  DEFAULT: 16,
  SM: 8,
  XS: 4
};
