// @revolut/ui-kit@11.21.2/esm/components/LimitsWidget/LimitsWidget.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled from 'styled-components';
import { css } from '@styled-system/css';
import { Space } from "../../constants";
import { Widget } from "../Widget";
import { Box } from "../Box";
export var LimitsWidget = Object.assign(styled(Widget).withConfig({
  displayName: "LimitsWidget",
  componentId: "sc-1po84up-0"
})(css({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center'
})), {
  Header: styled(Box).withConfig({
    displayName: "LimitsWidget__Header",
    componentId: "sc-1po84up-1"
  })(css({
    width: '100%'
  }))
}, {
  Chart: styled(Box).withConfig({
    displayName: "LimitsWidget__Chart",
    componentId: "sc-1po84up-2"
  })(css({
    p: Space.S_16
  }))
}, {
  Footer: styled(Box).withConfig({
    displayName: "LimitsWidget__Footer",
    componentId: "sc-1po84up-3"
  })(css({
    pb: Space.S_16,
    px: Space.S_16
  }))
});
