// @revolut/ui-kit@11.21.2/esm/tokens/unified/text-styles.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _TEXT_STYLES;

import { Font, FontSize, LetterSpace, LineHeight, TextStyle } from "../../constants";
export var TEXT_STYLES = (_TEXT_STYLES = {}, _TEXT_STYLES[TextStyle.DEFAULT] = null, _TEXT_STYLES[TextStyle.H0] = {
  fontFamily: Font.BRAND,
  fontWeight: 600,
  fontSize: FontSize.H0,
  lineHeight: LineHeight.H0,
  letterSpacing: LetterSpace.H0
}, _TEXT_STYLES[TextStyle.H1] = {
  fontFamily: Font.BRAND,
  fontWeight: 600,
  fontSize: FontSize.H1,
  lineHeight: LineHeight.H1,
  letterSpacing: LetterSpace.H1
}, _TEXT_STYLES[TextStyle.H2] = {
  fontFamily: Font.BRAND,
  fontWeight: 500,
  fontSize: FontSize.H2,
  lineHeight: LineHeight.H2,
  letterSpacing: LetterSpace.H2
}, _TEXT_STYLES[TextStyle.H3] = {
  fontFamily: Font.SYSTEM,
  fontWeight: 400,
  fontSize: FontSize.H3,
  lineHeight: LineHeight.H3,
  letterSpacing: LetterSpace.H3
}, _TEXT_STYLES[TextStyle.H4] = {
  fontFamily: Font.SYSTEM,
  fontWeight: 500,
  fontSize: FontSize.H4,
  lineHeight: LineHeight.H4,
  letterSpacing: LetterSpace.H4
}, _TEXT_STYLES[TextStyle.H5] = {
  fontFamily: Font.BRAND,
  fontWeight: 600,
  fontSize: FontSize.H5,
  lineHeight: LineHeight.H5,
  letterSpacing: LetterSpace.H5,
  textTransform: 'none'
}, _TEXT_STYLES[TextStyle.H6] = {
  fontFamily: Font.SYSTEM,
  fontWeight: 500,
  fontSize: FontSize.H6,
  lineHeight: LineHeight.H6,
  letterSpacing: LetterSpace.H6
}, _TEXT_STYLES[TextStyle.PRIMARY] = {
  fontFamily: Font.SYSTEM,
  fontWeight: 500,
  fontSize: FontSize.PRIMARY,
  lineHeight: LineHeight.PRIMARY,
  letterSpacing: LetterSpace.PRIMARY
}, _TEXT_STYLES[TextStyle.SECONDARY] = {
  fontFamily: Font.SYSTEM,
  fontWeight: 400,
  fontSize: FontSize.SECONDARY,
  lineHeight: LineHeight.SECONDARY,
  letterSpacing: LetterSpace.SECONDARY
}, _TEXT_STYLES[TextStyle.BULLET] = {
  fontFamily: Font.BRAND,
  fontWeight: 600,
  fontSize: FontSize.BULLET,
  lineHeight: LineHeight.BULLET,
  letterSpacing: LetterSpace.BULLET
}, _TEXT_STYLES[TextStyle.CAPTION] = {
  fontFamily: Font.SYSTEM,
  fontWeight: 400,
  fontSize: FontSize.CAPTION,
  lineHeight: LineHeight.CAPTION,
  letterSpacing: LetterSpace.CAPTION
}, _TEXT_STYLES[TextStyle.SMALL] = {
  fontFamily: Font.SYSTEM,
  fontWeight: 400,
  fontSize: FontSize.SMALL,
  lineHeight: LineHeight.SMALL,
  letterSpacing: LetterSpace.SMALL
}, _TEXT_STYLES[TextStyle.TINY] = {
  fontFamily: Font.SYSTEM,
  fontWeight: 600,
  fontSize: FontSize.TINY,
  lineHeight: LineHeight.TINY,
  letterSpacing: LetterSpace.TINY
}, _TEXT_STYLES[TextStyle.TILE] = {
  variant: "textStyles." + TextStyle.PRIMARY,
  fontFamily: Font.BRAND
}, _TEXT_STYLES[TextStyle.BUTTON] = null, _TEXT_STYLES[TextStyle.TEXT_BUTTON] = null, _TEXT_STYLES[TextStyle.ACTION] = {
  variant: "textStyles." + TextStyle.H6,
  fontWeight: null
}, _TEXT_STYLES[TextStyle.INPUT] = {
  variant: "textStyles." + TextStyle.SECONDARY
}, _TEXT_STYLES);
