// @revolut/ui-kit@11.21.2/esm/components/Layout/constants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _MD, _LG, _XL, _XXL, _MD2, _LG2, _XL2, _XXL2, _MD3, _LG3, _SideSize$MEDIUM, _SideSize$WIDE, _XL3, _SideSize$MEDIUM2, _SideSize$WIDE2, _XXL3, _SIDE;

import { rem } from 'polished';
export var ContainerSize = {
  MEDIUM: 'medium',
  WIDE: 'wide'
};
export var MainSize = {
  NARROW: 'narrow',
  MEDIUM: 'medium',
  WIDE: 'wide'
};
export var SideSize = {
  MEDIUM: 'medium',
  WIDE: 'wide'
};
export var Area = {
  MENU: 'menu',
  MAIN: 'main',
  SIDE: 'side',
  ACTS: 'acts',
  TOAST: 'toast'
};
export var SlotName = {
  MENU: 'layout-menu',
  SIDE: 'layout-side',
  ACTS: 'layout-acts'
};
export var Z_INDICES = {
  CONTAINER: 0,
  ACTS: 2,
  MENU: 3,
  TOAST: 4,
  SIDE: 5
};
export var MENU_MEDIUM_SIZES = {
  MD: 88,
  LG: 104,
  XL: 248,
  XXL: 400
};
export var MEDIUM_SIZES = {
  MENU: {
    MD: (_MD = {}, _MD[SideSize.MEDIUM] = rem(MENU_MEDIUM_SIZES.MD), _MD[SideSize.WIDE] = 0, _MD),
    LG: (_LG = {}, _LG[SideSize.MEDIUM] = rem(MENU_MEDIUM_SIZES.LG), _LG[SideSize.WIDE] = 0, _LG),
    XL: (_XL = {}, _XL[SideSize.MEDIUM] = rem(MENU_MEDIUM_SIZES.XL), _XL[SideSize.WIDE] = 0, _XL),
    XXL: (_XXL = {}, _XXL[SideSize.MEDIUM] = rem(MENU_MEDIUM_SIZES.XXL), _XXL[SideSize.WIDE] = 0, _XXL)
  },
  MAIN: {
    SM: rem(500),
    MD: rem(500),
    LG: rem(500),
    XL: rem(536)
  },
  SIDE: {
    MD: (_MD2 = {}, _MD2[SideSize.MEDIUM] = rem(MENU_MEDIUM_SIZES.MD), _MD2[SideSize.WIDE] = 0, _MD2),
    LG: (_LG2 = {}, _LG2[SideSize.MEDIUM] = rem(MENU_MEDIUM_SIZES.LG), _LG2[SideSize.WIDE] = rem(432), _LG2),
    XL: (_XL2 = {}, _XL2[SideSize.MEDIUM] = rem(MENU_MEDIUM_SIZES.XL), _XL2[SideSize.WIDE] = rem(680), _XL2),
    XXL: (_XXL2 = {}, _XXL2[SideSize.MEDIUM] = rem(MENU_MEDIUM_SIZES.XXL), _XXL2[SideSize.WIDE] = rem(1000), _XXL2)
  }
};
export var WIDE_SIZES = {
  NAV: {
    MD: 'max-content',
    LG: 'max-content',
    XL: 'max-content',
    XXL: rem(248)
  },
  NAV_GAP: {
    XL: "minmax(auto, " + rem(8) + ")",
    XXL: "minmax(auto, " + rem(120) + ")"
  },
  MAIN: {
    MD: (_MD3 = {}, _MD3[MainSize.NARROW] = "minmax(auto, " + rem(720) + ") 1fr", _MD3[MainSize.MEDIUM] = "1fr 0", _MD3[MainSize.WIDE] = "1fr 0", _MD3),
    LG: (_LG3 = {}, _LG3[MainSize.NARROW] = "minmax(auto, " + rem(720) + ")", _LG3[MainSize.MEDIUM] = "minmax(auto, " + rem(1088) + ")", _LG3[MainSize.WIDE] = "minmax(auto, " + rem(1272) + ")", _LG3),
    XL: (_XL3 = {}, _XL3[SideSize.MEDIUM] = (_SideSize$MEDIUM = {}, _SideSize$MEDIUM[MainSize.NARROW] = rem(720), _SideSize$MEDIUM[MainSize.MEDIUM] = "minmax(" + rem(640) + ", " + rem(1088) + ")", _SideSize$MEDIUM[MainSize.WIDE] = "minmax(" + rem(640) + ", " + rem(1456) + ")", _SideSize$MEDIUM), _XL3[SideSize.WIDE] = (_SideSize$WIDE = {}, _SideSize$WIDE[MainSize.NARROW] = "minmax(" + rem(500) + ", " + rem(720) + ")", _SideSize$WIDE[MainSize.MEDIUM] = "minmax(" + rem(500) + ", " + rem(1088) + ")", _SideSize$WIDE[MainSize.WIDE] = "minmax(" + rem(500) + ", " + rem(1456) + ")", _SideSize$WIDE), _XL3),
    XXL: (_XXL3 = {}, _XXL3[SideSize.MEDIUM] = (_SideSize$MEDIUM2 = {}, _SideSize$MEDIUM2[MainSize.NARROW] = rem(720), _SideSize$MEDIUM2[MainSize.MEDIUM] = rem(1088), _SideSize$MEDIUM2[MainSize.WIDE] = "minmax(" + rem(1248) + ", " + rem(1456) + ")", _SideSize$MEDIUM2), _XXL3[SideSize.WIDE] = (_SideSize$WIDE2 = {}, _SideSize$WIDE2[MainSize.NARROW] = rem(720), _SideSize$WIDE2[MainSize.MEDIUM] = rem(1088), _SideSize$WIDE2[MainSize.WIDE] = "minmax(" + rem(1056) + ", " + rem(1456) + ")", _SideSize$WIDE2), _XXL3)
  },
  SIDE: (_SIDE = {}, _SIDE[SideSize.MEDIUM] = rem(400), _SIDE[SideSize.WIDE] = rem(584), _SIDE)
};
