// @revolut/ui-kit@11.21.2/esm/components/Layout/LayoutMain.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _css;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import styled from 'styled-components';
import { css } from '@styled-system/css';
import { Space } from "../../constants";
import { StickyHeader } from "../Header/StickyHeader";
import { Box } from "../Box";
import { Area } from "./constants";
export var LayoutMainBase = /*#__PURE__*/styled(Box).withConfig({
  displayName: "LayoutMain__LayoutMainBase",
  componentId: "sc-17izcsp-0"
})(css((_css = {
  display: 'flex',
  flexDirection: 'column',
  gridArea: Area.MAIN,
  minWidth: 0,
  px: [Space.S_16, null, Space.NONE],
  py: [Space.S_8, null, Space.S_16]
}, _css[StickyHeader] = {
  order: -1
}, _css)));
export function LayoutMain(props) {
  return /*#__PURE__*/React.createElement(LayoutMainBase, _extends({
    use: "main"
  }, props));
}
