// @revolut/ui-kit@11.21.2/esm/components/Tooltip/Tooltip.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { Color, Radius, Space, TextStyle } from "../../constants";
import { Hint } from "../../legacy/components/Hint";
import { TransitionFade } from "../Transitions";
import { Text } from "../Text";
/** @private */

export var TooltipBox = function TooltipBox(props) {
  return /*#__PURE__*/React.createElement(Text, _extends({
    use: "div",
    bg: Color.FOREGROUND,
    color: Color.BACKGROUND,
    radius: Radius.TOOLTIP,
    m: Space.S_8,
    px: Space.S_8,
    py: Space.S_4,
    variant: TextStyle.SMALL
  }, props));
};
export var Tooltip = function Tooltip(props) {
  return /*#__PURE__*/React.createElement(Hint, _extends({
    useWrapper: TooltipBox,
    useTransition: TransitionFade
  }, props));
};
