// @revolut/ui-kit@11.21.2/esm/components/Media/Media.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { Box } from "../Box";
import { Flex } from "../Flex";
export var Media = function Media(props) {
  return /*#__PURE__*/React.createElement(Flex, _extends({
    flex: "1 1 auto",
    minWidth: 0,
    maxWidth: "100%"
  }, props));
};
Media.displayName = 'Media';
export var MediaSide = function MediaSide(props) {
  return /*#__PURE__*/React.createElement(Box, _extends({
    flex: "0 0 auto"
  }, props));
};
Media.Side = MediaSide;
MediaSide.displayName = 'Media.Side';
export var MediaContent = function MediaContent(props) {
  return /*#__PURE__*/React.createElement(Box, _extends({
    flex: "1 1 auto",
    minWidth: 0
  }, props));
};
Media.Content = MediaContent;
MediaContent.displayName = 'Media.Content';
