// @revolut/ui-kit@11.21.2/esm/components/Dots/Dots.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import styled, { keyframes } from 'styled-components';
import { Easing } from "../../constants";
import { EASINGS } from "../../tokens/default";
import { Box } from "../Box";
import { prop } from "../../styles";
var DEFAULT_DURATION = 900;
var move = /*#__PURE__*/keyframes(["0%,100%{transform:translate(0,0);}35%{transform:translate(0,8px);}"]);
var DotCircle = /*#__PURE__*/styled.circle.withConfig({
  displayName: "Dots__DotCircle",
  componentId: "lprp81-0"
})(["animation-name:", ";animation-duration:", ";animation-delay:", ";animation-iteration-count:infinite;animation-timing-function:cubic-bezier(", ");"], move, prop('duration', DEFAULT_DURATION, 'ms'), prop('delay', 0, 'ms'), EASINGS[Easing.DEFAULT]);
export var Dots = function Dots(_ref) {
  var _ref$size = _ref.size,
      size = _ref$size === void 0 ? 32 : _ref$size,
      _ref$circle = _ref.circle,
      circle = _ref$circle === void 0 ? 3 : _ref$circle,
      color = _ref.color,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? DEFAULT_DURATION : _ref$duration,
      rest = _objectWithoutPropertiesLoose(_ref, ["size", "circle", "color", "duration"]);

  return /*#__PURE__*/React.createElement(Box, _extends({
    use: "svg",
    display: "block",
    viewBox: "0 0 32 32",
    color: color,
    size: size
  }, rest), /*#__PURE__*/React.createElement("g", {
    fill: "currentColor"
  }, /*#__PURE__*/React.createElement(DotCircle, {
    duration: duration,
    delay: duration * 0.05,
    cx: "4",
    cy: "16",
    r: circle
  }), /*#__PURE__*/React.createElement(DotCircle, {
    duration: duration,
    delay: duration * 0.15,
    cx: "16",
    cy: "16",
    r: circle
  }), /*#__PURE__*/React.createElement(DotCircle, {
    duration: duration,
    delay: duration * 0.25,
    cx: "28",
    cy: "16",
    r: circle
  })));
};
