// @revolut/ui-kit@11.21.2/esm/components/Skeleton/Skeleton.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled, { keyframes } from 'styled-components';
import { Color, Easing, Radius, Space } from "../../constants";
import { EASINGS } from "../../tokens/default";
import { prop } from "../../styles";
import { isString, rgbaSafe, themeColor } from "../../utils";
import { Flex } from "../Flex";
var DEFAULT_DURATION = 1500;
var moveGradient = /*#__PURE__*/keyframes(["50%,100%{transform:translateX(100%);}"]);

function shimmerColor(props) {
  var _rgbaSafe;

  if (isString(props.shimmer)) {
    return themeColor(props.shimmer, props.shimmer)(props);
  }

  return (_rgbaSafe = rgbaSafe(themeColor(Color.GREY_TONE_5)(props), 0.5)) != null ? _rgbaSafe : 'rgba(255, 255, 255, 0)';
}

export var Skeleton = /*#__PURE__*/styled(Flex).withConfig({
  displayName: "Skeleton",
  componentId: "sc-1fnkrx-0"
})(["overflow:hidden;mask-image:radial-gradient(white,black);&::after{content:'';display:block;width:100%;height:100%;background-image:linear-gradient( 90deg,rgba(255,245,255,0),", ",rgba(255,255,255,0) );transform:translateX(-100%);animation-name:", ";animation-duration:", ";animation-delay:", ";animation-iteration-count:infinite;animation-timing-function:cubic-bezier(", ");}"], shimmerColor, moveGradient, prop('duration', 0, 'ms'), prop('delay', 0, 'ms'), EASINGS[Easing.DEFAULT]);
Skeleton.defaultProps = {
  height: Space.S_8,
  duration: DEFAULT_DURATION,
  radius: Radius.ROUND,
  bg: Color.GREY_TONE_10
};
