// @revolut/ui-kit@11.21.2/esm/legacy/components/Loader/Loader.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import styled, { keyframes } from 'styled-components';
import * as Icons from '@revolut/icons';
import { prop } from "../../../styles";
import { Circle } from "../../../components/Circle";
import { Spinner } from "../../../components/Spinner";
var DEFAULT_DURATION = 1700;
var AbsoluteSpinner = /*#__PURE__*/styled(Spinner).withConfig({
  displayName: "Loader__AbsoluteSpinner",
  componentId: "sc-1i2zcr-0"
})(["position:absolute;opacity:0.4;"]);
var pulse = /*#__PURE__*/keyframes(["0%,100%{opacity:1;}50%{opacity:0.4;}"]);
var Pulse = /*#__PURE__*/styled.span.withConfig({
  displayName: "Loader__Pulse",
  componentId: "sc-1i2zcr-1"
})(["animation-name:", ";animation-duration:", ";animation-timing-function:linear;animation-iteration-count:infinite;"], pulse, prop('duration', DEFAULT_DURATION, 'ms'));
export var Loader = function Loader(_ref) {
  var _ref$size = _ref.size,
      size = _ref$size === void 0 ? 120 : _ref$size,
      _ref$strokeWidth = _ref.strokeWidth,
      strokeWidth = _ref$strokeWidth === void 0 ? 2 : _ref$strokeWidth,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'grey-tone-20' : _ref$color,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? DEFAULT_DURATION : _ref$duration,
      _ref$children = _ref.children,
      children = _ref$children === void 0 ? /*#__PURE__*/React.createElement(Icons.Revolut, {
    size: 64
  }) : _ref$children,
      rest = _objectWithoutPropertiesLoose(_ref, ["size", "strokeWidth", "color", "duration", "children"]);

  return /*#__PURE__*/React.createElement(Circle, _extends({
    size: size,
    color: color
  }, rest), /*#__PURE__*/React.createElement(AbsoluteSpinner, {
    size: size,
    strokeWidth: strokeWidth,
    duration: duration
  }), /*#__PURE__*/React.createElement(Pulse, {
    duration: duration
  }, children));
};
