// @revolut/ui-kit@11.21.2/esm/components/Sticker/Sticker.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import React from 'react';
import css from '@styled-system/css';
import styled from 'styled-components';
import { useStorySwitcherContext } from "../StorySwitcher/StorySwitcherContext";
import { Box } from "../Box";
import { Flex } from "../Flex";
import { Widget } from "../Widget";
import { Color, Space, TextStyle } from "../../constants";
import { themeTextStyle } from "../../utils";
import { TransitionSticker } from "./TransitionSticker";
var Wrapper = /*#__PURE__*/styled(Widget).withConfig({
  displayName: "Sticker__Wrapper",
  componentId: "sc-1a9u1l7-0"
})(css({
  width: 'fit-content',
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  py: Space.S_16,
  px: Space.S_24
}));
export var Sticker = function Sticker(props) {
  var _useStorySwitcherCont = useStorySwitcherContext(),
      active = _useStorySwitcherCont.active;

  return active === null ? /*#__PURE__*/React.createElement(Wrapper, props) : /*#__PURE__*/React.createElement(TransitionSticker, {
    "in": active
  }, /*#__PURE__*/React.createElement(Wrapper, props));
};
var StickerAvatar = /*#__PURE__*/styled(Box).withConfig({
  displayName: "Sticker__StickerAvatar",
  componentId: "sc-1a9u1l7-1"
})(css({
  mb: Space.S_8
}));
var StickerCategory = /*#__PURE__*/styled(Box).withConfig({
  displayName: "Sticker__StickerCategory",
  componentId: "sc-1a9u1l7-2"
})(themeTextStyle(TextStyle.H4));
var StickerTitle = /*#__PURE__*/styled(Box).withConfig({
  displayName: "Sticker__StickerTitle",
  componentId: "sc-1a9u1l7-3"
})(themeTextStyle(TextStyle.H0));
var StickerDescription = /*#__PURE__*/styled(Flex).withConfig({
  displayName: "Sticker__StickerDescription",
  componentId: "sc-1a9u1l7-4"
})(themeTextStyle(TextStyle.CAPTION), css({
  color: Color.GREY_TONE_50
}));
Sticker.Avatar = StickerAvatar;
Sticker.Category = StickerCategory;
Sticker.Title = StickerTitle;
Sticker.Description = StickerDescription;
