// @revolut/ui-kit@11.21.2/esm/components/ZStack/ZStack.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled from 'styled-components';
import { zStack } from "../../styles";
import { Box } from "../Box";
/** @private */

export var Z_STACK_ITEM_AREA = 'rui-z-stack-item';
export var ZStack = /*#__PURE__*/styled(Box).withConfig({
  displayName: "ZStack",
  componentId: "sc-1v99yf7-0"
})(["display:grid;grid-template:'", "' 1fr / 1fr;::before,::after,> *{grid-area:", ";}", ""], Z_STACK_ITEM_AREA, Z_STACK_ITEM_AREA, zStack);
ZStack.defaultProps = {
  use: 'span',
  variantKey: 'zStackStyles',
  variant: 'default'
};
