// @revolut/ui-kit@11.21.2/esm/components/Side/Resizer.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import React, { useRef } from 'react';
import { css } from '@styled-system/css';
import styled from 'styled-components';
import { noop, themeTransition } from "../../utils";
import { useResize } from "../../hooks";
import { Color, Radius, Space, Transition } from "../../constants";
import { Absolute } from "../Position";
import { ButtonBase } from "../ButtonBase";
/** @private */

export var ResizerKnob = /*#__PURE__*/styled(Absolute).withConfig({
  displayName: "Resizer__ResizerKnob",
  componentId: "sc-13is38e-0"
})(css({
  cursor: 'ew-resize',
  userSelect: 'none',
  zIndex: 3,
  backgroundColor: Color.GREY_TONE_20,
  transform: 'translateY(-50%)',
  transition: function transition(theme) {
    return themeTransition(Transition.RESIZE_KNOB)({
      theme: theme
    });
  },
  '&:enabled': {
    cursor: 'ew-resize'
  },
  '&::after': {
    content: "''",
    position: 'absolute',
    top: Space.N_16,
    bottom: Space.N_16,
    left: Space.N_8,
    right: Space.N_8
  }
}));
export var Resizer = function Resizer(_ref) {
  var containerRef = _ref.containerRef,
      _ref$onResizeStart = _ref.onResizeStart,
      onResizeStart = _ref$onResizeStart === void 0 ? noop : _ref$onResizeStart,
      _ref$onResizeEnd = _ref.onResizeEnd,
      onResizeEnd = _ref$onResizeEnd === void 0 ? noop : _ref$onResizeEnd;
  var resizerRef = useRef(null);
  useResize({
    resizerRef: resizerRef,
    containerRef: containerRef,
    onResizeStart: onResizeStart,
    onResizeEnd: onResizeEnd
  });
  return /*#__PURE__*/React.createElement(ResizerKnob, {
    use: ButtonBase,
    ref: resizerRef,
    width: Space.S_4,
    height: Space.S_40,
    top: "50%",
    left: Space.S_8,
    radius: Radius.ROUND
  });
};
