// @revolut/ui-kit@11.21.2/esm/components/Header/constants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

export var BACKGROUND_COLOR_VAR = '--header-background-color';
export var TITLE_ANIMATION_DURATION = 100;
export var TITLE_ANIMATION_OFFSET_Y = -12;
export var HeaderSlots;

(function (HeaderSlots) {
  HeaderSlots[HeaderSlots["TITLE"] = 0] = "TITLE";
  HeaderSlots[HeaderSlots["NAVIGATION_BUTTON"] = 1] = "NAVIGATION_BUTTON";
  HeaderSlots[HeaderSlots["ACTIONS"] = 2] = "ACTIONS";
  HeaderSlots[HeaderSlots["DESCRIPTION"] = 3] = "DESCRIPTION";
  HeaderSlots[HeaderSlots["STATIC"] = 4] = "STATIC";
  HeaderSlots[HeaderSlots["STICKY"] = 5] = "STICKY";
})(HeaderSlots || (HeaderSlots = {}));

export var HeaderScrollMode = {
  AUTO: 'auto',
  MANUAL: 'manual'
};
export var HeaderDisplayMode = {
  AUTO: 'auto',
  INLINE: 'inline',
  LARGE: 'large'
};
