// @revolut/ui-kit@11.21.2/esm/components/Header/HeaderDescription.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import styled from 'styled-components';
import { HeaderBase } from "../HeaderBase";
import { useAccessibleId } from "../AccessibleId";
import { usePopupContext } from "../Popup/PopupContext";
import { PopupVariant } from "../Popup/variants";
import { getDescriptionId } from "./utils";
import { HeaderSlots } from "./constants";
import { HeaderSlotsContext } from "./HeaderSlotsContext";

function HeaderDescriptionBase(props) {
  var id = useAccessibleId();
  var color = usePopupContext() === PopupVariant.COLORFUL ? null : undefined;
  return /*#__PURE__*/React.createElement(HeaderSlotsContext.Fill, {
    name: HeaderSlots.DESCRIPTION
  }, /*#__PURE__*/React.createElement(HeaderBase.Description, _extends({
    id: getDescriptionId(id),
    color: color
  }, props)));
}

export var HeaderDescription = /*#__PURE__*/styled(HeaderDescriptionBase).withConfig({
  displayName: "HeaderDescription",
  componentId: "jtlzwp-0"
})([""]);
