// @revolut/ui-kit@11.21.2/esm/components/Header/HeaderNavigationButtons.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { HeaderBase } from "../HeaderBase";
import { useToggleContext } from "../Toggle";
import { HeaderSlots } from "./constants";
import { HeaderSlotsContext } from "./HeaderSlotsContext";

function useNavigationClickHandler(onClick) {
  var toggle = useToggleContext();
  return function (event) {
    if (onClick) {
      onClick(event);
    } else {
      toggle == null ? void 0 : toggle(false);
    }
  };
}

export var HeaderCloseButton = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var onClick = _ref.onClick,
      rest = _objectWithoutPropertiesLoose(_ref, ["onClick"]);

  var handleClick = useNavigationClickHandler(onClick);
  return /*#__PURE__*/React.createElement(HeaderSlotsContext.Fill, {
    name: HeaderSlots.NAVIGATION_BUTTON
  }, /*#__PURE__*/React.createElement(HeaderBase.CloseButton, _extends({
    ref: ref,
    onClick: handleClick
  }, rest)));
});
export var HeaderBackButton = /*#__PURE__*/React.forwardRef(function (_ref2, ref) {
  var onClick = _ref2.onClick,
      rest = _objectWithoutPropertiesLoose(_ref2, ["onClick"]);

  var handleClick = useNavigationClickHandler(onClick);
  return /*#__PURE__*/React.createElement(HeaderSlotsContext.Fill, {
    name: HeaderSlots.NAVIGATION_BUTTON
  }, /*#__PURE__*/React.createElement(HeaderBase.BackButton, _extends({
    ref: ref,
    onClick: handleClick
  }, rest)));
});
