// @revolut/ui-kit@11.21.2/esm/components/Header/HeaderStickyWrapper.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled from 'styled-components';
import { BACKGROUND_COLOR_VAR } from "./constants";
import { Space } from "../../constants";
import { Sticky } from "../Position";
/** @private */

/** @private */
export var HeaderStickyWrapper = /*#__PURE__*/styled(Sticky).withConfig({
  displayName: "HeaderStickyWrapper",
  componentId: "sc-4fg0wj-0"
})([""]);
HeaderStickyWrapper.defaultProps = {
  top: Space.S_56,
  mx: {
    all: Space.N_16,
    lg: Space.N_24
  },
  px: {
    all: Space.S_16,
    lg: Space.S_24
  },
  bg: "var(" + BACKGROUND_COLOR_VAR + ", inherit)",
  zIndex: 1
};
