// @revolut/ui-kit@11.21.2/esm/components/Header/HeaderSkeleton.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { rem } from 'polished';
import { Skeleton } from "../Skeleton";
import { MoreBarSkeleton } from "../MoreBar";
import { AvatarSkeleton } from "../Avatar";
import { Space } from "../../constants";
import { useHeaderBaseContext } from "../HeaderBase";
import { Header, HeaderVariant } from '.';

var HeaderTitleSkeleton = function HeaderTitleSkeleton(props) {
  var variant = useHeaderBaseContext();
  return /*#__PURE__*/React.createElement(Header.Title, props, /*#__PURE__*/React.createElement(Skeleton, {
    height: variant === HeaderVariant.MODAL || variant === HeaderVariant.BOTTOM_SHEET ? Space.S_16 : Space.S_24,
    width: rem(174),
    my: variant === HeaderVariant.MODAL || variant === HeaderVariant.BOTTOM_SHEET ? Space.S_6 : Space.S_8,
    maxWidth: "100%"
  }));
};

var HeaderSubtitleSkeleton = function HeaderSubtitleSkeleton(props) {
  return /*#__PURE__*/React.createElement(Header.Subtitle, props, /*#__PURE__*/React.createElement(Skeleton, {
    mt: Space.S_6,
    height: Space.S_16,
    width: rem(224),
    maxWidth: "100%"
  }));
};

var HeaderDescriptionSkeleton = function HeaderDescriptionSkeleton(props) {
  return /*#__PURE__*/React.createElement(Header.Description, props, /*#__PURE__*/React.createElement(Skeleton, {
    mt: Space.S_6,
    height: Space.S_16,
    width: rem(104),
    maxWidth: "100%"
  }));
};

var HeaderBarSkeleton = function HeaderBarSkeleton(_ref) {
  var children = _ref.children,
      rest = _objectWithoutPropertiesLoose(_ref, ["children"]);

  return /*#__PURE__*/React.createElement(Header.Bar, rest, /*#__PURE__*/React.createElement(MoreBarSkeleton, null));
};

var HeaderAvatarSkeleton = function HeaderAvatarSkeleton(_ref2) {
  var children = _ref2.children,
      rest = _objectWithoutPropertiesLoose(_ref2, ["children"]);

  return /*#__PURE__*/React.createElement(Header.Avatar, null, /*#__PURE__*/React.createElement(AvatarSkeleton, rest));
};

function HeaderSkeletonBase(_ref3) {
  var variant = _ref3.variant,
      onBack = _ref3.onBack,
      labelBackButton = _ref3.labelBackButton;

  switch (variant) {
    case HeaderVariant.ITEM:
      return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(HeaderSkeleton.BackButton, {
        onClick: onBack,
        "aria-label": labelBackButton
      }), /*#__PURE__*/React.createElement(HeaderSkeleton.Title, null), /*#__PURE__*/React.createElement(HeaderSkeleton.Subtitle, null), /*#__PURE__*/React.createElement(HeaderSkeleton.Description, null), /*#__PURE__*/React.createElement(HeaderSkeleton.Bar, null), /*#__PURE__*/React.createElement(HeaderSkeleton.Avatar, {
        size: 56
      }));

    case HeaderVariant.MAIN:
      return /*#__PURE__*/React.createElement(HeaderSkeleton.Title, null);

    case HeaderVariant.FORM:
      return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(HeaderSkeleton.BackButton, {
        onClick: onBack,
        "aria-label": labelBackButton
      }), /*#__PURE__*/React.createElement(HeaderSkeleton.Title, null), /*#__PURE__*/React.createElement(HeaderSkeleton.Subtitle, null), /*#__PURE__*/React.createElement(HeaderSkeleton.Bar, null));

    case HeaderVariant.MODAL:
      return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(HeaderSkeleton.BackButton, {
        onClick: onBack,
        "aria-label": labelBackButton
      }), /*#__PURE__*/React.createElement(HeaderSkeleton.Title, null));

    case HeaderVariant.BOTTOM_SHEET:
      return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(HeaderSkeleton.Title, null), /*#__PURE__*/React.createElement(HeaderSkeleton.Subtitle, null));

    default:
      return null;
  }
}

export var HeaderSkeleton = function HeaderSkeleton(_ref4) {
  var children = _ref4.children,
      onBack = _ref4.onBack,
      rest = _objectWithoutPropertiesLoose(_ref4, ["children", "onBack"]);

  return /*#__PURE__*/React.createElement(Header, rest, children != null ? children : /*#__PURE__*/React.createElement(HeaderSkeletonBase, _extends({
    onBack: onBack
  }, rest)));
};
HeaderSkeleton.CloseButton = Header.CloseButton;
HeaderSkeleton.BackButton = Header.BackButton;
HeaderSkeleton.Title = HeaderTitleSkeleton;
HeaderSkeleton.Subtitle = HeaderSubtitleSkeleton;
HeaderSkeleton.Description = HeaderDescriptionSkeleton;
HeaderSkeleton.Bar = HeaderBarSkeleton;
HeaderSkeleton.Avatar = HeaderAvatarSkeleton;
