// @revolut/ui-kit@11.21.2/esm/components/SliderBox/styles.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled, { css } from 'styled-components';
import { Breakpoint, Color, Radius, Size, Space, Transition } from "../../constants";
import { rgbaSafe, themeColor, themeRadius, themeSize, themeSpace, themeTransition } from "../../utils";
import { mq } from "../../styles";
import { ButtonBase } from "../ButtonBase";
import { Absolute, Relative } from "../Position";
export var Wrapper = /*#__PURE__*/styled(Relative).withConfig({
  displayName: "styles__Wrapper",
  componentId: "sc-1fbikzf-0"
})(["display:flex;flex-direction:column;", " ", ""], function (props) {
  switch (props.fillSpace) {
    case 'layout':
      return css(["--rui-slider-wrapper:", ";overflow:hidden;margin-left:calc(var(--rui-slider-wrapper) * -1);margin-right:calc(var(--rui-slider-wrapper) * -1);padding-left:var(--rui-slider-wrapper);padding-right:var(--rui-slider-wrapper);@media ", "{--rui-slider-wrapper:", ";}@media ", "{--rui-slider-wrapper:initial;overflow:initial;}"], themeSpace(Space.S_16), mq(Breakpoint.MD), themeSpace(Space.S_24), mq(Breakpoint.LG));

    default:
      return null;
  }
}, function (_ref) {
  var buttonTransformSpace = _ref.buttonTransformSpace;
  return buttonTransformSpace && css(["--rui-slider-wrapper-button-transform:", ";"], themeSpace(buttonTransformSpace));
});
var ScrollButton = /*#__PURE__*/styled(ButtonBase).withConfig({
  displayName: "styles__ScrollButton",
  componentId: "sc-1fbikzf-1"
})(["position:absolute;top:50%;padding:", ";background-color:transparent;border-radius:", ";color:", ";opacity:0;transition:", ",", ",", ",", ";::after{content:'';position:absolute;top:0;bottom:0;left:0;right:0;transition:", ";z-index:1;}@media (hover:hover) and (pointer:fine){", ":hover &{background-color:", ";box-shadow:0px 2px 3px rgba(0,0,0,0.05);opacity:1;}}", ""], themeSpace(Space.S_8), themeRadius(Radius.MENU_TAB), themeColor(Color.GREY_TONE_50), themeTransition(Transition.TRANSFORM), themeTransition(Transition.BACKGROUND), themeTransition(Transition.SHADOW), themeTransition(Transition.OPACITY), themeTransition(Transition.TRANSFORM), Wrapper, themeColor(Color.WIDGET_BACKGROUND), function (_ref2) {
  var _ref2$hidePreview = _ref2.hidePreview,
      hidePreview = _ref2$hidePreview === void 0 ? Breakpoint.MD : _ref2$hidePreview;
  return hidePreview ? css(["@media ", "{opacity:1;}"], mq(hidePreview)) : css(["opacity:1;"]);
});
export var ScrollLeftButton = /*#__PURE__*/styled(ScrollButton).withConfig({
  displayName: "styles__ScrollLeftButton",
  componentId: "sc-1fbikzf-2"
})(["left:var(--rui-slider-wrapper,0);transform:translate(var(--rui-slider-wrapper-button-transform,-100%),-50%);", ":hover &{transform:translate( var(--rui-slider-wrapper-button-transform,-50%),-50% );::after{transform:translate(-25%,0);}}"], Wrapper);
export var ScrollRightButton = /*#__PURE__*/styled(ScrollButton).withConfig({
  displayName: "styles__ScrollRightButton",
  componentId: "sc-1fbikzf-3"
})(["right:var(--rui-slider-wrapper,0);transform:translate(var(--rui-slider-wrapper-button-transform,100%),-50%);", ":hover &{transform:translate(var(--rui-slider-wrapper-button-transform,50%),-50%);::after{transform:translate(25%,0);}}"], Wrapper);
export var ScrollButtonPlaceholder = /*#__PURE__*/styled(Absolute).withConfig({
  displayName: "styles__ScrollButtonPlaceholder",
  componentId: "sc-1fbikzf-4"
})(["top:0;height:100%;width:", ";"], themeSize(Size.S_40));
var Gradient = /*#__PURE__*/styled(Absolute).withConfig({
  displayName: "styles__Gradient",
  componentId: "sc-1fbikzf-5"
})(["width:", ";height:100%;top:0;background:", ";"], themeSize(Space.S_56), function (_ref3) {
  var theme = _ref3.theme;
  return "linear-gradient(\n    90deg,\n    " + rgbaSafe(theme.colors[Color.LAYOUT_BACKGROUND], 1) + " 65%,\n    " + rgbaSafe(theme.colors[Color.LAYOUT_BACKGROUND], 0.9) + " 70%,\n    " + rgbaSafe(theme.colors[Color.LAYOUT_BACKGROUND], 0) + " 90%\n  )";
});
export var GradientLeft = /*#__PURE__*/styled(Gradient).withConfig({
  displayName: "styles__GradientLeft",
  componentId: "sc-1fbikzf-6"
})(["left:var(--rui-slider-wrapper,0);transform:translate(-0.25rem,0);"]);
export var GradientRight = /*#__PURE__*/styled(Gradient).withConfig({
  displayName: "styles__GradientRight",
  componentId: "sc-1fbikzf-7"
})(["right:var(--rui-slider-wrapper,0);transform:translate(0.25rem,0) rotate(180deg);"]);
