// @revolut/ui-kit@11.21.2/esm/components/Details/DetailsSkeleton.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { rem } from 'polished';
import { Details, DetailsCell } from "./Details";
import { Skeleton } from "../Skeleton";

var DetailsSkeletonTitle = function DetailsSkeletonTitle(props) {
  return /*#__PURE__*/React.createElement(Details.Title, props, /*#__PURE__*/React.createElement(Skeleton, {
    height: rem(16),
    width: 0.5,
    minWidth: "7rem"
  }));
};

var DetailsSkeletonContent = function DetailsSkeletonContent(props) {
  return /*#__PURE__*/React.createElement(Details.Content, props, /*#__PURE__*/React.createElement(Skeleton, {
    height: rem(16),
    width: 0.5,
    minWidth: "5rem"
  }));
};

var DetailsSkeletonNote = function DetailsSkeletonNote(props) {
  return /*#__PURE__*/React.createElement(Details.Note, props, /*#__PURE__*/React.createElement(Skeleton, {
    height: rem(16),
    width: 0.6,
    minWidth: "10rem"
  }));
};

export var DetailsSkeleton = function DetailsSkeleton(_ref) {
  var children = _ref.children,
      rest = _objectWithoutPropertiesLoose(_ref, ["children"]);

  return /*#__PURE__*/React.createElement(Details, rest, children != null ? children : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(DetailsSkeletonTitle, null), /*#__PURE__*/React.createElement(DetailsSkeletonContent, null)));
};
DetailsSkeleton.Content = DetailsSkeletonContent;
DetailsSkeleton.Note = DetailsSkeletonNote;
DetailsSkeleton.Title = DetailsSkeletonTitle;
export var DetailsCellSkeleton = function DetailsCellSkeleton(_ref2) {
  var children = _ref2.children,
      rest = _objectWithoutPropertiesLoose(_ref2, ["children"]);

  return /*#__PURE__*/React.createElement(DetailsCell, rest, children != null ? children : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(DetailsSkeletonTitle, null), /*#__PURE__*/React.createElement(DetailsSkeletonContent, null)));
};
DetailsCellSkeleton.Content = DetailsSkeletonContent;
DetailsCellSkeleton.Note = DetailsSkeletonNote;
DetailsCellSkeleton.Title = DetailsSkeletonTitle;
