// @revolut/ui-kit@11.21.2/esm/legacy/components/DatePicker/styles.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import ReactDayPicker from 'react-day-picker';
import styled, { css } from 'styled-components';
import { width } from 'styled-system';
import { themeGet } from '@styled-system/theme-get';
import { transition } from "../../../styles";
import { Circle } from "../../../components/Circle";
import { TextBox } from "../../../components/Text";
import { DatePickerSize } from "./constants";
var classes = ReactDayPicker.defaultProps.classNames;
export var DatePickerContainer = /*#__PURE__*/styled(TextBox).withConfig({
  displayName: "styles__DatePickerContainer",
  componentId: "sc-9nlgsi-0"
})([".", "{display:inline-block;width:", ";max-width:100%;", "}.", "{position:relative;user-select:none;flex-direction:row;outline:0;}.", "{display:flex;flex-wrap:wrap;justify-content:center;}.", "{display:table;border-collapse:collapse;border-spacing:0;user-select:none;table-layout:fixed;width:100%;}.", "{display:table-header-group;}.", "{display:table-row;}.", "{display:table-cell;& abbr[title]{border-bottom:none;text-decoration:none;}}.", "{display:table-row-group;}.", "{display:table-row;}.", "{display:table-cell;cursor:pointer;vertical-align:middle;outline:none;}.", " .", "{cursor:default;}"], classes.container, DatePickerSize.BOX, width, classes.wrapper, classes.months, classes.month, classes.weekdays, classes.weekdaysRow, classes.weekday, classes.body, classes.week, classes.day, classes.interactionDisabled, classes.day);
export var DatePickerMonth = /*#__PURE__*/styled(TextBox).withConfig({
  displayName: "styles__DatePickerMonth",
  componentId: "sc-9nlgsi-1"
})(["display:table-caption;font-weight:500;"]);
export var DatePickerWeekDay = /*#__PURE__*/styled(TextBox).withConfig({
  displayName: "styles__DatePickerWeekDay",
  componentId: "sc-9nlgsi-2"
})(["display:table-cell;width:", ";height:", ";text-align:center;color:", ";"], DatePickerSize.DAY, DatePickerSize.DAY, themeGet('colors.grey-50'));
export var DatePickerDay = /*#__PURE__*/styled(Circle).withConfig({
  displayName: "styles__DatePickerDay",
  componentId: "sc-9nlgsi-3"
})(["width:", ";height:", ";transition:all ", ";text-align:center;&:hover{background-color:rgba(0,0,0,0.05);}", " ", " ", " ", ""], DatePickerSize.DAY, DatePickerSize.DAY, transition(), function (props) {
  return props.isToday && css(["font-weight:500;color:", ";"], themeGet('colors.accent'));
}, function (props) {
  return props.isDisabled && css(["color:", ";pointer-events:none;"], themeGet('colors.grey-50'));
}, function (props) {
  return props.outside && css(["opacity:0.3;"]);
}, function (props) {
  return (props.isSelected || props.selected) && css(["&,&:hover{background-color:", ";color:", ";opacity:", ";}"], themeGet('colors.primary'), themeGet('colors.white'), (props.isDisabled || props.outside) && 0.3);
});
export var DatePickerNavBarButton = /*#__PURE__*/styled(Circle).withConfig({
  displayName: "styles__DatePickerNavBarButton",
  componentId: "sc-9nlgsi-4"
})(["cursor:pointer;"]);
