// @revolut/ui-kit@11.21.2/esm/components/Avatar/constants.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

export var AvatarVariant = {
  DEFAULT: 'default',
  BRAND: 'brand',

  /** @deprecated Use `AppIcon` component instead */
  APP: 'app'
};
export var AvatarBadgePosition = {
  TOP_RIGHT: 'top-right',
  BOTTOM_RIGHT: 'bottom-right',

  /** @private  */
  TOP_LEFT: 'top-left',

  /** @private  */
  BOTTOM_LEFT: 'bottom-left'
};

/** @private */
export var AvatarComponentSize = {
  S_20: 20,
  S_24: 24,
  S_32: 32,
  S_40: 40,
  S_56: 56,
  S_60: 60,
  S_76: 76,
  S_96: 96
};
/** @private */
