// @revolut/ui-kit@11.21.2/esm/constants/font-sizes.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

/** @deprecated */
export var ResponsiveFontSize = {
  /** @deprecated */
  H0_PLUS: 'h0+',

  /** @deprecated */
  H1_PLUS: 'h1+',

  /** @deprecated */
  H2_PLUS: 'h2+',

  /** @deprecated */
  H3_PLUS: 'h3+',

  /** @deprecated */
  H4_PLUS: 'h4+'
};
export var BaseFontSize = {
  H0: 'h0',
  H1: 'h1',
  H2: 'h2',
  H3: 'h3',
  H4: 'h4',
  H5: 'h5',
  H6: 'h6',
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  BULLET: 'bullet',
  CAPTION: 'caption',
  SMALL: 'small',
  TINY: 'tiny',

  /** @deprecated */
  HELPER: 'helper'
};
export var FontSize = _extends({}, ResponsiveFontSize, BaseFontSize, {
  /** @deprecated */
  AVATAR: 'avatar'
});
