// @revolut/ui-kit@11.21.2/esm/constants/radii.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

export var Radius = {
  UNSET: 'unset',
  ROUND: 'round',
  DROPDOWN: 'dropdown',
  WIDGET: 'widget',
  CARD: 'card',
  CARD_TOP: 'card-top',
  CARD_BOTTOM: 'card-bottom',
  CELL: 'cell',
  INPUT: 'input',
  IMAGE: 'image',
  BUTTON_SM: 'button-sm',
  BUTTON_MD: 'button-md',
  BUTTON_BAR: 'button-bar',
  TAG: 'tag',
  TOAST: 'toast',
  TOOLTIP: 'tooltip',
  POPUP: 'popup',
  POPUP_TOP: 'popup-top',
  POPUP_BOTTOM: 'popup-bottom',
  POPUP_FULL_SCREEN: 'popup-full-screen',
  SHEET: 'sheet',
  SHEET_TOP: 'sheet-top',
  SHEET_BOTTOM: 'sheet-bottom',
  BUBBLE_TOP_LEFT: 'bubble-top-left',
  BUBBLE_TOP_RIGHT: 'bubble-top-right',
  BUBBLE_BOTTOM_LEFT: 'bubble-bottom-left',
  BUBBLE_BOTTOM_RIGHT: 'bubble-bottom-right',
  TAB_OUTER: 'tab-outer',
  TAB_INNER: 'tab-inner',
  MENU_TAB: 'menu-tab',
  SEARCH: 'search',
  CALENDAR_DAY: 'calendar-day',
  LEGEND: 'legend',
  APP: 'app',
  STORY: 'story',
  IMAGE_CLIP: 'image-clip',
  REWARD_STICKER: 'reward-sticker',
  BOTTOM_NAV: 'bottom-nav',
  ATTACHMENT: 'attachment',

  /** @deprecated */
  BUTTON_XS: 'button-xs',

  /** @deprecated */
  POPUP_PLUS: 'popup+',

  /** @deprecated */
  BUTTON_LG: 'button-lg'
};
