// @revolut/ui-kit@11.21.2/esm/constants/sizes.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { Space } from "./space";
export var AvatarSize = {
  DEFAULT: 'default',
  S_24: 24,
  S_40: 40,
  S_96: 96,

  /** @deprecated */
  XS: 'xs',

  /** @deprecated */
  SM: 'sm',

  /** @deprecated */
  MD: 'md',

  /** @deprecated */
  MD_PLUS: 'md+',

  /** @deprecated */
  LG: 'lg',

  /** @deprecated */
  LG_PLUS: 'lg+',

  /** @deprecated */
  XL: 'xl',

  /** @deprecated */
  XXL: 'xxl'
};
export var ButtonSize = {
  XS: 'xs',
  SM: 'sm',
  MD: 'md',

  /** @private */
  TAG: 'tag',

  /** @deprecated */
  MD_PLUS: 'md+'
};
export var SizeScale = _extends({
  ICON: 'icon',
  AVATAR: 'avatar',
  BUTTON: 'button',
  LINE: 'line',
  DETAILS: 'details',
  LAYOUT: 'layout',
  LAYOUT_ACTIONS: 'layout-actions',
  TAB_NAVIGATION: 'tab-navigation',
  TAB_SEGMENTED: 'tab-segmented',
  INDICATOR_NAVIGATION: 'indicator-navigation',
  INDICATOR_SEGMENTED: 'indicator-segmented'
}, Space);
export var Size = _extends({
  ICON_10: 'icon.10',
  ICON_16: 'icon.16',
  ICON_20: 'icon.20',
  ICON_24: 'icon.24',

  /** @deprecated */
  ICON_32: 'icon.32',
  ICON_40: 'icon.40',

  /** @deprecated */
  ICON_48: 'icon.48',

  /** @deprecated */
  ICON_64: 'icon.64',
  ICON_96: 'icon.96',
  ICON_104: 'icon.104',

  /** @deprecated */
  ICON_120: 'icon.120',
  AVATAR_DEFAULT: 'avatar.default',
  AVATAR_24: 'avatar.24',
  AVATAR_40: 'avatar.40',
  AVATAR_96: 'avatar.96',

  /** @deprecated */
  AVATAR_XS: 'avatar.xs',

  /** @deprecated */
  AVATAR_SM: 'avatar.sm',

  /** @deprecated */
  AVATAR_MD: 'avatar.md',

  /** @deprecated */
  AVATAR_MD_PLUS: 'avatar.md+',

  /** @deprecated */
  AVATAR_LG: 'avatar.lg',

  /** @deprecated */
  AVATAR_LG_PLUS: 'avatar.lg+',

  /** @deprecated */
  AVATAR_XL: 'avatar.xl',

  /** @deprecated */
  AVATAR_XXL: 'avatar.xxl',

  /** @deprecated */
  BUTTON_XS: 'button.xs',
  BUTTON_SM: 'button.sm',
  BUTTON_MD: 'button.md',

  /** @deprecated */
  BUTTON_MD_PLUS: 'button.md+',
  LINE: 'line',
  DETAILS: 'details',
  LAYOUT: 'layout',
  LAYOUT_ACTIONS: 'layout-actions',
  TAB_NAVIGATION: 'tab-navigation',
  TAB_SEGMENTED: 'tab-segmented',
  INDICATOR_NAVIGATION: 'indicator-navigation',
  INDICATOR_SEGMENTED: 'indicator-segmented'
}, Space);
