// @revolut/ui-kit@11.21.2/esm/constants/space.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _BASE_SPACE;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { rem } from 'polished';
export var PositiveSpace = {
  S_2: 's-2',
  S_4: 's-4',
  S_6: 's-6',
  S_8: 's-8',
  S_12: 's-12',
  S_16: 's-16',
  S_20: 's-20',
  S_24: 's-24',
  S_32: 's-32',
  S_40: 's-40',
  S_48: 's-48',
  S_56: 's-56',
  S_72: 's-72',
  GRID: 'grid'
};
export var NegativeSpace = {
  N_2: '-s-2',
  N_4: '-s-4',
  N_6: '-s-6',
  N_8: '-s-8',
  N_12: '-s-12',
  N_16: '-s-16',
  N_20: '-s-20',
  N_24: '-s-24',
  N_32: '-s-32',
  N_40: '-s-40',
  N_48: '-s-48',
  N_56: '-s-56',
  N_72: '-s-72',
  N_GRID: '-grid'
};
export var Space = _extends({
  NONE: 'none'
}, PositiveSpace, NegativeSpace);
export var BASE_SPACE = (_BASE_SPACE = {}, _BASE_SPACE[Space.NONE] = 0, _BASE_SPACE[Space.S_2] = rem(2), _BASE_SPACE[Space.S_4] = rem(4), _BASE_SPACE[Space.S_6] = rem(6), _BASE_SPACE[Space.S_8] = rem(8), _BASE_SPACE[Space.S_12] = rem(12), _BASE_SPACE[Space.S_16] = rem(16), _BASE_SPACE[Space.S_20] = rem(20), _BASE_SPACE[Space.S_24] = rem(24), _BASE_SPACE[Space.S_32] = rem(32), _BASE_SPACE[Space.S_40] = rem(40), _BASE_SPACE[Space.S_48] = rem(48), _BASE_SPACE[Space.S_56] = rem(56), _BASE_SPACE[Space.S_72] = rem(72), _BASE_SPACE[Space.GRID] = rem(16), _BASE_SPACE[Space.N_2] = rem(-2), _BASE_SPACE[Space.N_4] = rem(-4), _BASE_SPACE[Space.N_6] = rem(-6), _BASE_SPACE[Space.N_8] = rem(-8), _BASE_SPACE[Space.N_12] = rem(-12), _BASE_SPACE[Space.N_16] = rem(-16), _BASE_SPACE[Space.N_20] = rem(-20), _BASE_SPACE[Space.N_24] = rem(-24), _BASE_SPACE[Space.N_32] = rem(-32), _BASE_SPACE[Space.N_40] = rem(-40), _BASE_SPACE[Space.N_48] = rem(-48), _BASE_SPACE[Space.N_56] = rem(-56), _BASE_SPACE[Space.N_72] = rem(-72), _BASE_SPACE[Space.N_GRID] = rem(-16), _BASE_SPACE);
/** @private */

export var PRIVATE_SPACE = [Space.NONE, Space.N_2, Space.N_4, Space.N_6, Space.N_8, Space.N_12, Space.N_16, Space.N_20, Space.N_24, Space.N_32, Space.N_40, Space.N_48, Space.N_56, Space.N_72, Space.N_GRID];
