// @revolut/ui-kit@11.21.2/esm/constants/text-styles.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

export var ComponentTextStyle = {
  BUTTON: 'button',
  INPUT: 'input',
  TILE: 'tile',
  ACTION: 'action',

  /** @deprecated */
  AVATAR: 'avatar',

  /** @deprecated */
  TEXT_BUTTON: 'text-button'
};
export var TextStyle = _extends({
  DEFAULT: 'default',
  H0: 'h0',
  H1: 'h1',
  H2: 'h2',
  H3: 'h3',
  H4: 'h4',
  H5: 'h5',
  H6: 'h6',
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  BULLET: 'bullet',
  CAPTION: 'caption',
  SMALL: 'small',
  TINY: 'tiny',

  /** @deprecated */
  HELPER: 'helper'
}, ComponentTextStyle);
export var TEXT_STYLES_SCALE = [TextStyle.H0, TextStyle.H1, TextStyle.H2, TextStyle.H3, TextStyle.H4, TextStyle.H5, TextStyle.H6, TextStyle.PRIMARY, TextStyle.SECONDARY, TextStyle.BULLET, TextStyle.CAPTION, TextStyle.SMALL, TextStyle.TINY];
