// @revolut/ui-kit@11.21.2/esm/components/Amount/AmountSkeleton.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { Color, Space } from "../../constants";
import { Flex } from "../Flex";
import { Skeleton } from "../Skeleton";
import { Amount } from "./Amount";
import { AmountInputWrapper } from "./AmountInputWrapper";
import { AmountStatus } from "./AmountStatus";
import { Area } from "./constants";
var SkeletonStyled = /*#__PURE__*/styled(Skeleton).withConfig({
  displayName: "AmountSkeleton__SkeletonStyled",
  componentId: "o4l9pe-0"
})(["opacity:0.4;"]);

var AmountMainSkeleton = function AmountMainSkeleton(props) {
  return /*#__PURE__*/React.createElement(SkeletonStyled, _extends({
    my: Space.S_4,
    height: rem(20),
    width: rem(80),
    bg: Color.GREY_TONE_20
  }, props));
};

var AmountMessageSkeleton = function AmountMessageSkeleton(props) {
  return /*#__PURE__*/React.createElement(SkeletonStyled, _extends({
    mt: Space.S_6,
    mb: Space.S_4,
    height: rem(12),
    width: rem(100),
    bg: Color.GREY_TONE_20
  }, props));
};

function AmountCurrencySkeleton(_ref) {
  var children = _ref.children,
      rest = _objectWithoutPropertiesLoose(_ref, ["children"]);

  return /*#__PURE__*/React.createElement(Amount.Currency, _extends({
    value: /*#__PURE__*/React.createElement(AmountMainSkeleton, null)
  }, rest));
}

function AmountInputSkeleton(_ref2) {
  var message = _ref2.message,
      rest = _objectWithoutPropertiesLoose(_ref2, ["message"]);

  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(AmountInputWrapper, null, /*#__PURE__*/React.createElement(Flex, {
    justifyContent: "flex-end"
  }, /*#__PURE__*/React.createElement(AmountMainSkeleton, rest))), message && /*#__PURE__*/React.createElement(AmountStatus, {
    gridArea: Area.D,
    justifySelf: "end",
    textAlign: "right"
  }, message));
}

export var AmountSkeleton = function AmountSkeleton(_ref3) {
  var children = _ref3.children,
      rest = _objectWithoutPropertiesLoose(_ref3, ["children"]);

  return /*#__PURE__*/React.createElement(Amount, rest, children != null ? children : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(AmountSkeleton.Currency, null), /*#__PURE__*/React.createElement(AmountSkeleton.Input, null)));
};
AmountSkeleton.Currency = AmountCurrencySkeleton;
AmountSkeleton.Input = AmountInputSkeleton;
AmountSkeleton.Message = AmountMessageSkeleton;
