// @revolut/ui-kit@11.21.2/esm/components/Amount/Amount.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled from 'styled-components';
import { css } from '@styled-system/css';
import { Space } from "../../constants";
import { InputBox, InputBoxVariant } from "../InputBox";
import { Area } from "./constants";
import { AmountStatus } from "./AmountStatus";
import { AmountInput } from "./AmountInput";
import { AmountCurrency } from "./AmountCurrency";
var AmountContainer = /*#__PURE__*/styled(InputBox).withConfig({
  displayName: "Amount__AmountContainer",
  componentId: "sc-1559bx9-0"
})(function (props) {
  var _css;

  return css((_css = {
    display: 'grid',
    paddingY: props.variant === InputBoxVariant.GREY ? Space.S_16 : Space.S_8,
    paddingX: Space.S_16,
    columnGap: Space.S_16,
    gridTemplateAreas: "\n      '" + Area.A + " " + Area.B + " " + Area.B + " " + Area.B + "'\n      '" + Area.C + " " + Area.C + " " + Area.D + " " + Area.D + "'\n    "
  }, _css[AmountStatus] = props.variant === InputBoxVariant.GREY ? {
    marginTop: Space.S_4
  } : {}, _css));
});
export var Amount = Object.assign(AmountContainer, {
  Currency: AmountCurrency,
  Input: AmountInput
});
