// @revolut/ui-kit@11.21.2/esm/components/SideNav/styles.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

var _css, _css3;

import styled from 'styled-components';
import { css } from '@styled-system/css';
import { position } from 'styled-system';
import { Color, Radius, Space, TextStyle, Transition } from "../../constants";
import { themeTextStyle, themeTransition } from "../../utils";
import { Box } from "../Box";
import { ButtonBase } from "../ButtonBase";
import { Flex } from "../Flex";
import { Text } from "../Text";
import { Badge, BadgeVariant } from "../Badge";

var dropDisable = function dropDisable(props) {
  return {
    transition: themeTransition(Transition.OPACITY)(props),
    opacity: props.disabled ? 0.3 : undefined,
    pointerEvents: props.disabled ? 'none' : undefined
  };
};

export var SideNavItemsBase = /*#__PURE__*/styled(Box).withConfig({
  displayName: "styles__SideNavItemsBase",
  componentId: "sc-1obogf8-0"
})(css({
  my: Space.S_24
}));
export var SideNavContainer = /*#__PURE__*/styled(Box).withConfig({
  displayName: "styles__SideNavContainer",
  componentId: "sc-1obogf8-1"
})(css((_css = {}, _css["& > " + SideNavItemsBase + " + " + SideNavItemsBase] = {
  mt: 0
}, _css)));
export var SideNavHeaderButton = /*#__PURE__*/styled(ButtonBase).withConfig({
  displayName: "styles__SideNavHeaderButton",
  componentId: "sc-1obogf8-2"
})(dropDisable, css({
  width: '100%',
  borderRadius: Radius.MENU_TAB,
  ':hover:enabled': {
    backgroundColor: Color.GREY_TONE_10
  },
  "&[aria-pressed='true'], &[aria-selected='true'], &[aria-current='page']": {
    backgroundColor: Color.WIDGET_BACKGROUND,
    color: Color.BLUE
  }
}));
export var SideNavHeaderBase = /*#__PURE__*/styled(Box).withConfig({
  displayName: "styles__SideNavHeaderBase",
  componentId: "sc-1obogf8-3"
})(css({
  alignItems: 'center',
  columnGap: Space.S_12,
  display: 'grid',
  gridTemplateColumns: 'auto',
  gridAutoColumns: '1fr',
  gridTemplateRows: '1fr auto'
}));
export var SideNavAvatar = /*#__PURE__*/styled(Box).withConfig({
  displayName: "styles__SideNavAvatar",
  componentId: "sc-1obogf8-4"
})({
  alignSelf: 'center',
  gridColumn: 1,
  gridRow: '1 / span 2',
  justifySelf: 'center'
});
SideNavAvatar.defaultProps = {
  use: 'span'
};
export var SideNavTitleBase = /*#__PURE__*/styled(Text).withConfig({
  displayName: "styles__SideNavTitleBase",
  componentId: "sc-1obogf8-5"
})(themeTextStyle(TextStyle.TILE), {
  gridColumn: 2,
  gridRow: 1,
  overflow: 'hidden',
  justifySelf: 'start',
  maxWidth: '100%'
});
export var SideNavDescriptionBase = /*#__PURE__*/styled(Text).withConfig({
  displayName: "styles__SideNavDescriptionBase",
  componentId: "sc-1obogf8-6"
})(themeTextStyle(TextStyle.CAPTION), function (_ref) {
  var onClick = _ref.onClick;
  return css({
    alignItems: 'center',
    color: Color.GREY_TONE_50,
    gridColumn: 2,
    gridRow: 2,
    overflow: 'hidden',
    ':hover': {
      color: onClick ? Color.BLUE : undefined
    }
  });
});
export var SideNavItemBase = /*#__PURE__*/styled(ButtonBase).withConfig({
  displayName: "styles__SideNavItemBase",
  componentId: "sc-1obogf8-7"
})(dropDisable, function (props) {
  return css({
    borderRadius: Radius.MENU_TAB,
    color: Color.GREY_TONE_50,
    display: 'block',
    width: '100%',
    transition: themeTransition(Transition.BUTTON)(props),
    "&[aria-pressed='true'], &[aria-selected='true'], &[aria-current='page']": {
      backgroundColor: Color.WIDGET_BACKGROUND,
      color: Color.BLUE
    },
    ":hover:not([disabled]):not(:disabled):not([aria-pressed='true']):not([aria-selected='true']):not([aria-current='page'])": {
      backgroundColor: Color.GREY_TONE_10
    }
  });
});
export var SideNavItemLayoutBase = /*#__PURE__*/styled(Flex).withConfig({
  displayName: "styles__SideNavItemLayoutBase",
  componentId: "sc-1obogf8-8"
})(css({
  alignItems: 'center',
  position: 'relative',
  paddingY: Space.S_12
}));
export var SideNavItemMenuContainer = /*#__PURE__*/styled(Box).withConfig({
  displayName: "styles__SideNavItemMenuContainer",
  componentId: "sc-1obogf8-9"
})(function (props) {
  var _css2;

  return css((_css2 = {
    color: Color.GREY_TONE_20,
    position: 'absolute',
    right: props.collapsed ? Space.N_24 : Space.S_12,
    // Keeps button's above tooltip
    zIndex: 1,
    '> *:hover:not([disabled])': {
      color: Color.GREY_TONE_50
    },
    opacity: 0,
    transition: themeTransition(Transition.OPACITY)(props)
  }, _css2[SideNavItemLayoutBase + ":hover &"] = {
    opacity: 1
  }, _css2));
});
export var SideNavDraggableIcon = /*#__PURE__*/styled(Box).withConfig({
  displayName: "styles__SideNavDraggableIcon",
  componentId: "sc-1obogf8-10"
})(css((_css3 = {
  color: Color.GREY_TONE_20,
  position: 'absolute',
  right: Space.S_12
}, _css3[SideNavItemLayoutBase + ":hover &"] = {
  color: Color.GREY_TONE_50
}, _css3)));
export var SideNavItemsBadge = /*#__PURE__*/styled(Badge).withConfig({
  displayName: "styles__SideNavItemsBadge",
  componentId: "sc-1obogf8-11"
})(position);
SideNavItemsBadge.defaultProps = {
  variant: BadgeVariant.DOT
};
