// @revolut/ui-kit@11.21.2/esm/components/Overlay/Overlay.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled, { css } from 'styled-components';
import * as SS from 'styled-system';
import { Color, Transition } from "../../constants";
import { isBoolean, themeTransition } from "../../utils";
import { Position } from "../Position";
import { OverlayCloseButton } from "./OverlayCloseButton";
export var OverlayBase = /*#__PURE__*/styled(Position).withConfig({
  displayName: "Overlay__OverlayBase",
  componentId: "sc-1nj0ikp-0"
})(["", " overflow-x:hidden;overflow-y:auto;-webkit-overflow-scrolling:touch;:focus{outline:none;}", " ", ""], SS.position, function (props) {
  return props.__noBackground && css(["background-color:transparent;"]);
}, function (props) {
  return props.__layerFilter && isBoolean(props.__isTailLayer) && css(["> *{transition:", ";}> [role='dialog']{filter:brightness(", ");}>:not([role='dialog']){filter:brightness(", ");}"], themeTransition(Transition.FILTER), props.__isTailLayer ? 0.4 : 1, props.__isTailLayer ? 0.2 : 1);
});
export var Overlay = Object.assign(OverlayBase, {
  CloseButton: OverlayCloseButton
});
Overlay.defaultProps = {
  position: 'fixed',
  top: 0,
  left: 0,
  right: 0,
  bottom: 0,
  bg: Color.MODAL_OVERLAY
};
