// @revolut/ui-kit@11.21.2/esm/components/Overlay/OverlayCloseButton.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import * as Icons from '@revolut/icons';
import styled from 'styled-components';
import { css } from '@styled-system/css';
import { Color, Space } from "../../constants";
import { useMessage } from "../../hooks";
import { ButtonBase } from "../ButtonBase";

var OverlayCloseButtonBase = function OverlayCloseButtonBase(_ref) {
  var ariaLabel = _ref['aria-label'],
      rest = _objectWithoutPropertiesLoose(_ref, ["aria-label"]);

  var m = useMessage();
  return /*#__PURE__*/React.createElement(ButtonBase, _extends({
    "aria-label": ariaLabel != null ? ariaLabel : m('action.close')
  }, rest), /*#__PURE__*/React.createElement(Icons.Cross, null));
};

export var OverlayCloseButton = /*#__PURE__*/styled(OverlayCloseButtonBase).withConfig({
  displayName: "OverlayCloseButton",
  componentId: "sc-1yb3dco-0"
})(css({
  position: 'absolute',
  color: Color.WHITE,
  top: [Space.S_12, null, Space.S_32],
  right: [Space.S_16, null, Space.S_40]
}));
