// @revolut/ui-kit@11.21.2/esm/components/TextButton/TextButton.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled from 'styled-components';
import { color, fontFamily, fontSize, fontWeight, height, letterSpacing, lineHeight, maxWidth, space, textAlign, width } from 'styled-system';
import { textButtonSize, textStyle, variant } from "../../styles";
import { TextStyle } from "../../constants";
import { ButtonBase } from "../ButtonBase";
import { TEXT_BUTTON_STYLES, TextButtonVariant } from "./variants";
export var TextButton = /*#__PURE__*/styled(ButtonBase).withConfig({
  displayName: "TextButton",
  componentId: "wk5noy-0"
})(textButtonSize, variant({
  scale: 'textButtonStyles',
  variants: TEXT_BUTTON_STYLES
}), textStyle, fontFamily, fontSize, fontWeight, letterSpacing, lineHeight, textAlign, color, space, width, height, maxWidth);
TextButton.defaultProps = {
  textStyle: TextStyle.BUTTON,
  variant: TextButtonVariant.DEFAULT,
  display: 'inline-block',
  width: 'max-content',
  maxWidth: '100%'
};
