// @revolut/ui-kit@11.21.2/esm/components/PerformanceChain/PerformanceChain.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { Color, TextStyle } from "../../constants";
import { themeTextStyle } from "../../utils";
import { Flex } from "../Flex";
import { Badge, BadgeVariant } from "../Badge";
export var PerformanceChainColor = {
  RED: Color.RED,
  ORANGE: Color.ORANGE,
  YELLOW: Color.YELLOW,
  GREEN: Color.GREEN,
  GREY_50: Color.GREY_50
};
export var PerformanceChain = function PerformanceChain(_ref) {
  var children = _ref.children;
  return /*#__PURE__*/React.createElement(Flex, null, children);
};

var PerformanceChainItem = function PerformanceChainItem(_ref2) {
  var color = _ref2.color,
      children = _ref2.children;
  return /*#__PURE__*/React.createElement(PerformanceChainItemBase, {
    color: color
  }, /*#__PURE__*/React.createElement(Flex, {
    alignItems: "center",
    justifyContent: "center",
    size: rem(16)
  }, /*#__PURE__*/React.createElement(Badge, {
    variant: BadgeVariant.DOT,
    bg: color
  })), children);
};

PerformanceChain.Item = PerformanceChainItem;
var PerformanceChainItemBase = /*#__PURE__*/styled(Flex).withConfig({
  displayName: "PerformanceChain__PerformanceChainItemBase",
  componentId: "sc-1tpychn-0"
})(themeTextStyle(TextStyle.CAPTION), {
  alignItems: 'center'
});
