// @revolut/ui-kit@11.21.2/esm/components/SliderWidget/SliderWidget.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import styled from 'styled-components';
import { Color, Space, TextStyle } from "../../constants";
import { themeSpace } from "../../utils";
import { Text } from "../Text";
import { Widget } from "../Widget";
import { VStack } from "../VStack";
import { HStack } from "../HStack";
import { Spacer } from "../Spacer";
export var SliderWidgetAlignment = {
  AUTO: 'auto',
  CENTER: 'center'
};
var SliderWidgetTitle = /*#__PURE__*/styled(HStack).withConfig({
  displayName: "SliderWidget__SliderWidgetTitle",
  componentId: "sc-1qceywo-0"
})([""]);
SliderWidgetTitle.defaultProps = {
  align: 'center',
  space: Space.S_8,
  color: Color.FOREGROUND,
  textStyle: TextStyle.TILE
};
var SliderWidgetDescription = /*#__PURE__*/styled(Text).withConfig({
  displayName: "SliderWidget__SliderWidgetDescription",
  componentId: "sc-1qceywo-1"
})(["", " + &{margin-top:", ";}"], SliderWidgetTitle, themeSpace(Space.N_8));
SliderWidgetDescription.defaultProps = {
  color: Color.GREY_TONE_50,
  variant: TextStyle.CAPTION
};

var SliderWidgetLegend = function SliderWidgetLegend(_ref) {
  var _ref$min = _ref.min,
      min = _ref$min === void 0 ? 0 : _ref$min,
      _ref$max = _ref.max,
      max = _ref$max === void 0 ? 1 : _ref$max,
      rest = _objectWithoutPropertiesLoose(_ref, ["min", "max"]);

  return /*#__PURE__*/React.createElement(HStack, _extends({
    color: Color.GREY_TONE_50
  }, rest), /*#__PURE__*/React.createElement(Text, {
    variant: TextStyle.SMALL
  }, min), /*#__PURE__*/React.createElement(Spacer, null), /*#__PURE__*/React.createElement(Text, {
    variant: TextStyle.SMALL
  }, max));
};

var SliderWidgetBase = /*#__PURE__*/styled(Widget).withConfig({
  displayName: "SliderWidget__SliderWidgetBase",
  componentId: "sc-1qceywo-2"
})(["", "{justify-content:", ";}", "{text-align:", ";}"], SliderWidgetTitle, function (props) {
  return props.align === SliderWidgetAlignment.CENTER ? 'center' : 'space-between';
}, SliderWidgetDescription, function (props) {
  return props.align === SliderWidgetAlignment.CENTER ? 'center' : 'start';
});
/**
 * Docs: {@link https://ui-kit.revolut.codes/widgets/slider|SliderWidget}
 */

export var SliderWidget = function SliderWidget(_ref2) {
  var children = _ref2.children,
      rest = _objectWithoutPropertiesLoose(_ref2, ["children"]);

  return /*#__PURE__*/React.createElement(SliderWidgetBase, _extends({
    align: SliderWidgetAlignment.AUTO,
    p: Space.S_16
  }, rest), /*#__PURE__*/React.createElement(VStack, {
    space: Space.S_8
  }, children));
};
SliderWidget.Title = SliderWidgetTitle;
SliderWidget.Description = SliderWidgetDescription;
SliderWidget.Legend = SliderWidgetLegend;
