// @revolut/ui-kit@11.21.2/esm/constants/colors/default.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { UnifiedSemanticColor } from "./unified.generated";
/** @deprecated */

export var DefaultLiteralColor = {
  BLACK: 'black',
  WHITE: 'white',
  BLUE: 'blue',
  BROWN: 'brown',
  RED: 'red',
  TERRAKOTTA: 'terrakotta',
  ORANGE: 'orange',
  YELLOW: 'yellow',
  LIME: 'lime',
  LIGHT_GREEN: 'light-green',
  GREEN: 'green',
  TEAL: 'teal',
  CYAN: 'cyan',
  SKY: 'sky',
  INDIGO: 'indigo',
  DEEP_PURPLE: 'deep-purple',
  PURPLE: 'purple',
  PINK: 'pink',
  DEEP_GREY: 'deep-grey',
  GREY_35: 'grey-35',
  GREY_50: 'grey-50',
  GREY_80: 'grey-80',
  GREY_90: 'grey-90',
  GREY_95: 'grey-95',
  GREY_100: 'grey-100',
  TRANSPARENT_BLACK: 'transparent-black',
  TRANSPARENT_GREY_35: 'transparent-grey-35',
  TRANSPARENT_GREY_50: 'transparent-grey-50',
  TRANSPARENT_GREY_80: 'transparent-grey-80',
  TRANSPARENT_GREY_90: 'transparent-grey-90',
  TRANSPARENT_GREY_95: 'transparent-grey-95',
  TRANSPARENT_GREY_100: 'transparent-grey-100',
  TRANSPARENT_WHITE: 'transparent-white',
  TRANSPARENT_BROWN: 'transparent-brown',
  TRANSPARENT_RED: 'transparent-red',
  TRANSPARENT_TERRAKOTTA: 'transparent-terrakotta',
  TRANSPARENT_ORANGE: 'transparent-orange',
  TRANSPARENT_YELLOW: 'transparent-yellow',
  TRANSPARENT_LIME: 'transparent-lime',
  TRANSPARENT_LIGHT_GREEN: 'transparent-light-green',
  TRANSPARENT_GREEN: 'transparent-green',
  TRANSPARENT_TEAL: 'transparent-teal',
  TRANSPARENT_CYAN: 'transparent-cyan',
  TRANSPARENT_SKY: 'transparent-sky',
  TRANSPARENT_INDIGO: 'transparent-indigo',
  TRANSPARENT_DEEP_PURPLE: 'transparent-deep-purple',
  TRANSPARENT_PURPLE: 'transparent-purple',
  TRANSPARENT_PINK: 'transparent-pink',
  TRANSPARENT_DEEP_GREY: 'transparent-deep-grey'
};
/** @deprecated */

/** @deprecated */
export var DefaultSemanticColor = {
  ACCENT: 'accent',
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  ERROR: 'error',
  SUCCESS: 'success',
  WARNING: 'warning',
  TRANSPARENT_ACCENT: 'transparent-accent',
  TRANSPARENT_PRIMARY: 'transparent-primary',
  TRANSPARENT_ERROR: 'transparent-error',
  TRANSPARENT_WARNING: 'transparent-warning',
  TRANSPARENT_SUCCESS: 'transparent-success',
  MODAL_OVERLAY: 'modal-overlay',
  LAYOUT_BACKGROUND: 'layout-background'
};
/** @deprecated */

/** @deprecated */
export var DefaultColor = _extends({}, DefaultLiteralColor, DefaultSemanticColor, UnifiedSemanticColor);
/** @deprecated */

/** @deprecated */
export var DefaultColorMode = {
  LIGHT: 'light'
};
/** @deprecated */
