// @revolut/ui-kit@11.21.2/esm/constants/colors/unified.generated.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

export var UnifiedLiteralColor = {
  BLUE: 'blue',
  BLUE_90: 'blue-90',
  BLUE_80: 'blue-80',
  BLUE_70: 'blue-70',
  BLUE_60: 'blue-60',
  BLUE_50: 'blue-50',
  BLUE_40: 'blue-40',
  BLUE_30: 'blue-30',
  BLUE_20: 'blue-20',
  BLUE_10: 'blue-10',
  BLUE_5: 'blue-5',
  BLUE_OPAQUE_90: 'blue-opaque-90',
  BLUE_OPAQUE_80: 'blue-opaque-80',
  BLUE_OPAQUE_70: 'blue-opaque-70',
  BLUE_OPAQUE_60: 'blue-opaque-60',
  BLUE_OPAQUE_50: 'blue-opaque-50',
  BLUE_OPAQUE_40: 'blue-opaque-40',
  BLUE_OPAQUE_30: 'blue-opaque-30',
  BLUE_OPAQUE_20: 'blue-opaque-20',
  BLUE_OPAQUE_10: 'blue-opaque-10',
  BLUE_OPAQUE_5: 'blue-opaque-5',
  PINK: 'pink',
  PINK_90: 'pink-90',
  PINK_80: 'pink-80',
  PINK_70: 'pink-70',
  PINK_60: 'pink-60',
  PINK_50: 'pink-50',
  PINK_40: 'pink-40',
  PINK_30: 'pink-30',
  PINK_20: 'pink-20',
  PINK_10: 'pink-10',
  PINK_5: 'pink-5',
  PINK_OPAQUE_90: 'pink-opaque-90',
  PINK_OPAQUE_80: 'pink-opaque-80',
  PINK_OPAQUE_70: 'pink-opaque-70',
  PINK_OPAQUE_60: 'pink-opaque-60',
  PINK_OPAQUE_50: 'pink-opaque-50',
  PINK_OPAQUE_40: 'pink-opaque-40',
  PINK_OPAQUE_30: 'pink-opaque-30',
  PINK_OPAQUE_20: 'pink-opaque-20',
  PINK_OPAQUE_10: 'pink-opaque-10',
  PINK_OPAQUE_5: 'pink-opaque-5',
  GREEN: 'green',
  GREEN_90: 'green-90',
  GREEN_80: 'green-80',
  GREEN_70: 'green-70',
  GREEN_60: 'green-60',
  GREEN_50: 'green-50',
  GREEN_40: 'green-40',
  GREEN_30: 'green-30',
  GREEN_20: 'green-20',
  GREEN_10: 'green-10',
  GREEN_5: 'green-5',
  GREEN_OPAQUE_90: 'green-opaque-90',
  GREEN_OPAQUE_80: 'green-opaque-80',
  GREEN_OPAQUE_70: 'green-opaque-70',
  GREEN_OPAQUE_60: 'green-opaque-60',
  GREEN_OPAQUE_50: 'green-opaque-50',
  GREEN_OPAQUE_40: 'green-opaque-40',
  GREEN_OPAQUE_30: 'green-opaque-30',
  GREEN_OPAQUE_20: 'green-opaque-20',
  GREEN_OPAQUE_10: 'green-opaque-10',
  GREEN_OPAQUE_5: 'green-opaque-5',
  ORANGE: 'orange',
  ORANGE_90: 'orange-90',
  ORANGE_80: 'orange-80',
  ORANGE_70: 'orange-70',
  ORANGE_60: 'orange-60',
  ORANGE_50: 'orange-50',
  ORANGE_40: 'orange-40',
  ORANGE_30: 'orange-30',
  ORANGE_20: 'orange-20',
  ORANGE_10: 'orange-10',
  ORANGE_5: 'orange-5',
  ORANGE_OPAQUE_90: 'orange-opaque-90',
  ORANGE_OPAQUE_80: 'orange-opaque-80',
  ORANGE_OPAQUE_70: 'orange-opaque-70',
  ORANGE_OPAQUE_60: 'orange-opaque-60',
  ORANGE_OPAQUE_50: 'orange-opaque-50',
  ORANGE_OPAQUE_40: 'orange-opaque-40',
  ORANGE_OPAQUE_30: 'orange-opaque-30',
  ORANGE_OPAQUE_20: 'orange-opaque-20',
  ORANGE_OPAQUE_10: 'orange-opaque-10',
  ORANGE_OPAQUE_5: 'orange-opaque-5',
  RED: 'red',
  RED_90: 'red-90',
  RED_80: 'red-80',
  RED_70: 'red-70',
  RED_60: 'red-60',
  RED_50: 'red-50',
  RED_40: 'red-40',
  RED_30: 'red-30',
  RED_20: 'red-20',
  RED_10: 'red-10',
  RED_5: 'red-5',
  RED_OPAQUE_90: 'red-opaque-90',
  RED_OPAQUE_80: 'red-opaque-80',
  RED_OPAQUE_70: 'red-opaque-70',
  RED_OPAQUE_60: 'red-opaque-60',
  RED_OPAQUE_50: 'red-opaque-50',
  RED_OPAQUE_40: 'red-opaque-40',
  RED_OPAQUE_30: 'red-opaque-30',
  RED_OPAQUE_20: 'red-opaque-20',
  RED_OPAQUE_10: 'red-opaque-10',
  RED_OPAQUE_5: 'red-opaque-5',
  DEEP_PINK: 'deep-pink',
  DEEP_PINK_90: 'deep-pink-90',
  DEEP_PINK_80: 'deep-pink-80',
  DEEP_PINK_70: 'deep-pink-70',
  DEEP_PINK_60: 'deep-pink-60',
  DEEP_PINK_50: 'deep-pink-50',
  DEEP_PINK_40: 'deep-pink-40',
  DEEP_PINK_30: 'deep-pink-30',
  DEEP_PINK_20: 'deep-pink-20',
  DEEP_PINK_10: 'deep-pink-10',
  DEEP_PINK_5: 'deep-pink-5',
  DEEP_PINK_OPAQUE_90: 'deep-pink-opaque-90',
  DEEP_PINK_OPAQUE_80: 'deep-pink-opaque-80',
  DEEP_PINK_OPAQUE_70: 'deep-pink-opaque-70',
  DEEP_PINK_OPAQUE_60: 'deep-pink-opaque-60',
  DEEP_PINK_OPAQUE_50: 'deep-pink-opaque-50',
  DEEP_PINK_OPAQUE_40: 'deep-pink-opaque-40',
  DEEP_PINK_OPAQUE_30: 'deep-pink-opaque-30',
  DEEP_PINK_OPAQUE_20: 'deep-pink-opaque-20',
  DEEP_PINK_OPAQUE_10: 'deep-pink-opaque-10',
  DEEP_PINK_OPAQUE_5: 'deep-pink-opaque-5',
  TERRACOTTA: 'terracotta',
  TERRACOTTA_90: 'terracotta-90',
  TERRACOTTA_80: 'terracotta-80',
  TERRACOTTA_70: 'terracotta-70',
  TERRACOTTA_60: 'terracotta-60',
  TERRACOTTA_50: 'terracotta-50',
  TERRACOTTA_40: 'terracotta-40',
  TERRACOTTA_30: 'terracotta-30',
  TERRACOTTA_20: 'terracotta-20',
  TERRACOTTA_10: 'terracotta-10',
  TERRACOTTA_5: 'terracotta-5',
  TERRACOTTA_OPAQUE_90: 'terracotta-opaque-90',
  TERRACOTTA_OPAQUE_80: 'terracotta-opaque-80',
  TERRACOTTA_OPAQUE_70: 'terracotta-opaque-70',
  TERRACOTTA_OPAQUE_60: 'terracotta-opaque-60',
  TERRACOTTA_OPAQUE_50: 'terracotta-opaque-50',
  TERRACOTTA_OPAQUE_40: 'terracotta-opaque-40',
  TERRACOTTA_OPAQUE_30: 'terracotta-opaque-30',
  TERRACOTTA_OPAQUE_20: 'terracotta-opaque-20',
  TERRACOTTA_OPAQUE_10: 'terracotta-opaque-10',
  TERRACOTTA_OPAQUE_5: 'terracotta-opaque-5',
  YELLOW: 'yellow',
  YELLOW_90: 'yellow-90',
  YELLOW_80: 'yellow-80',
  YELLOW_70: 'yellow-70',
  YELLOW_60: 'yellow-60',
  YELLOW_50: 'yellow-50',
  YELLOW_40: 'yellow-40',
  YELLOW_30: 'yellow-30',
  YELLOW_20: 'yellow-20',
  YELLOW_10: 'yellow-10',
  YELLOW_5: 'yellow-5',
  YELLOW_OPAQUE_90: 'yellow-opaque-90',
  YELLOW_OPAQUE_80: 'yellow-opaque-80',
  YELLOW_OPAQUE_70: 'yellow-opaque-70',
  YELLOW_OPAQUE_60: 'yellow-opaque-60',
  YELLOW_OPAQUE_50: 'yellow-opaque-50',
  YELLOW_OPAQUE_40: 'yellow-opaque-40',
  YELLOW_OPAQUE_30: 'yellow-opaque-30',
  YELLOW_OPAQUE_20: 'yellow-opaque-20',
  YELLOW_OPAQUE_10: 'yellow-opaque-10',
  YELLOW_OPAQUE_5: 'yellow-opaque-5',
  LIME: 'lime',
  LIME_90: 'lime-90',
  LIME_80: 'lime-80',
  LIME_70: 'lime-70',
  LIME_60: 'lime-60',
  LIME_50: 'lime-50',
  LIME_40: 'lime-40',
  LIME_30: 'lime-30',
  LIME_20: 'lime-20',
  LIME_10: 'lime-10',
  LIME_5: 'lime-5',
  LIME_OPAQUE_90: 'lime-opaque-90',
  LIME_OPAQUE_80: 'lime-opaque-80',
  LIME_OPAQUE_70: 'lime-opaque-70',
  LIME_OPAQUE_60: 'lime-opaque-60',
  LIME_OPAQUE_50: 'lime-opaque-50',
  LIME_OPAQUE_40: 'lime-opaque-40',
  LIME_OPAQUE_30: 'lime-opaque-30',
  LIME_OPAQUE_20: 'lime-opaque-20',
  LIME_OPAQUE_10: 'lime-opaque-10',
  LIME_OPAQUE_5: 'lime-opaque-5',
  LIGHT_GREEN: 'light-green',
  LIGHT_GREEN_90: 'light-green-90',
  LIGHT_GREEN_80: 'light-green-80',
  LIGHT_GREEN_70: 'light-green-70',
  LIGHT_GREEN_60: 'light-green-60',
  LIGHT_GREEN_50: 'light-green-50',
  LIGHT_GREEN_40: 'light-green-40',
  LIGHT_GREEN_30: 'light-green-30',
  LIGHT_GREEN_20: 'light-green-20',
  LIGHT_GREEN_10: 'light-green-10',
  LIGHT_GREEN_5: 'light-green-5',
  LIGHT_GREEN_OPAQUE_90: 'light-green-opaque-90',
  LIGHT_GREEN_OPAQUE_80: 'light-green-opaque-80',
  LIGHT_GREEN_OPAQUE_70: 'light-green-opaque-70',
  LIGHT_GREEN_OPAQUE_60: 'light-green-opaque-60',
  LIGHT_GREEN_OPAQUE_50: 'light-green-opaque-50',
  LIGHT_GREEN_OPAQUE_40: 'light-green-opaque-40',
  LIGHT_GREEN_OPAQUE_30: 'light-green-opaque-30',
  LIGHT_GREEN_OPAQUE_20: 'light-green-opaque-20',
  LIGHT_GREEN_OPAQUE_10: 'light-green-opaque-10',
  LIGHT_GREEN_OPAQUE_5: 'light-green-opaque-5',
  TEAL: 'teal',
  TEAL_90: 'teal-90',
  TEAL_80: 'teal-80',
  TEAL_70: 'teal-70',
  TEAL_60: 'teal-60',
  TEAL_50: 'teal-50',
  TEAL_40: 'teal-40',
  TEAL_30: 'teal-30',
  TEAL_20: 'teal-20',
  TEAL_10: 'teal-10',
  TEAL_5: 'teal-5',
  TEAL_OPAQUE_90: 'teal-opaque-90',
  TEAL_OPAQUE_80: 'teal-opaque-80',
  TEAL_OPAQUE_70: 'teal-opaque-70',
  TEAL_OPAQUE_60: 'teal-opaque-60',
  TEAL_OPAQUE_50: 'teal-opaque-50',
  TEAL_OPAQUE_40: 'teal-opaque-40',
  TEAL_OPAQUE_30: 'teal-opaque-30',
  TEAL_OPAQUE_20: 'teal-opaque-20',
  TEAL_OPAQUE_10: 'teal-opaque-10',
  TEAL_OPAQUE_5: 'teal-opaque-5',
  CYAN: 'cyan',
  CYAN_90: 'cyan-90',
  CYAN_80: 'cyan-80',
  CYAN_70: 'cyan-70',
  CYAN_60: 'cyan-60',
  CYAN_50: 'cyan-50',
  CYAN_40: 'cyan-40',
  CYAN_30: 'cyan-30',
  CYAN_20: 'cyan-20',
  CYAN_10: 'cyan-10',
  CYAN_5: 'cyan-5',
  CYAN_OPAQUE_90: 'cyan-opaque-90',
  CYAN_OPAQUE_80: 'cyan-opaque-80',
  CYAN_OPAQUE_70: 'cyan-opaque-70',
  CYAN_OPAQUE_60: 'cyan-opaque-60',
  CYAN_OPAQUE_50: 'cyan-opaque-50',
  CYAN_OPAQUE_40: 'cyan-opaque-40',
  CYAN_OPAQUE_30: 'cyan-opaque-30',
  CYAN_OPAQUE_20: 'cyan-opaque-20',
  CYAN_OPAQUE_10: 'cyan-opaque-10',
  CYAN_OPAQUE_5: 'cyan-opaque-5',
  LIGHT_BLUE: 'light-blue',
  LIGHT_BLUE_90: 'light-blue-90',
  LIGHT_BLUE_80: 'light-blue-80',
  LIGHT_BLUE_70: 'light-blue-70',
  LIGHT_BLUE_60: 'light-blue-60',
  LIGHT_BLUE_50: 'light-blue-50',
  LIGHT_BLUE_40: 'light-blue-40',
  LIGHT_BLUE_30: 'light-blue-30',
  LIGHT_BLUE_20: 'light-blue-20',
  LIGHT_BLUE_10: 'light-blue-10',
  LIGHT_BLUE_5: 'light-blue-5',
  LIGHT_BLUE_OPAQUE_90: 'light-blue-opaque-90',
  LIGHT_BLUE_OPAQUE_80: 'light-blue-opaque-80',
  LIGHT_BLUE_OPAQUE_70: 'light-blue-opaque-70',
  LIGHT_BLUE_OPAQUE_60: 'light-blue-opaque-60',
  LIGHT_BLUE_OPAQUE_50: 'light-blue-opaque-50',
  LIGHT_BLUE_OPAQUE_40: 'light-blue-opaque-40',
  LIGHT_BLUE_OPAQUE_30: 'light-blue-opaque-30',
  LIGHT_BLUE_OPAQUE_20: 'light-blue-opaque-20',
  LIGHT_BLUE_OPAQUE_10: 'light-blue-opaque-10',
  LIGHT_BLUE_OPAQUE_5: 'light-blue-opaque-5',
  INDIGO: 'indigo',
  INDIGO_90: 'indigo-90',
  INDIGO_80: 'indigo-80',
  INDIGO_70: 'indigo-70',
  INDIGO_60: 'indigo-60',
  INDIGO_50: 'indigo-50',
  INDIGO_40: 'indigo-40',
  INDIGO_30: 'indigo-30',
  INDIGO_20: 'indigo-20',
  INDIGO_10: 'indigo-10',
  INDIGO_5: 'indigo-5',
  INDIGO_OPAQUE_90: 'indigo-opaque-90',
  INDIGO_OPAQUE_80: 'indigo-opaque-80',
  INDIGO_OPAQUE_70: 'indigo-opaque-70',
  INDIGO_OPAQUE_60: 'indigo-opaque-60',
  INDIGO_OPAQUE_50: 'indigo-opaque-50',
  INDIGO_OPAQUE_40: 'indigo-opaque-40',
  INDIGO_OPAQUE_30: 'indigo-opaque-30',
  INDIGO_OPAQUE_20: 'indigo-opaque-20',
  INDIGO_OPAQUE_10: 'indigo-opaque-10',
  INDIGO_OPAQUE_5: 'indigo-opaque-5',
  DEEP_PURPLE: 'deep-purple',
  DEEP_PURPLE_90: 'deep-purple-90',
  DEEP_PURPLE_80: 'deep-purple-80',
  DEEP_PURPLE_70: 'deep-purple-70',
  DEEP_PURPLE_60: 'deep-purple-60',
  DEEP_PURPLE_50: 'deep-purple-50',
  DEEP_PURPLE_40: 'deep-purple-40',
  DEEP_PURPLE_30: 'deep-purple-30',
  DEEP_PURPLE_20: 'deep-purple-20',
  DEEP_PURPLE_10: 'deep-purple-10',
  DEEP_PURPLE_5: 'deep-purple-5',
  DEEP_PURPLE_OPAQUE_90: 'deep-purple-opaque-90',
  DEEP_PURPLE_OPAQUE_80: 'deep-purple-opaque-80',
  DEEP_PURPLE_OPAQUE_70: 'deep-purple-opaque-70',
  DEEP_PURPLE_OPAQUE_60: 'deep-purple-opaque-60',
  DEEP_PURPLE_OPAQUE_50: 'deep-purple-opaque-50',
  DEEP_PURPLE_OPAQUE_40: 'deep-purple-opaque-40',
  DEEP_PURPLE_OPAQUE_30: 'deep-purple-opaque-30',
  DEEP_PURPLE_OPAQUE_20: 'deep-purple-opaque-20',
  DEEP_PURPLE_OPAQUE_10: 'deep-purple-opaque-10',
  DEEP_PURPLE_OPAQUE_5: 'deep-purple-opaque-5',
  PURPLE: 'purple',
  PURPLE_90: 'purple-90',
  PURPLE_80: 'purple-80',
  PURPLE_70: 'purple-70',
  PURPLE_60: 'purple-60',
  PURPLE_50: 'purple-50',
  PURPLE_40: 'purple-40',
  PURPLE_30: 'purple-30',
  PURPLE_20: 'purple-20',
  PURPLE_10: 'purple-10',
  PURPLE_5: 'purple-5',
  PURPLE_OPAQUE_90: 'purple-opaque-90',
  PURPLE_OPAQUE_80: 'purple-opaque-80',
  PURPLE_OPAQUE_70: 'purple-opaque-70',
  PURPLE_OPAQUE_60: 'purple-opaque-60',
  PURPLE_OPAQUE_50: 'purple-opaque-50',
  PURPLE_OPAQUE_40: 'purple-opaque-40',
  PURPLE_OPAQUE_30: 'purple-opaque-30',
  PURPLE_OPAQUE_20: 'purple-opaque-20',
  PURPLE_OPAQUE_10: 'purple-opaque-10',
  PURPLE_OPAQUE_5: 'purple-opaque-5',
  BROWN: 'brown',
  BROWN_90: 'brown-90',
  BROWN_80: 'brown-80',
  BROWN_70: 'brown-70',
  BROWN_60: 'brown-60',
  BROWN_50: 'brown-50',
  BROWN_40: 'brown-40',
  BROWN_30: 'brown-30',
  BROWN_20: 'brown-20',
  BROWN_10: 'brown-10',
  BROWN_5: 'brown-5',
  BROWN_OPAQUE_90: 'brown-opaque-90',
  BROWN_OPAQUE_80: 'brown-opaque-80',
  BROWN_OPAQUE_70: 'brown-opaque-70',
  BROWN_OPAQUE_60: 'brown-opaque-60',
  BROWN_OPAQUE_50: 'brown-opaque-50',
  BROWN_OPAQUE_40: 'brown-opaque-40',
  BROWN_OPAQUE_30: 'brown-opaque-30',
  BROWN_OPAQUE_20: 'brown-opaque-20',
  BROWN_OPAQUE_10: 'brown-opaque-10',
  BROWN_OPAQUE_5: 'brown-opaque-5',
  DEEP_GREY: 'deep-grey',
  DEEP_GREY_90: 'deep-grey-90',
  DEEP_GREY_80: 'deep-grey-80',
  DEEP_GREY_70: 'deep-grey-70',
  DEEP_GREY_60: 'deep-grey-60',
  DEEP_GREY_50: 'deep-grey-50',
  DEEP_GREY_40: 'deep-grey-40',
  DEEP_GREY_30: 'deep-grey-30',
  DEEP_GREY_20: 'deep-grey-20',
  DEEP_GREY_10: 'deep-grey-10',
  DEEP_GREY_5: 'deep-grey-5',
  DEEP_GREY_OPAQUE_90: 'deep-grey-opaque-90',
  DEEP_GREY_OPAQUE_80: 'deep-grey-opaque-80',
  DEEP_GREY_OPAQUE_70: 'deep-grey-opaque-70',
  DEEP_GREY_OPAQUE_60: 'deep-grey-opaque-60',
  DEEP_GREY_OPAQUE_50: 'deep-grey-opaque-50',
  DEEP_GREY_OPAQUE_40: 'deep-grey-opaque-40',
  DEEP_GREY_OPAQUE_30: 'deep-grey-opaque-30',
  DEEP_GREY_OPAQUE_20: 'deep-grey-opaque-20',
  DEEP_GREY_OPAQUE_10: 'deep-grey-opaque-10',
  DEEP_GREY_OPAQUE_5: 'deep-grey-opaque-5',
  BLACK: 'black',
  BLACK_90: 'black-90',
  BLACK_80: 'black-80',
  BLACK_70: 'black-70',
  BLACK_60: 'black-60',
  BLACK_50: 'black-50',
  BLACK_40: 'black-40',
  BLACK_30: 'black-30',
  BLACK_20: 'black-20',
  BLACK_10: 'black-10',
  BLACK_5: 'black-5',
  BLACK_OPAQUE_90: 'black-opaque-90',
  BLACK_OPAQUE_80: 'black-opaque-80',
  BLACK_OPAQUE_70: 'black-opaque-70',
  BLACK_OPAQUE_60: 'black-opaque-60',
  BLACK_OPAQUE_50: 'black-opaque-50',
  BLACK_OPAQUE_40: 'black-opaque-40',
  BLACK_OPAQUE_30: 'black-opaque-30',
  BLACK_OPAQUE_20: 'black-opaque-20',
  BLACK_OPAQUE_10: 'black-opaque-10',
  BLACK_OPAQUE_5: 'black-opaque-5',
  GREY_50: 'grey-50',
  GREY_50_90: 'grey-50-90',
  GREY_50_80: 'grey-50-80',
  GREY_50_70: 'grey-50-70',
  GREY_50_60: 'grey-50-60',
  GREY_50_50: 'grey-50-50',
  GREY_50_40: 'grey-50-40',
  GREY_50_30: 'grey-50-30',
  GREY_50_20: 'grey-50-20',
  GREY_50_10: 'grey-50-10',
  GREY_50_5: 'grey-50-5',
  GREY_50_OPAQUE_90: 'grey-50-opaque-90',
  GREY_50_OPAQUE_80: 'grey-50-opaque-80',
  GREY_50_OPAQUE_70: 'grey-50-opaque-70',
  GREY_50_OPAQUE_60: 'grey-50-opaque-60',
  GREY_50_OPAQUE_50: 'grey-50-opaque-50',
  GREY_50_OPAQUE_40: 'grey-50-opaque-40',
  GREY_50_OPAQUE_30: 'grey-50-opaque-30',
  GREY_50_OPAQUE_20: 'grey-50-opaque-20',
  GREY_50_OPAQUE_10: 'grey-50-opaque-10',
  GREY_50_OPAQUE_5: 'grey-50-opaque-5',
  GREY_20: 'grey-20',
  GREY_20_90: 'grey-20-90',
  GREY_20_80: 'grey-20-80',
  GREY_20_70: 'grey-20-70',
  GREY_20_60: 'grey-20-60',
  GREY_20_50: 'grey-20-50',
  GREY_20_40: 'grey-20-40',
  GREY_20_30: 'grey-20-30',
  GREY_20_20: 'grey-20-20',
  GREY_20_10: 'grey-20-10',
  GREY_20_5: 'grey-20-5',
  GREY_20_OPAQUE_90: 'grey-20-opaque-90',
  GREY_20_OPAQUE_80: 'grey-20-opaque-80',
  GREY_20_OPAQUE_70: 'grey-20-opaque-70',
  GREY_20_OPAQUE_60: 'grey-20-opaque-60',
  GREY_20_OPAQUE_50: 'grey-20-opaque-50',
  GREY_20_OPAQUE_40: 'grey-20-opaque-40',
  GREY_20_OPAQUE_30: 'grey-20-opaque-30',
  GREY_20_OPAQUE_20: 'grey-20-opaque-20',
  GREY_20_OPAQUE_10: 'grey-20-opaque-10',
  GREY_20_OPAQUE_5: 'grey-20-opaque-5',
  GREY_10: 'grey-10',
  GREY_10_90: 'grey-10-90',
  GREY_10_80: 'grey-10-80',
  GREY_10_70: 'grey-10-70',
  GREY_10_60: 'grey-10-60',
  GREY_10_50: 'grey-10-50',
  GREY_10_40: 'grey-10-40',
  GREY_10_30: 'grey-10-30',
  GREY_10_20: 'grey-10-20',
  GREY_10_10: 'grey-10-10',
  GREY_10_5: 'grey-10-5',
  GREY_10_OPAQUE_90: 'grey-10-opaque-90',
  GREY_10_OPAQUE_80: 'grey-10-opaque-80',
  GREY_10_OPAQUE_70: 'grey-10-opaque-70',
  GREY_10_OPAQUE_60: 'grey-10-opaque-60',
  GREY_10_OPAQUE_50: 'grey-10-opaque-50',
  GREY_10_OPAQUE_40: 'grey-10-opaque-40',
  GREY_10_OPAQUE_30: 'grey-10-opaque-30',
  GREY_10_OPAQUE_20: 'grey-10-opaque-20',
  GREY_10_OPAQUE_10: 'grey-10-opaque-10',
  GREY_10_OPAQUE_5: 'grey-10-opaque-5',
  GREY_5: 'grey-5',
  GREY_5_90: 'grey-5-90',
  GREY_5_80: 'grey-5-80',
  GREY_5_70: 'grey-5-70',
  GREY_5_60: 'grey-5-60',
  GREY_5_50: 'grey-5-50',
  GREY_5_40: 'grey-5-40',
  GREY_5_30: 'grey-5-30',
  GREY_5_20: 'grey-5-20',
  GREY_5_10: 'grey-5-10',
  GREY_5_5: 'grey-5-5',
  GREY_5_OPAQUE_90: 'grey-5-opaque-90',
  GREY_5_OPAQUE_80: 'grey-5-opaque-80',
  GREY_5_OPAQUE_70: 'grey-5-opaque-70',
  GREY_5_OPAQUE_60: 'grey-5-opaque-60',
  GREY_5_OPAQUE_50: 'grey-5-opaque-50',
  GREY_5_OPAQUE_40: 'grey-5-opaque-40',
  GREY_5_OPAQUE_30: 'grey-5-opaque-30',
  GREY_5_OPAQUE_20: 'grey-5-opaque-20',
  GREY_5_OPAQUE_10: 'grey-5-opaque-10',
  GREY_5_OPAQUE_5: 'grey-5-opaque-5',
  GREY_8: 'grey-8',
  GREY_8_90: 'grey-8-90',
  GREY_8_80: 'grey-8-80',
  GREY_8_70: 'grey-8-70',
  GREY_8_60: 'grey-8-60',
  GREY_8_50: 'grey-8-50',
  GREY_8_40: 'grey-8-40',
  GREY_8_30: 'grey-8-30',
  GREY_8_20: 'grey-8-20',
  GREY_8_10: 'grey-8-10',
  GREY_8_5: 'grey-8-5',
  GREY_8_OPAQUE_90: 'grey-8-opaque-90',
  GREY_8_OPAQUE_80: 'grey-8-opaque-80',
  GREY_8_OPAQUE_70: 'grey-8-opaque-70',
  GREY_8_OPAQUE_60: 'grey-8-opaque-60',
  GREY_8_OPAQUE_50: 'grey-8-opaque-50',
  GREY_8_OPAQUE_40: 'grey-8-opaque-40',
  GREY_8_OPAQUE_30: 'grey-8-opaque-30',
  GREY_8_OPAQUE_20: 'grey-8-opaque-20',
  GREY_8_OPAQUE_10: 'grey-8-opaque-10',
  GREY_8_OPAQUE_5: 'grey-8-opaque-5',
  GREY_2: 'grey-2',
  GREY_2_90: 'grey-2-90',
  GREY_2_80: 'grey-2-80',
  GREY_2_70: 'grey-2-70',
  GREY_2_60: 'grey-2-60',
  GREY_2_50: 'grey-2-50',
  GREY_2_40: 'grey-2-40',
  GREY_2_30: 'grey-2-30',
  GREY_2_20: 'grey-2-20',
  GREY_2_10: 'grey-2-10',
  GREY_2_5: 'grey-2-5',
  GREY_2_OPAQUE_90: 'grey-2-opaque-90',
  GREY_2_OPAQUE_80: 'grey-2-opaque-80',
  GREY_2_OPAQUE_70: 'grey-2-opaque-70',
  GREY_2_OPAQUE_60: 'grey-2-opaque-60',
  GREY_2_OPAQUE_50: 'grey-2-opaque-50',
  GREY_2_OPAQUE_40: 'grey-2-opaque-40',
  GREY_2_OPAQUE_30: 'grey-2-opaque-30',
  GREY_2_OPAQUE_20: 'grey-2-opaque-20',
  GREY_2_OPAQUE_10: 'grey-2-opaque-10',
  GREY_2_OPAQUE_5: 'grey-2-opaque-5',
  WHITE: 'white',
  WHITE_90: 'white-90',
  WHITE_80: 'white-80',
  WHITE_70: 'white-70',
  WHITE_60: 'white-60',
  WHITE_50: 'white-50',
  WHITE_40: 'white-40',
  WHITE_30: 'white-30',
  WHITE_20: 'white-20',
  WHITE_10: 'white-10',
  WHITE_5: 'white-5',
  WHITE_OPAQUE_90: 'white-opaque-90',
  WHITE_OPAQUE_80: 'white-opaque-80',
  WHITE_OPAQUE_70: 'white-opaque-70',
  WHITE_OPAQUE_60: 'white-opaque-60',
  WHITE_OPAQUE_50: 'white-opaque-50',
  WHITE_OPAQUE_40: 'white-opaque-40',
  WHITE_OPAQUE_30: 'white-opaque-30',
  WHITE_OPAQUE_20: 'white-opaque-20',
  WHITE_OPAQUE_10: 'white-opaque-10',
  WHITE_OPAQUE_5: 'white-opaque-5',
  ACTION_BLUE: 'action-blue',
  ACTION_BLUE_90: 'action-blue-90',
  ACTION_BLUE_80: 'action-blue-80',
  ACTION_BLUE_70: 'action-blue-70',
  ACTION_BLUE_60: 'action-blue-60',
  ACTION_BLUE_50: 'action-blue-50',
  ACTION_BLUE_40: 'action-blue-40',
  ACTION_BLUE_30: 'action-blue-30',
  ACTION_BLUE_20: 'action-blue-20',
  ACTION_BLUE_10: 'action-blue-10',
  ACTION_BLUE_5: 'action-blue-5',
  ACTION_BLUE_OPAQUE_90: 'action-blue-opaque-90',
  ACTION_BLUE_OPAQUE_80: 'action-blue-opaque-80',
  ACTION_BLUE_OPAQUE_70: 'action-blue-opaque-70',
  ACTION_BLUE_OPAQUE_60: 'action-blue-opaque-60',
  ACTION_BLUE_OPAQUE_50: 'action-blue-opaque-50',
  ACTION_BLUE_OPAQUE_40: 'action-blue-opaque-40',
  ACTION_BLUE_OPAQUE_30: 'action-blue-opaque-30',
  ACTION_BLUE_OPAQUE_20: 'action-blue-opaque-20',
  ACTION_BLUE_OPAQUE_10: 'action-blue-opaque-10',
  ACTION_BLUE_OPAQUE_5: 'action-blue-opaque-5'
};
export var UnifiedSemanticColor = {
  SEARCH_BACKGROUND: 'search-background',
  SEGMENTED_WIDGET_BACKGROUND: 'segmented-widget-background',
  SEGMENTED_BACKGROUND: 'segmented-background',
  SEGMENTED_ACTIVE: 'segmented-active',
  ACTION_PHOTO_HEADER_TEXT: 'action-photo-header-text',
  ACTION_PHOTO_HEADER_BACKGROUND: 'action-photo-header-background',
  RED_ACTION_BACKGROUND: 'red-action-background',
  ACTION_BACKGROUND: 'action-background',
  WIDGET_BACKGROUND: 'widget-background',
  GROUPED_BACKGROUND: 'grouped-background',
  BACKGROUND: 'background',
  GREY_TONE_2: 'grey-tone-2',
  GREY_TONE_5: 'grey-tone-5',
  GREY_TONE_8: 'grey-tone-8',
  GREY_TONE_10: 'grey-tone-10',
  GREY_TONE_20: 'grey-tone-20',
  GREY_TONE_50: 'grey-tone-50',
  FOREGROUND: 'foreground',
  ACTION_LABEL: 'action-label',
  INPUT_ERROR: 'input-error',
  INPUT_ERROR_FOCUS: 'input-error-focus'
};
export var UnifiedCustomColor = {
  LAYOUT_BACKGROUND: 'layout-background',
  BOTTOM_NAV: 'bottom-nav',
  MODAL_OVERLAY: 'modal-overlay',
  PRIMARY: 'primary',
  SUCCESS: 'success',
  WARNING: 'warning',
  ERROR: 'error'
};
export var UnifiedColor = _extends({}, UnifiedLiteralColor, UnifiedSemanticColor, UnifiedCustomColor);
export var UnifiedColorMode = {
  LIGHT: 'light',
  DARK: 'dark',
  LIGHT_ELEVATED: 'light-elevated',
  DARK_ELEVATED: 'dark-elevated'
};
