// @revolut/ui-kit@11.21.2/esm/components/Button/Button.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

import styled from 'styled-components';
import { color, height, maxWidth, space, textAlign, width } from 'styled-system';
import { buttonElevation, hasDot, radius, textStyle, variant } from "../../styles";
import { ButtonLayout } from "../ButtonLayout";
import { DEFAULT_ELEMENT } from "../ButtonBase";
import { ButtonSizeStyle, ButtonVariant, TextStyle } from "../../constants";

var attributes = function attributes(props) {
  if (props.children != null || props.width != null) {
    return {};
  }

  if (props.use === undefined || props.use === DEFAULT_ELEMENT) {
    return {
      width: 'auto'
    };
  }

  return {
    width: 'max-content',
    maxWidth: '100%'
  };
};

export var Button = /*#__PURE__*/styled(ButtonLayout).attrs(attributes).withConfig({
  displayName: "Button",
  componentId: "sc-1gvtpv4-0"
})(["", " ", " ", " ", " ", " ", " ", " ", " ", " ", " ", " ", ""], variant({
  prop: 'size',
  scale: 'buttonSizeStyles'
}), variant({
  scale: 'buttonStyles'
}), buttonElevation, textStyle, textAlign, radius, color, space, height, width, maxWidth, hasDot({
  right: -2,
  top: -2
}));
Button.defaultProps = {
  variant: ButtonVariant.DEFAULT,
  size: ButtonSizeStyle.MD,
  textStyle: TextStyle.BUTTON,
  textAlign: 'center',
  disabled: false
};
