// @revolut/ui-kit@11.21.2/esm/components/Button/ButtonSkeleton.js
/**
* Copyright 2018-present Revolut LTD
* 
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
* 
*     http://www.apache.org/licenses/LICENSE-2.0
* 
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
* 
**/

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { rem } from 'polished';
import React from 'react';
import { useTheme } from 'styled-components';
import { Color, Radius } from "../../constants";
import { rgbaSafe, themeColor } from "../../utils";
import { Skeleton } from "../Skeleton";
export var ButtonSkeleton = function ButtonSkeleton(props) {
  var theme = useTheme();
  return /*#__PURE__*/React.createElement(Skeleton, _extends({
    width: rem(82),
    height: rem(36),
    radius: Radius.BUTTON_SM,
    shimmer: rgbaSafe(themeColor(Color.BLUE)({
      theme: theme
    }), 0.03),
    bg: Color.ACTION_BACKGROUND
  }, props));
};
